<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yixuan (Elisa) Chen" />

<meta name="date" content="2025-05-30" />

<title>Relation between lung gene expression with Asthma PRS</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/yixuan-chen-elisa/myproject">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Relation between lung gene expression with
Asthma PRS</h1>
<h4 class="author">Yixuan (Elisa) Chen</h4>
<h4 class="date">2025-05-30</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>prs/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250417code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250417)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250417code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250417)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyixuanchenelisaprstreee8f339ea00f96826f28172da445b2432ee64c45btargetblanke8f339ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/yixuan-chen-elisa/prs/tree/e8f339ea00f96826f28172da445b2432ee64c45b" target="_blank">e8f339e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomyixuanchenelisaprstreee8f339ea00f96826f28172da445b2432ee64c45btargetblanke8f339ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/yixuan-chen-elisa/prs/tree/e8f339ea00f96826f28172da445b2432ee64c45b" target="_blank">e8f339e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  analysis/asthma_lung/
    Untracked:  analysis/asthma_wb/
    Untracked:  analysis/continuous_artery/
    Untracked:  analysis/continuous_wb/
    Untracked:  analysis/continuous_wb_m1/
    Untracked:  analysis/continuous_wb_m2/
    Untracked:  analysis/metadata.txt
    Untracked:  analysis/metadata_artery.txt
    Untracked:  analysis/metadata_artery_quantile.txt
    Untracked:  analysis/metadata_lung_asthma.txt
    Untracked:  analysis/metadata_lung_asthma_quantile.txt
    Untracked:  analysis/metadata_quantile.txt
    Untracked:  analysis/metadata_wb_asthma.txt
    Untracked:  analysis/metadata_wb_asthma_quantile.txt
    Untracked:  analysis/normalized_counts.rda
    Untracked:  analysis/prs_asthma.txt
    Untracked:  analysis/quantile_artery/
    Untracked:  analysis/quantile_wb/
    Untracked:  analysis/quantile_wb_m1/
    Untracked:  analysis/quantile_wb_m2/
    Untracked:  analysis/vst norm counts.rda
    Untracked:  data/Artery_Aorta.v8.covariates.txt
    Untracked:  data/GTEx_v8.bk
    Untracked:  data/GTEx_v8.rds
    Untracked:  data/Lung.v8.covariates.txt
    Untracked:  data/PGS001787.txt
    Untracked:  data/PGS001787_hmPOS_GRCh38.txt
    Untracked:  data/T2D_hmPOS_GRCh38.txt
    Untracked:  data/Whole_Blood.v8.covariates.txt
    Untracked:  data/blood_cell/
    Untracked:  data/endotype_sumstats/
    Untracked:  data/gene_reads_2017-06-05_v8_artery_aorta.gct
    Untracked:  data/gene_reads_2017-06-05_v8_lung.gct
    Untracked:  data/gene_reads_2017-06-05_v8_whole_blood.gct
    Untracked:  data/immune/
    Untracked:  data/pca.eigenval
    Untracked:  data/pca.eigenvec
    Untracked:  data/protein-coding_gene.txt
    Untracked:  lung_exp_count.csv
    Untracked:  metadata_asthma_m2.csv
    Untracked:  metadata_lung_asthma_M2.csv
    Untracked:  metadata_lung_asthma_quantile_M2.csv
    Untracked:  metadata_quantile_asthma_m2.csv
    Untracked:  wb_exp_count.csv

Unstaged changes:
    Modified:   analysis/PRS.Rmd
    Deleted:    analysis/QC.Rmd
    Modified:   analysis/asthma_wb.Rmd
    Deleted:    analysis/normalized_counts.txt
    Modified:   analysis/prs_blood_cell.txt
    Modified:   prs.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/asthma_lung.Rmd</code>) and HTML
(<code>docs/asthma_lung.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/e8f339ea00f96826f28172da445b2432ee64c45b/analysis/asthma_lung.Rmd" target="_blank">e8f339e</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-06-02
</td>
<td>
workflowr::wflow_publish("analysis/asthma_lung.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/36ca2c8b754243b6309f8be5580a0b35181850e3/docs/asthma_lung.html" target="_blank">36ca2c8</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/51461273f5c8272ef874a7375acb1f72a67e36e2/analysis/asthma_lung.Rmd" target="_blank">5146127</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-30
</td>
<td>
workflowr::wflow_publish(c("analysis/index.Rmd",
"analysis/asthma_wb.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="correlation-between-prs-expression-pcs" class="section level1">
<h1>Correlation between PRS &amp; expression PCs</h1>
<pre class="r"><code># load prs &amp; pcs
metadata_file &lt;- &quot;analysis/metadata_lung_asthma.txt&quot;
metadata &lt;- read.csv(metadata_file, header = T, sep = &quot;\t&quot;, stringsAsFactors = T)
metadata$sex &lt;- as.factor(metadata$sex)

traits &lt;- metadata$asthma
pc &lt;- metadata[, c(1:5)]

# Calculate the correlation between each trait and each PC
correlation_matrix &lt;- cor(traits, pc)
range(correlation_matrix)</code></pre>
<pre><code>[1] -0.1441741  0.1685583</code></pre>
<pre class="r"><code>correlation_matrix &lt;- t(correlation_matrix)
correlation_matrix</code></pre>
<pre><code>            [,1]
PC1  0.168558314
PC2  0.005256598
PC3 -0.144174130
PC4  0.083689022
PC5 -0.083910604</code></pre>
</div>
<div id="model-1-expression-prs-sex-expression-pcs"
class="section level1">
<h1>Model 1: expression ~ PRS + sex + expression PCs</h1>
<div id="continuous-prs" class="section level2">
<h2>Continuous PRS</h2>
<div id="deseq2-differential-expression" class="section level3">
<h3>DESeq2 Differential Expression</h3>
<pre class="r"><code># Load the gene expression data
gene_expr_file &lt;- &quot;data/gene_reads_2017-06-05_v8_lung.gct&quot;
raw_count_df &lt;- fread(gene_expr_file, header = TRUE, sep = &quot;\t&quot;, drop = &quot;id&quot;)

# load protein_coding list
protein_coding &lt;- fread(&quot;data/protein-coding_gene.txt&quot;, sep = &quot;\t&quot;)
protein_coding &lt;- protein_coding[, c(&quot;symbol&quot;, &quot;ensembl_gene_id&quot;)]

# keep only protein-coding genes
raw_count_df &lt;- raw_count_df[raw_count_df$Description %in% protein_coding$symbol, ]

id &lt;- raw_count_df$Name
raw_count &lt;- raw_count_df[, -c(1:2)]

# modify GTEx sample names matching names used in PRS data
colnames(raw_count) &lt;- sub(&quot;^(GTEX-[^-.]+).*&quot;, &quot;\\1&quot;, colnames(raw_count))

matching_samples &lt;- intersect(rownames(metadata), colnames(raw_count))
final_count &lt;- raw_count[ , ..matching_samples]

# prefilter: keep only rows that have a count of at least 10
keep_genes &lt;- rowSums(final_count &gt;= 10) &gt; 0
final_count &lt;- final_count[keep_genes, ]
id &lt;- id[keep_genes]
dim(final_count)

prs_trait &lt;- scale(traits) # Standardize PRS to mean = 0, sd = 1
    
# Add the standardized PRS to the metadata for continuous trait
metadata$asthma &lt;- scale(metadata$asthma)
  
# Create the DESeqDataSet for the current trait
dds &lt;- DESeqDataSetFromMatrix(
  countData = as.matrix(final_count),  # Raw counts
  colData = metadata,
  design = as.formula(&quot;~ PC1 + PC2 + PC3 + PC4 + PC5 + sex + asthma&quot;)
  )
rownames(dds) &lt;- id
  
# Run DESeq2 analysis
dds &lt;- DESeq(dds, parallel = TRUE, BPPARAM = MulticoreParam(4))
  
# Get the results for the current trait
res &lt;- results(dds)
  
# Save the results to a file
write.csv(res, &quot;differential_expression_asthma_results.csv&quot;)
  
# print a summary of the results
print(paste(&quot;Results for trait: asthma&quot;))
print(summary(res))
  
# plot the MA-plot for the current trait
png(paste0(&quot;ma_plot_asthma.png&quot;), width = 800, height = 600)
plotMA(res, main = paste(&quot;Continuous: MA Plot for Asthma&quot;))
dev.off()
  
# volcano plot
res_tableOE &lt;- as.data.frame(res)
res_tableOE$gene_name &lt;- raw_count_df$Description[keep_genes]
res_tableOE &lt;- mutate(res_tableOE, threshold_OE = padj &lt; 0.1)
res_tableOE &lt;- res_tableOE %&gt;% arrange(padj) %&gt;% mutate(genelabels = &quot;&quot;)
res_tableOE$genelabels[1:10] &lt;- res_tableOE$gene_name[1:10]
  
volcano_plot &lt;- ggplot(res_tableOE, aes(x = log2FoldChange, y = -log10(padj))) +
    geom_point(aes(colour = threshold_OE)) +
    geom_text_repel(aes(label = genelabels)) +
    ggtitle(paste(&quot;Continuous: Volcano Plot for Asthma&quot;)) +
    xlab(&quot;log2 fold change&quot;) + 
    ylab(&quot;-log10 adjusted p-value&quot;) +
    theme(legend.position = &quot;none&quot;,
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
  
# Save the volcano plot
png(paste0(&quot;volcano_plot_asthma.png&quot;), width = 800, height = 600)
print(volcano_plot)
dev.off()</code></pre>
</div>
<div id="go-enrichment-analysis" class="section level3">
<h3>GO enrichment Analysis</h3>
<pre class="r"><code>deGenes &lt;- res_tableOE[res_tableOE$padj &lt; 0.1 &amp; 
                         abs(res_tableOE$log2FoldChange) &gt;= 0.5, ]
deGenes$gene_id &lt;- gsub(&quot;\\.\\d+$&quot;, &quot;&quot;, rownames(deGenes))

# Separate upregulated and downregulated genes
upregulated_genes &lt;- deGenes[deGenes$log2FoldChange &gt; 0, ]$gene_id
downregulated_genes &lt;- deGenes[deGenes$log2FoldChange &lt; 0, ]$gene_id
  
# Run GO enrichment for upregulated genes
gse_up &lt;- enrichGO(gene = upregulated_genes, ont = &quot;BP&quot;, 
                   OrgDb = &quot;org.Hs.eg.db&quot;, keyType = &quot;ENSEMBL&quot;, readable = T)

# Run GO enrichment for downregulated genes
gse_down &lt;- enrichGO(gene = downregulated_genes, ont = &quot;BP&quot;, 
                     OrgDb = &quot;org.Hs.eg.db&quot;, keyType = &quot;ENSEMBL&quot;, readable = T)
  
# Convert enrichment results to data frames and calculate additional ratios
gse_up &lt;- as.data.frame(gse_up)
gse_down &lt;- as.data.frame(gse_down)

gse_up$GeneRatio_num &lt;- as.numeric(sapply(strsplit(gse_up$GeneRatio, &quot;/&quot;), function(x) x[1])) / 
                        as.numeric(sapply(strsplit(gse_up$GeneRatio, &quot;/&quot;), function(x) x[2]))
gse_up$BgRatio_num &lt;- as.numeric(sapply(strsplit(gse_up$BgRatio, &quot;/&quot;), function(x) x[1])) / 
                        as.numeric(sapply(strsplit(gse_up$BgRatio, &quot;/&quot;), function(x) x[2]))
gse_up &lt;- cbind(gse_up, FoldEnrich = gse_up$GeneRatio_num / gse_up$BgRatio_num)

gse_down$GeneRatio_num &lt;- as.numeric(sapply(strsplit(gse_down$GeneRatio, &quot;/&quot;), function(x) x[1])) / 
                          as.numeric(sapply(strsplit(gse_down$GeneRatio, &quot;/&quot;), function(x) x[2]))
gse_down$BgRatio_num &lt;- as.numeric(sapply(strsplit(gse_down$BgRatio, &quot;/&quot;), function(x) x[1])) / 
                          as.numeric(sapply(strsplit(gse_down$BgRatio, &quot;/&quot;), function(x) x[2]))
gse_down &lt;- cbind(gse_down, FoldEnrich = gse_down$GeneRatio_num / gse_down$BgRatio_num)

if (nrow(gse_up) &gt;= 20) {
    enrich_plot_up &lt;- plotEnrich(gse_up[1:20,], plot_type = &quot;dot&quot;, scale_ratio = 0.5) +
    labs(title = &quot;Continuous: Upregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 10)) 
}else{
    enrich_plot_up &lt;- plotEnrich(gse_up, plot_type = &quot;dot&quot;, scale_ratio = 0.5) +
    labs(title = &quot;Continuous: Upregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 10)) 
}

if (nrow(gse_down) &gt;= 20) {
    enrich_plot_down &lt;- plotEnrich(gse_down[1:20,], plot_type = &quot;dot&quot;, scale_ratio = 0.5) +
    labs(title = &quot;Continuous: Downregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 10)) 
}else{
    enrich_plot_down &lt;- plotEnrich(gse_down, plot_type = &quot;dot&quot;, scale_ratio = 0.5) +
    labs(title = &quot;Continuous: Downregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 10)) 
}

# Arrange the two plots side by side
combined_plot &lt;- grid.arrange(enrich_plot_up, enrich_plot_down, ncol = 2)

# Save the combined plot
ggsave(&quot;enrichment_plot_asthma.png&quot;, plot = combined_plot, width = 12, height = 6)

# Save the GO enrichment results to CSV
write.csv(gse_up, &quot;GO_enrichment_asthma_upregulated.csv&quot;)
write.csv(gse_down, &quot;GO_enrichment_asthma_downregulated.csv&quot;)</code></pre>
</div>
</div>
<div id="quantile-prs" class="section level2">
<h2>Quantile PRS</h2>
<div id="deseq2-differential-expression-1" class="section level3">
<h3>DESeq2 Differential Expression</h3>
<pre class="r"><code>metadata_file &lt;- &quot;analysis/metadata_lung_asthma_quantile.txt&quot;
metadata &lt;- read.csv(metadata_file, header = T, sep = &quot;\t&quot;, stringsAsFactors = T)
metadata$sex &lt;- as.factor(metadata$sex)

# Create the DESeqDataSet for the current trait
dds &lt;- DESeqDataSetFromMatrix(
  countData = as.matrix(final_count),
  colData = metadata,
  design = as.formula(&quot;~ PC1 + PC2 + PC3 + PC4 + PC5 + sex + asthma&quot;)
  )
rownames(dds) &lt;- id
  
# Run DESeq2 analysis
dds &lt;- DESeq(dds, parallel = TRUE, BPPARAM = MulticoreParam(4))
  
# Get the results for the current trait
res &lt;- results(dds)
  
# Save the results to a file
write.csv(res, &quot;differential_expression_asthma_quantile_results.csv&quot;)
  
# print a summary of the results
print(paste(&quot;Results for trait: asthma&quot;))
print(summary(res))
  
# plot the MA-plot for the current trait
png(paste0(&quot;ma_plot_quantile_asthma.png&quot;), width = 800, height = 600)
plotMA(res, main = paste(&quot;Quantile: MA Plot for Asthma&quot;))
dev.off()
  
# volcano plot
res_tableOE &lt;- as.data.frame(res)
res_tableOE$gene_name &lt;- raw_count_df$Description[keep_genes]
res_tableOE &lt;- mutate(res_tableOE, threshold_OE = padj &lt; 0.1)
res_tableOE &lt;- res_tableOE %&gt;% arrange(padj) %&gt;% mutate(genelabels = &quot;&quot;)
res_tableOE$genelabels[1:10] &lt;- res_tableOE$gene_name[1:10]
  
volcano_plot &lt;- ggplot(res_tableOE, aes(x = log2FoldChange, y = -log10(padj))) +
    geom_point(aes(colour = threshold_OE)) +
    geom_text_repel(aes(label = genelabels)) +
    ggtitle(paste(&quot;Quantile: Volcano Plot for Asthma&quot;)) +
    xlab(&quot;log2 fold change&quot;) + 
    ylab(&quot;-log10 adjusted p-value&quot;) +
    theme(legend.position = &quot;none&quot;,
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
  
# Save the volcano plot
png(paste0(&quot;volcano_plot_quantile_asthma.png&quot;), width = 800, height = 600)
print(volcano_plot)
dev.off()</code></pre>
</div>
<div id="go-enrichment-analysis-1" class="section level3">
<h3>GO enrichment Analysis</h3>
<pre class="r"><code>deGenes &lt;- res_tableOE[res_tableOE$padj &lt; 0.1 &amp; 
                         abs(res_tableOE$log2FoldChange) &gt;= 0.5, ]
deGenes$gene_id &lt;- gsub(&quot;\\.\\d+$&quot;, &quot;&quot;, rownames(deGenes))

# Separate upregulated and downregulated genes
upregulated_genes &lt;- deGenes[deGenes$log2FoldChange &gt; 0, ]$gene_id
downregulated_genes &lt;- deGenes[deGenes$log2FoldChange &lt; 0, ]$gene_id
  
# Run GO enrichment for upregulated genes
gse_up &lt;- enrichGO(gene = upregulated_genes, ont = &quot;BP&quot;, 
                   OrgDb = &quot;org.Hs.eg.db&quot;, keyType = &quot;ENSEMBL&quot;, readable = T)

# Run GO enrichment for downregulated genes
gse_down &lt;- enrichGO(gene = downregulated_genes, ont = &quot;BP&quot;, 
                     OrgDb = &quot;org.Hs.eg.db&quot;, keyType = &quot;ENSEMBL&quot;, readable = T)
  
# Convert enrichment results to data frames and calculate additional ratios
gse_up &lt;- as.data.frame(gse_up)
gse_down &lt;- as.data.frame(gse_down)

gse_up$GeneRatio_num &lt;- as.numeric(sapply(strsplit(gse_up$GeneRatio, &quot;/&quot;), function(x) x[1])) / 
                        as.numeric(sapply(strsplit(gse_up$GeneRatio, &quot;/&quot;), function(x) x[2]))
gse_up$BgRatio_num &lt;- as.numeric(sapply(strsplit(gse_up$BgRatio, &quot;/&quot;), function(x) x[1])) / 
                        as.numeric(sapply(strsplit(gse_up$BgRatio, &quot;/&quot;), function(x) x[2]))
gse_up &lt;- cbind(gse_up, FoldEnrich = gse_up$GeneRatio_num / gse_up$BgRatio_num)

gse_down$GeneRatio_num &lt;- as.numeric(sapply(strsplit(gse_down$GeneRatio, &quot;/&quot;), function(x) x[1])) / 
                          as.numeric(sapply(strsplit(gse_down$GeneRatio, &quot;/&quot;), function(x) x[2]))
gse_down$BgRatio_num &lt;- as.numeric(sapply(strsplit(gse_down$BgRatio, &quot;/&quot;), function(x) x[1])) / 
                          as.numeric(sapply(strsplit(gse_down$BgRatio, &quot;/&quot;), function(x) x[2]))
gse_down &lt;- cbind(gse_down, FoldEnrich = gse_down$GeneRatio_num / gse_down$BgRatio_num)

if (nrow(gse_up) &gt;= 20) {
    enrich_plot_up &lt;- plotEnrich(gse_up[1:20,], plot_type = &quot;dot&quot;, scale_ratio = 0.4) +
    labs(title = &quot;Quantile: Upregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 6)) 
}else{
    enrich_plot_up &lt;- plotEnrich(gse_up, plot_type = &quot;dot&quot;, scale_ratio = 0.4) +
    labs(title = &quot;Quantile: Upregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 6)) 
}

if (nrow(gse_down) &gt;= 20) {
    enrich_plot_down &lt;- plotEnrich(gse_down[1:20,], plot_type = &quot;dot&quot;, scale_ratio = 0.4) +
    labs(title = &quot;Quantile: Downregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 6)) 
}else{
    enrich_plot_down &lt;- plotEnrich(gse_down, plot_type = &quot;dot&quot;, scale_ratio = 0.4) +
    labs(title = &quot;Quantile: Downregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 6)) 
}

# Arrange the two plots side by side
combined_plot &lt;- grid.arrange(enrich_plot_up, enrich_plot_down, ncol = 2)

# Save the combined plot
ggsave(&quot;enrichment_plot_quantile_asthma.png&quot;, plot = combined_plot, width = 12, height = 6)

# Save the GO enrichment results to CSV
write.csv(gse_up, &quot;GO_enrichment_quantile_asthma_upregulated.csv&quot;)
write.csv(gse_down, &quot;GO_enrichment_quantile_asthma_downregulated.csv&quot;)</code></pre>
</div>
</div>
</div>
<div id="model-2-expression-prs-sex-expression-pcs-2-genotype-pcs"
class="section level1">
<h1>Model 2: expression ~ PRS + sex + expression PCs + 2 genotype
PCs</h1>
<div id="obtain-genotype-pcs" class="section level2">
<h2>Obtain genotype PCs</h2>
<pre class="r"><code># Load the gene expression data
gene_expr_file &lt;- &quot;data/gene_reads_2017-06-05_v8_lung.gct&quot;
raw_count_df &lt;- fread(gene_expr_file, header = TRUE, sep = &quot;\t&quot;, drop = &quot;id&quot;)

# load protein_coding list
protein_coding &lt;- fread(&quot;data/protein-coding_gene.txt&quot;, 
                        sep = &quot;\t&quot;)
protein_coding &lt;- protein_coding[, c(&quot;symbol&quot;, &quot;ensembl_gene_id&quot;)]

# keep only protein-coding genes
raw_count_df &lt;- raw_count_df[raw_count_df$Description %in% protein_coding$symbol, ]

id &lt;- raw_count_df$Name
raw_count &lt;- raw_count_df[, -c(1:2)]

# modify GTEx sample names matching names used in PRS data
colnames(raw_count) &lt;- sub(&quot;^(GTEX-[^-.]+).*&quot;, &quot;\\1&quot;, colnames(raw_count))

matching_samples &lt;- intersect(rownames(metadata), colnames(raw_count))
final_count &lt;- raw_count[ , ..matching_samples]

# prefilter: keep only rows that have a count of at least 10
keep_genes &lt;- rowSums(final_count &gt;= 10) &gt; 0
final_count &lt;- final_count[keep_genes, ]
id &lt;- id[keep_genes]
dim(final_count)</code></pre>
<pre><code>[1] 17297   515</code></pre>
<pre class="r"><code># obtain genotype PCs
geno_pc &lt;- read.table(&quot;data/pca.eigenvec&quot;)
names(geno_pc) = c(&quot;FID&quot;,&quot;IID&quot;,paste0(&quot;geno_PC&quot;, c(1:(ncol(geno_pc)-2))))
geno_pc &lt;- geno_pc[geno_pc$FID %in% matching_samples, ]

eigenval &lt;- scan(&quot;data/pca.eigenval&quot;)
pve &lt;- data.frame(PC = 1:20, pve = eigenval/sum(eigenval)*100)

ggplot(geno_pc) + geom_point(aes(x = geno_PC1, y = geno_PC2)) + 
  labs(x = paste0(&quot;PC1 (&quot;, signif(pve$pve[1], 3), &quot;%)&quot;), 
       y = paste0(&quot;PC2 (&quot;, signif(pve$pve[2], 3), &quot;%)&quot;)) + theme_classic()</code></pre>
<p><img src="figure/asthma_lung.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(geno_pc) + geom_point(aes(x = geno_PC1, y = geno_PC3)) + 
    labs(x = paste0(&quot;PC1 (&quot;, signif(pve$pve[1], 3), &quot;%)&quot;), 
         y = paste0(&quot;PC3 (&quot;, signif(pve$pve[3], 3), &quot;%)&quot;)) + theme_classic()</code></pre>
<p><img src="figure/asthma_lung.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pve, aes(PC, pve)) + geom_point() + geom_line() +
  labs(x = &quot;Genotype PC&quot;, y = &quot;Percentage variance explained&quot;) + theme_classic()</code></pre>
<p><img src="figure/asthma_lung.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="continuous-prs-1" class="section level2">
<h2>Continuous PRS</h2>
<div id="deseq2-differential-expression-2" class="section level3">
<h3>DESeq2 Differential Expression</h3>
<pre class="r"><code>metadata_file &lt;- &quot;analysis/metadata_lung_asthma.txt&quot;
metadata &lt;- read.csv(metadata_file, header = T, sep = &quot;\t&quot;)
metadata$sex &lt;- as.factor(metadata$sex)

metadata &lt;- cbind(metadata, geno_pc[,3:ncol(geno_pc)])

# Standardize PRS for the current trait
#prs_trait &lt;- scale(metadata$asthma)

# Add the standardized PRS to the metadata for continuous trait
metadata$asthma &lt;- scale(metadata$asthma)

# Create the DESeqDataSet for the current trait
dds &lt;- DESeqDataSetFromMatrix(
  countData = as.matrix(final_count),  # Raw counts
  colData = metadata[, 1:9],
  design = as.formula(&quot;~ PC1 + PC2 + PC3 + PC4 + PC5 + sex + geno_PC1 + geno_PC2 + asthma&quot;)
)
rownames(dds) &lt;- id

# Run DESeq2 analysis
dds &lt;- DESeq(dds, parallel = TRUE, BPPARAM = MulticoreParam(4))

# Get the results for the current trait
res &lt;- results(dds)
  
# Save the results to a file
write.csv(res, &quot;differential_expression_asthma_results_M2.csv&quot;)
  
# print a summary of the results
print(&quot;Results for trait: Asthma&quot;)
print(summary(res))
  
# plot the MA-plot for the current trait
png(paste0(&quot;ma_plot_asthma_M2.png&quot;), width = 800, height = 600)
plotMA(res, main = &quot;Continuous (M2): MA Plot for Asthma&quot;)
dev.off()
  
# volcano plot
res_tableOE &lt;- as.data.frame(res)
res_tableOE$gene_name &lt;- raw_count_df$Description[keep_genes]
res_tableOE &lt;- mutate(res_tableOE, threshold_OE = padj &lt; 0.1)
res_tableOE &lt;- res_tableOE %&gt;% arrange(padj) %&gt;% mutate(genelabels = &quot;&quot;)
res_tableOE$genelabels[1:10] &lt;- res_tableOE$gene_name[1:10]
  
volcano_plot &lt;- ggplot(res_tableOE, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(aes(colour = threshold_OE)) +
  geom_text_repel(aes(label = genelabels)) +
  ggtitle(&quot;Continuous (M2): Volcano Plot for Asthma&quot;) +
    xlab(&quot;log2 fold change&quot;) + 
    ylab(&quot;-log10 adjusted p-value&quot;) +
    theme(legend.position = &quot;none&quot;,
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
  
# Save the volcano plot
png(paste0(&quot;volcano_plot_asthma_M2.png&quot;), width = 800, height = 600)
print(volcano_plot)
dev.off()</code></pre>
</div>
<div id="go-enrichment-analysis-2" class="section level3">
<h3>GO enrichment Analysis</h3>
<pre class="r"><code>deGenes &lt;- res_tableOE[res_tableOE$padj &lt; 0.1 &amp; 
                         abs(res_tableOE$log2FoldChange) &gt;= 0.5, ]
deGenes$gene_id &lt;- gsub(&quot;\\.\\d+$&quot;, &quot;&quot;, rownames(deGenes))

# Separate upregulated and downregulated genes
upregulated_genes &lt;- deGenes[deGenes$log2FoldChange &gt; 0, ]$gene_id
downregulated_genes &lt;- deGenes[deGenes$log2FoldChange &lt; 0, ]$gene_id
  
# Run GO enrichment for upregulated genes
gse_up &lt;- enrichGO(gene = upregulated_genes, ont = &quot;BP&quot;, 
                   OrgDb = &quot;org.Hs.eg.db&quot;, keyType = &quot;ENSEMBL&quot;, readable = T)

# Run GO enrichment for downregulated genes
gse_down &lt;- enrichGO(gene = downregulated_genes, ont = &quot;BP&quot;, 
                     OrgDb = &quot;org.Hs.eg.db&quot;, keyType = &quot;ENSEMBL&quot;, readable = T)

# Convert enrichment results to data frames and calculate additional ratios
gse_up &lt;- as.data.frame(gse_up)
gse_down &lt;- as.data.frame(gse_down)

gse_up$GeneRatio_num &lt;- as.numeric(sapply(strsplit(gse_up$GeneRatio, &quot;/&quot;), function(x) x[1])) / 
                        as.numeric(sapply(strsplit(gse_up$GeneRatio, &quot;/&quot;), function(x) x[2]))
gse_up$BgRatio_num &lt;- as.numeric(sapply(strsplit(gse_up$BgRatio, &quot;/&quot;), function(x) x[1])) / 
                        as.numeric(sapply(strsplit(gse_up$BgRatio, &quot;/&quot;), function(x) x[2]))
gse_up &lt;- cbind(gse_up, FoldEnrich = gse_up$GeneRatio_num / gse_up$BgRatio_num)

gse_down$GeneRatio_num &lt;- as.numeric(sapply(strsplit(gse_down$GeneRatio, &quot;/&quot;), function(x) x[1])) / 
                          as.numeric(sapply(strsplit(gse_down$GeneRatio, &quot;/&quot;), function(x) x[2]))
gse_down$BgRatio_num &lt;- as.numeric(sapply(strsplit(gse_down$BgRatio, &quot;/&quot;), function(x) x[1])) / 
                          as.numeric(sapply(strsplit(gse_down$BgRatio, &quot;/&quot;), function(x) x[2]))
gse_down &lt;- cbind(gse_down, FoldEnrich = gse_down$GeneRatio_num / gse_down$BgRatio_num)

if (nrow(gse_up) &gt;= 20) {
    enrich_plot_up &lt;- plotEnrich(gse_up[1:20,], plot_type = &quot;dot&quot;, scale_ratio = 0.4) +
    labs(title = &quot;Continuous (M2): Upregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 6)) 
}else{
    enrich_plot_up &lt;- plotEnrich(gse_up, plot_type = &quot;dot&quot;, scale_ratio = 0.4) +
    labs(title = &quot;Continuous (M2): Upregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 6)) 
}

if (nrow(gse_down) &gt;= 20) {
    enrich_plot_down &lt;- plotEnrich(gse_down[1:20,], plot_type = &quot;dot&quot;, scale_ratio = 0.4) +
    labs(title = &quot;Continuous (M2): Downregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 6)) 
}else{
    enrich_plot_down &lt;- plotEnrich(gse_down, plot_type = &quot;dot&quot;, scale_ratio = 0.4) +
    labs(title = &quot;Continuous (M2): Downregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 6)) 
}

# Arrange the two plots side by side
combined_plot &lt;- grid.arrange(enrich_plot_up, enrich_plot_down, ncol = 2)

# Save the combined plot
ggsave(&quot;enrichment_plot_asthma_M2.png&quot;, plot = combined_plot, width = 12, height = 6)

# Save the GO enrichment results to CSV
write.csv(gse_up, &quot;GO_enrichment_asthma_upregulated_M2.csv&quot;)
write.csv(gse_down, &quot;GO_enrichment_asthma_downregulated_M2.csv&quot;)</code></pre>
</div>
</div>
<div id="quantile-prs-1" class="section level2">
<h2>Quantile PRS</h2>
<div id="deseq2-differential-expression-3" class="section level3">
<h3>DESeq2 Differential Expression</h3>
<pre class="r"><code>metadata_file &lt;- &quot;analysis/metadata_lung_asthma_quantile.txt&quot;
metadata &lt;- read.csv(metadata_file, header = T, sep = &quot;\t&quot;)
metadata$sex &lt;- as.factor(metadata$sex)

metadata &lt;- cbind(metadata, geno_pc[,3:ncol(geno_pc)])

# Create the DESeqDataSet for the current trait
dds &lt;- DESeqDataSetFromMatrix(
  countData = as.matrix(final_count),  # Raw counts
  colData = metadata[, 1:9],
  design = as.formula(&quot;~ PC1 + PC2 + PC3 + PC4 + PC5 + sex + geno_PC1 + geno_PC2 + asthma&quot;)
)
rownames(dds) &lt;- id

# Run DESeq2 analysis
dds &lt;- DESeq(dds, parallel = TRUE, BPPARAM = MulticoreParam(4))

# Get the results for the current trait
res &lt;- results(dds)
  
# Save the results to a file
write.csv(res, &quot;differential_expression_asthma_quantile_results_M2.csv&quot;)
  
# print a summary of the results
print(&quot;Results for trait: Asthma&quot;)
print(summary(res))
  
# plot the MA-plot for the current trait
png(paste0(&quot;ma_plot_quantile_asthma_M2.png&quot;), width = 800, height = 600)
plotMA(res, main = &quot;Quantile (M2): MA Plot for Asthma&quot;)
dev.off()
  
# volcano plot
res_tableOE &lt;- as.data.frame(res)
res_tableOE$gene_name &lt;- raw_count_df$Description[keep_genes]
res_tableOE &lt;- mutate(res_tableOE, threshold_OE = padj &lt; 0.1)
res_tableOE &lt;- res_tableOE %&gt;% arrange(padj) %&gt;% mutate(genelabels = &quot;&quot;)
res_tableOE$genelabels[1:10] &lt;- res_tableOE$gene_name[1:10]
  
volcano_plot &lt;- ggplot(res_tableOE, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(aes(colour = threshold_OE)) +
  geom_text_repel(aes(label = genelabels)) +
  ggtitle(&quot;Quantile (M2): Volcano Plot for Asthma&quot;) +
    xlab(&quot;log2 fold change&quot;) + 
    ylab(&quot;-log10 adjusted p-value&quot;) +
    theme(legend.position = &quot;none&quot;,
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
  
# Save the volcano plot
png(paste0(&quot;volcano_plot_quantile_asthma_M2.png&quot;), width = 800, height = 600)
print(volcano_plot)
dev.off()</code></pre>
</div>
<div id="go-enrichment-analysis-3" class="section level3">
<h3>GO enrichment Analysis</h3>
<pre class="r"><code>deGenes &lt;- res_tableOE[res_tableOE$padj &lt; 0.1 &amp; 
                         abs(res_tableOE$log2FoldChange) &gt;= 0.5, ]
deGenes$gene_id &lt;- gsub(&quot;\\.\\d+$&quot;, &quot;&quot;, rownames(deGenes))

# Separate upregulated and downregulated genes
upregulated_genes &lt;- deGenes[deGenes$log2FoldChange &gt; 0, ]$gene_id
downregulated_genes &lt;- deGenes[deGenes$log2FoldChange &lt; 0, ]$gene_id
  
# Run GO enrichment for upregulated genes
gse_up &lt;- enrichGO(gene = upregulated_genes, ont = &quot;BP&quot;, 
                   OrgDb = &quot;org.Hs.eg.db&quot;, keyType = &quot;ENSEMBL&quot;, readable = T)

# Run GO enrichment for downregulated genes
gse_down &lt;- enrichGO(gene = downregulated_genes, ont = &quot;BP&quot;, 
                     OrgDb = &quot;org.Hs.eg.db&quot;, keyType = &quot;ENSEMBL&quot;, readable = T)
  
# Convert enrichment results to data frames and calculate additional ratios
gse_up &lt;- as.data.frame(gse_up)
gse_down &lt;- as.data.frame(gse_down)

gse_up$GeneRatio_num &lt;- as.numeric(sapply(strsplit(gse_up$GeneRatio, &quot;/&quot;), function(x) x[1])) / 
                        as.numeric(sapply(strsplit(gse_up$GeneRatio, &quot;/&quot;), function(x) x[2]))
gse_up$BgRatio_num &lt;- as.numeric(sapply(strsplit(gse_up$BgRatio, &quot;/&quot;), function(x) x[1])) / 
                        as.numeric(sapply(strsplit(gse_up$BgRatio, &quot;/&quot;), function(x) x[2]))
gse_up &lt;- cbind(gse_up, FoldEnrich = gse_up$GeneRatio_num / gse_up$BgRatio_num)

gse_down$GeneRatio_num &lt;- as.numeric(sapply(strsplit(gse_down$GeneRatio, &quot;/&quot;), function(x) x[1])) / 
                          as.numeric(sapply(strsplit(gse_down$GeneRatio, &quot;/&quot;), function(x) x[2]))
gse_down$BgRatio_num &lt;- as.numeric(sapply(strsplit(gse_down$BgRatio, &quot;/&quot;), function(x) x[1])) / 
                          as.numeric(sapply(strsplit(gse_down$BgRatio, &quot;/&quot;), function(x) x[2]))
gse_down &lt;- cbind(gse_down, FoldEnrich = gse_down$GeneRatio_num / gse_down$BgRatio_num)

if (nrow(gse_up) &gt;= 20) {
    enrich_plot_up &lt;- plotEnrich(gse_up[1:20,], plot_type = &quot;dot&quot;, scale_ratio = 0.4) +
    labs(title = &quot;Quantile (M2): Upregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 6)) 
}else{
    enrich_plot_up &lt;- plotEnrich(gse_up, plot_type = &quot;dot&quot;, scale_ratio = 0.4) +
    labs(title = &quot;Quantile (M2): Upregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 6)) 
}

if (nrow(gse_down) &gt;= 20) {
    enrich_plot_down &lt;- plotEnrich(gse_down[1:20,], plot_type = &quot;dot&quot;, scale_ratio = 0.4) +
    labs(title = &quot;Quantile (M2): Downregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 6)) 
}else{
    enrich_plot_down &lt;- plotEnrich(gse_down, plot_type = &quot;dot&quot;, scale_ratio = 0.4) +
    labs(title = &quot;Quantile (M2): Downregulated Enrichment Pathways for Asthma&quot;) + 
    theme(plot.title = element_text(size = 6)) 
}

# Arrange the two plots side by side
combined_plot &lt;- grid.arrange(enrich_plot_up, enrich_plot_down, ncol = 2)

# Save the combined plot
ggsave(&quot;enrichment_plot_quantile_asthma_M2.png&quot;, plot = combined_plot, width = 12, height = 6)

# Save the GO enrichment results to CSV
write.csv(gse_up, &quot;GO_enrichment_quantile_asthma_upregulated_M2.csv&quot;)
write.csv(gse_down, &quot;GO_enrichment_quantil_asthma_downregulated_M2.csv&quot;)</code></pre>
</div>
</div>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<div id="summary-table" class="section level2">
<h2>Summary table</h2>
<table>
<caption>Summary of Differential Expression and Enriched Pathways for
Asthma PRS Models</caption>
<colgroup>
<col width="11%" />
<col width="16%" />
<col width="14%" />
<col width="16%" />
<col width="19%" />
<col width="21%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Model</th>
<th align="right">Significant DE genes</th>
<th align="right">Up-regulated genes</th>
<th align="right">Down-regulated genes</th>
<th align="right">Up-regulated GO pathways</th>
<th align="right">Down-regulated GO pathways</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">continuous</td>
<td align="right">138</td>
<td align="right">23</td>
<td align="right">115</td>
<td align="right">0</td>
<td align="right">225</td>
</tr>
<tr class="even">
<td align="left">continuous_M2</td>
<td align="right">136</td>
<td align="right">22</td>
<td align="right">114</td>
<td align="right">0</td>
<td align="right">225</td>
</tr>
<tr class="odd">
<td align="left">quantile</td>
<td align="right">90</td>
<td align="right">33</td>
<td align="right">57</td>
<td align="right">14</td>
<td align="right">163</td>
</tr>
<tr class="even">
<td align="left">quantile_M2</td>
<td align="right">84</td>
<td align="right">31</td>
<td align="right">53</td>
<td align="right">15</td>
<td align="right">143</td>
</tr>
</tbody>
</table>
</div>
<div id="model-1-adjust-without-genotype-pcs" class="section level2">
<h2>Model 1: Adjust without genotype PCs</h2>
<div id="continuous-prs-2" class="section level3">
<h3>Continuous PRS</h3>
<p><img src="figure/asthma_lung.Rmd/ma_plot_asthma.png" /> <img
src="figure/asthma_lung.Rmd/volcano_plot_asthma.png" /> <img
src="figure/asthma_lung.Rmd/enrichment_plot_asthma.png" /></p>
</div>
<div id="quantile-prs-2" class="section level3">
<h3>Quantile PRS</h3>
<p><img src="figure/asthma_lung.Rmd/ma_plot_quantile_asthma.png" /> <img
src="figure/asthma_lung.Rmd/volcano_plot_quantile_asthma.png" /> <img
src="figure/asthma_lung.Rmd/enrichment_plot_quantile_asthma.png" /></p>
</div>
</div>
<div id="model-2-adjust-with-genotype-pcs" class="section level2">
<h2>Model 2: Adjust with genotype PCs</h2>
<div id="continuous-prs-3" class="section level3">
<h3>Continuous PRS</h3>
<p><img src="figure/asthma_lung.Rmd/ma_plot_asthma_M2.png" /> <img
src="figure/asthma_lung.Rmd/volcano_plot_asthma_M2.png" /> <img
src="figure/asthma_lung.Rmd/enrichment_plot_asthma_M2.png" /></p>
</div>
<div id="quantile-prs-3" class="section level3">
<h3>Quantile PRS</h3>
<p><img src="figure/asthma_lung.Rmd/ma_plot_quantile_asthma_M2.png" />
<img src="figure/asthma_lung.Rmd/volcano_plot_quantile_asthma_M2.png" />
<img
src="figure/asthma_lung.Rmd/enrichment_plot_quantile_asthma_M2.png" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gridExtra_2.3               clusterProfiler_4.6.2      
 [3] enrichplot_1.18.4           org.Hs.eg.db_3.16.0        
 [5] AnnotationDbi_1.60.2        genekitr_1.2.8             
 [7] ggrepel_0.9.6               BiocParallel_1.32.6        
 [9] DESeq2_1.38.3               SummarizedExperiment_1.28.0
[11] Biobase_2.58.0              MatrixGenerics_1.10.0      
[13] matrixStats_1.2.0           GenomicRanges_1.50.2       
[15] GenomeInfoDb_1.34.9         IRanges_2.32.0             
[17] S4Vectors_0.36.2            BiocGenerics_0.44.0        
[19] corrplot_0.95               lubridate_1.9.4            
[21] forcats_1.0.0               stringr_1.5.1              
[23] dplyr_1.1.4                 purrr_1.0.2                
[25] readr_2.1.5                 tidyr_1.3.1                
[27] tibble_3.2.1                ggplot2_3.5.1              
[29] tidyverse_2.0.0             data.table_1.16.4          
[31] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] shadowtext_0.1.4       fastmatch_1.1-6        plyr_1.8.9            
  [4] igraph_1.5.1           lazyeval_0.2.2         splines_4.2.2         
  [7] usethis_3.1.0          urltools_1.7.3         digest_0.6.37         
 [10] yulab.utils_0.2.0      htmltools_0.5.8.1      GOSemSim_2.24.0       
 [13] viridis_0.6.5          GO.db_3.16.0           magrittr_2.0.3        
 [16] memoise_2.0.1          remotes_2.5.0          openxlsx_4.2.5.2      
 [19] tzdb_0.4.0             Biostrings_2.66.0      annotate_1.76.0       
 [22] graphlayouts_1.0.1     vroom_1.6.5            timechange_0.3.0      
 [25] prettyunits_1.2.0      colorspace_2.1-1       blob_1.2.4            
 [28] xfun_0.50              callr_3.7.6            crayon_1.5.3          
 [31] RCurl_1.98-1.16        jsonlite_1.8.9         scatterpie_0.2.4      
 [34] ape_5.7-1              glue_1.8.0             polyclip_1.10-7       
 [37] gtable_0.3.6           zlibbioc_1.44.0        XVector_0.38.0        
 [40] DelayedArray_0.24.0    pkgbuild_1.4.6         scales_1.3.0          
 [43] DOSE_3.24.2            DBI_1.2.3              miniUI_0.1.1.1        
 [46] Rcpp_1.0.14            progress_1.2.3         viridisLite_0.4.2     
 [49] xtable_1.8-4           gridGraphics_0.5-1     tidytree_0.4.6        
 [52] europepmc_0.4.3        bit_4.5.0.1            profvis_0.4.0         
 [55] htmlwidgets_1.6.4      httr_1.4.7             fgsea_1.24.0          
 [58] RColorBrewer_1.1-3     ellipsis_0.3.2         urlchecker_1.0.1      
 [61] pkgconfig_2.0.3        XML_3.99-0.18          farver_2.1.2          
 [64] sass_0.4.9             locfit_1.5-9.8         labeling_0.4.3        
 [67] ggplotify_0.1.2        tidyselect_1.2.1       rlang_1.1.5           
 [70] reshape2_1.4.4         later_1.4.1            munsell_0.5.1         
 [73] tools_4.2.2            cachem_1.1.0           downloader_0.4        
 [76] cli_3.6.3              generics_0.1.3         RSQLite_2.3.9         
 [79] gson_0.1.0             devtools_2.4.5         evaluate_1.0.3        
 [82] fastmap_1.2.0          yaml_2.3.10            ggtree_3.6.2          
 [85] processx_3.8.5         knitr_1.49             bit64_4.6.0-1         
 [88] fs_1.6.5               tidygraph_1.3.0        zip_2.3.2             
 [91] KEGGREST_1.38.0        ggraph_2.1.0           nlme_3.1-160          
 [94] mime_0.12              whisker_0.4.1          aplot_0.2.4           
 [97] ggvenn_0.1.10          xml2_1.3.6             compiler_4.2.2        
[100] rstudioapi_0.17.1      png_0.1-8              treeio_1.22.0         
[103] tweenr_2.0.3           geneplotter_1.76.0     bslib_0.9.0           
[106] stringi_1.8.4          ps_1.8.1               lattice_0.22-6        
[109] Matrix_1.6-4           vctrs_0.6.5            pillar_1.10.1         
[112] lifecycle_1.0.4        triebeard_0.4.1        jquerylib_0.1.4       
[115] cowplot_1.1.3          bitops_1.0-9           httpuv_1.6.15         
[118] patchwork_1.3.0        qvalue_2.30.0          R6_2.5.1              
[121] promises_1.3.2         sessioninfo_1.2.2      codetools_0.2-20      
[124] pkgload_1.4.0          MASS_7.3-58.1          rprojroot_2.0.4       
[127] withr_3.0.2            GenomeInfoDbData_1.2.9 parallel_4.2.2        
[130] hms_1.1.3              grid_4.2.2             ggfun_0.1.8           
[133] HDO.db_0.99.1          rmarkdown_2.29         git2r_0.33.0          
[136] getPass_0.2-4          ggforce_0.4.1          shiny_1.10.0          
[139] geneset_0.2.7         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
