<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Differential expression</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/yixuan-chen-elisa/myproject">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Differential expression</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-20
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>prs/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250417code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250417)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250417code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250417)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyixuanchenelisaprstreec2d7279074ba2e70d8ad9331d0794a6a802a2d0atargetblankc2d7279a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/yixuan-chen-elisa/prs/tree/c2d7279074ba2e70d8ad9331d0794a6a802a2d0a" target="_blank">c2d7279</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomyixuanchenelisaprstreec2d7279074ba2e70d8ad9331d0794a6a802a2d0atargetblankc2d7279a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/yixuan-chen-elisa/prs/tree/c2d7279074ba2e70d8ad9331d0794a6a802a2d0a" target="_blank">c2d7279</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  analysis/continuous_artery/
    Untracked:  analysis/continuous_wb/
    Untracked:  analysis/continuous_wb_m1/
    Untracked:  analysis/continuous_wb_m2/
    Untracked:  analysis/metadata.txt
    Untracked:  analysis/metadata_artery.txt
    Untracked:  analysis/metadata_artery_quantile.txt
    Untracked:  analysis/metadata_quantile.txt
    Untracked:  analysis/normalized_counts.rda
    Untracked:  analysis/quantile_artery/
    Untracked:  analysis/quantile_wb/
    Untracked:  analysis/quantile_wb_m1/
    Untracked:  analysis/quantile_wb_m2/
    Untracked:  analysis/vst norm counts.rda
    Untracked:  data/Artery_Aorta.v8.covariates.txt
    Untracked:  data/GTEx_v8.bk
    Untracked:  data/GTEx_v8.rds
    Untracked:  data/T2D_hmPOS_GRCh38.txt
    Untracked:  data/Whole_Blood.v8.covariates.txt
    Untracked:  data/blood_cell/
    Untracked:  data/gene_reads_2017-06-05_v8_artery_aorta.gct
    Untracked:  data/gene_reads_2017-06-05_v8_whole_blood.gct
    Untracked:  data/gene_tpm_2017-06-05_v8_whole_blood.gct.gz
    Untracked:  data/immune/
    Untracked:  data/pca.eigenval
    Untracked:  data/pca.eigenvec
    Untracked:  data/protein-coding_gene.txt
    Untracked:  metadata_m2.csv

Unstaged changes:
    Deleted:    analysis/QC.Rmd
    Deleted:    analysis/normalized_counts.txt
    Modified:   analysis/prs_blood_cell.txt
    Modified:   prs.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/differential_expression.Rmd</code>) and HTML
(<code>docs/differential_expression.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/c2d7279074ba2e70d8ad9331d0794a6a802a2d0a/docs/differential_expression.html" target="_blank">c2d7279</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/e3d43a74946995b65931e39892b4c6e2564bc088/analysis/differential_expression.Rmd" target="_blank">e3d43a7</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-20
</td>
<td>
workflowr::wflow_publish("analysis/differential_expression.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/cf16954a68991e3dcd18c035b7279cea3c286362/docs/differential_expression.html" target="_blank">cf16954</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/1200397e06307b1beb5aa94e8d78ad9b18cb02ca/analysis/differential_expression.Rmd" target="_blank">1200397</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-20
</td>
<td>
workflowr::wflow_publish("analysis/differential_expression.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/51ffd48702cd994a05a864da79f2fcc8ad9989b9/docs/differential_expression.html" target="_blank">51ffd48</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/3a1e4dd5a25f479cda2266471c3e7c51553525eb/analysis/differential_expression.Rmd" target="_blank">3a1e4dd</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-20
</td>
<td>
workflowr::wflow_publish("analysis/differential_expression.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/e4dd48b9ed59a365f48a3c14dd47fbd409152bb1/docs/differential_expression.html" target="_blank">e4dd48b</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/4bdd6d20e954341b6f5f44beefcf788d935abc16/analysis/differential_expression.Rmd" target="_blank">4bdd6d2</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-16
</td>
<td>
workflowr::wflow_publish("analysis/differential_expression.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/d78bc7e169a1574a049a1694694cc0fd7b737975/docs/differential_expression.html" target="_blank">d78bc7e</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/f9c370a928f33d4acd99b613a1d7af103065e720/analysis/differential_expression.Rmd" target="_blank">f9c370a</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-16
</td>
<td>
workflowr::wflow_publish("analysis/differential_expression.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/1ef6ae9c14bef5ecf10bedaf99ef37f345826afb/docs/differential_expression.html" target="_blank">1ef6ae9</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/0591bd80827f6b0f1edb51f1a7a397b1afb68d66/analysis/differential_expression.Rmd" target="_blank">0591bd8</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-14
</td>
<td>
workflowr::wflow_publish(c("analysis/differential_expression.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/3bb86a4cc06bad8b46207046d1c5e169dd3567a2/docs/differential_expression.html" target="_blank">3bb86a4</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/a2649641c72080088cacd37821f9a5d7c7231166/analysis/differential_expression.Rmd" target="_blank">a264964</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-13
</td>
<td>
workflowr::wflow_publish("analysis/differential_expression.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/2b73cc36eaf2923feaccb6ca107242d45d039ad4/docs/differential_expression.html" target="_blank">2b73cc3</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/c5d5446df49e64e682e21fb9717e7bb438e267a6/analysis/differential_expression.Rmd" target="_blank">c5d5446</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-13
</td>
<td>
workflowr::wflow_publish(c("analysis/PRS.Rmd", "analysis/index.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/daf51702f2cac8c4b52ef591cbd56d264c0595dc/docs/differential_expression.html" target="_blank">daf5170</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/060c7acceea40b4cc08de64679980145f5ea4f37/analysis/differential_expression.Rmd" target="_blank">060c7ac</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-07
</td>
<td>
wflow_publish("analysis/differential_expression.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/d129b3e136dc67029de8b15f88711452342a73dd/docs/differential_expression.html" target="_blank">d129b3e</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/7a5b0e4b564f2a60d2b440de79814384160f09fd/analysis/differential_expression.Rmd" target="_blank">7a5b0e4</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-07
</td>
<td>
wflow_publish(c("analysis/index.Rmd", "analysis/metadata.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/5e7a42ebb4ef83cfb89955c4548bf8d215ba1a08/docs/differential_expression.html" target="_blank">5e7a42e</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-04-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/6a1157a2bb6789de0e180c9a7f198be9e113a978/analysis/differential_expression.Rmd" target="_blank">6a1157a</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-04-30
</td>
<td>
workflowr::wflow_publish("analysis/differential_expression.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/959e0318c56ad9f5edaa8ddde67a201bcc010f7e/docs/differential_expression.html" target="_blank">959e031</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-04-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/9017040dcd0d02e9db257f816e709459bea482ad/analysis/differential_expression.Rmd" target="_blank">9017040</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-04-30
</td>
<td>
workflowr::wflow_publish("analysis/differential_expression.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/ae6b2a5a84a3f81a852cfc7d12b2415ef4477224/docs/differential_expression.html" target="_blank">ae6b2a5</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-04-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/6488595341c3d6d4422847d065b8d8e5d65b6a4d/analysis/differential_expression.Rmd" target="_blank">6488595</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-04-30
</td>
<td>
workflowr::wflow_publish("analysis/differential_expression.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yixuan-chen-elisa/prs/c4ece77e6e04279cb27efb04a88e27335aa144a4/docs/differential_expression.html" target="_blank">c4ece77</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-04-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/5b80e272b1564a6835a64729103a8791c1dba21f/analysis/differential_expression.Rmd" target="_blank">5b80e27</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-04-30
</td>
<td>
workflowr::wflow_publish(c("analysis/differential_expression.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="correlation-between-prs-expression-pcs-whole-blood"
class="section level1">
<h1>Correlation between PRS &amp; expression PCs (whole blood)</h1>
<pre class="r"><code># load prs &amp; pcs
metadata_file &lt;- &quot;analysis/metadata.txt&quot;
metadata &lt;- read.csv(metadata_file, header = T, sep = &quot;\t&quot;, stringsAsFactors = T)
metadata$sex &lt;- as.factor(metadata$sex)

traits &lt;- metadata[, 7:43]
pc &lt;- metadata[, c(1:5)]

# Calculate the correlation between each trait and each PC
correlation_matrix &lt;- cor(traits, pc)
range(correlation_matrix)</code></pre>
<pre><code>[1] -0.2850667  0.2787222</code></pre>
<pre class="r"><code>correlation_matrix &lt;- t(correlation_matrix)

# Create the heatmap using corrplot
corrplot(correlation_matrix, method = &quot;color&quot;, 
         col = colorRampPalette(c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;))(200), # color palette
         addCoef.col = &quot;black&quot;, # Add correlation coefficients to the plot
         number.cex = 0.4, # Adjust the font size of the numbers
         tl.col = &quot;black&quot;, # text label color
         tl.srt = 90, # rotate text labels
         tl.cex = 0.5, 
         title = &quot;Correlation between Traits and expression PCs&quot;, 
         mar = c(0, 0, 1, 0)
)</code></pre>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/2b73cc36eaf2923feaccb6ca107242d45d039ad4/docs/figure/differential_expression.Rmd/unnamed-chunk-1-1.png" target="_blank">2b73cc3</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/c4ece77e6e04279cb27efb04a88e27335aa144a4/docs/figure/differential_expression.Rmd/unnamed-chunk-1-1.png" target="_blank">c4ece77</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-04-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="perform-deseq2-differential-expression-analysis-for-each-trait"
class="section level1">
<h1>Perform DESeq2 differential expression analysis for each trait</h1>
<div id="model-1-expression-prs-sex-expression-pcs"
class="section level2">
<h2>Model 1: expression ~ PRS + sex + expression PCs</h2>
<p><strong>Continuous PRS:</strong></p>
<pre class="r"><code># Load the gene expression data
gene_expr_file &lt;- &quot;data/gene_reads_2017-06-05_v8_whole_blood.gct&quot;
raw_count_df &lt;- fread(gene_expr_file, header = TRUE, sep = &quot;\t&quot;, drop = &quot;id&quot;)

# load protein_coding list
protein_coding &lt;- fread(&quot;data/protein-coding_gene.txt&quot;, sep = &quot;\t&quot;)
protein_coding &lt;- protein_coding[, c(&quot;symbol&quot;, &quot;ensembl_gene_id&quot;)]

# keep only protein-coding genes
raw_count_df &lt;- raw_count_df[raw_count_df$Description %in% protein_coding$symbol, ]

id &lt;- raw_count_df$Name
raw_count &lt;- raw_count_df[, -c(1:2)]

# modify GTEx sample names matching names used in PRS data
colnames(raw_count) &lt;- sub(&quot;^(GTEX-[^-.]+).*&quot;, &quot;\\1&quot;, colnames(raw_count))

matching_samples &lt;- intersect(rownames(metadata), colnames(raw_count))
final_count &lt;- raw_count[ , ..matching_samples]

# prefilter: keep only rows that have a count of at least 10
keep_genes &lt;- rowSums(final_count &gt;= 10) &gt; 0
final_count &lt;- final_count[keep_genes, ]
id &lt;- id[keep_genes]
dim(final_count)

# Loop through each trait and run DESeq2
for (trait in colnames(traits)) {
  
  # Standardize PRS for the current trait
  prs_trait &lt;- traits[,trait]
  prs_trait &lt;- scale(prs_trait) # Standardize PRS to mean = 0, sd = 1
    
  # Add the standardized PRS to the metadata for continuous trait
  metadata[,trait] &lt;- prs_trait
  
  # Create the DESeqDataSet for the current trait
  dds &lt;- DESeqDataSetFromMatrix(
    countData = as.matrix(final_count),  # Raw counts
    colData = metadata[, c(1:6, which(colnames(metadata) == trait))],
    design = as.formula(paste(&quot;~ PC1 + PC2 + PC3 + PC4 + PC5 + sex +&quot;, trait))
  )
  rownames(dds) &lt;- id
  
  # Run DESeq2 analysis
  dds &lt;- DESeq(dds, parallel = TRUE, BPPARAM = MulticoreParam(4))
  
  # Get the results for the current trait
  res &lt;- results(dds)
  
  # Save the results to a file
  write.csv(res, paste0(&quot;differential_expression_&quot;, trait, &quot;_results.csv&quot;))
  
  # print a summary of the results
  print(paste(&quot;Results for trait:&quot;, trait))
  print(summary(res))
  
  # plot the MA-plot for the current trait
  png(paste0(&quot;ma_plot_&quot;, trait, &quot;.png&quot;), width = 800, height = 600)
  plotMA(res, main = paste(&quot;Continuous: MA Plot for&quot;, trait))
  dev.off()
  
  # volcano plot
  res_tableOE &lt;- as.data.frame(res)
  res_tableOE$gene_name &lt;- raw_count_df$Description[keep_genes]
  res_tableOE &lt;- mutate(res_tableOE, threshold_OE = padj &lt; 0.1)
  res_tableOE &lt;- res_tableOE %&gt;% arrange(padj) %&gt;% mutate(genelabels = &quot;&quot;)
  res_tableOE$genelabels[1:10] &lt;- res_tableOE$gene_name[1:10]
  
  volcano_plot &lt;- ggplot(res_tableOE, aes(x = log2FoldChange, y = -log10(padj))) +
    geom_point(aes(colour = threshold_OE)) +
    geom_text_repel(aes(label = genelabels)) +
    ggtitle(paste(&quot;Continuous: Volcano Plot for&quot;, trait)) +
    xlab(&quot;log2 fold change&quot;) + 
    ylab(&quot;-log10 adjusted p-value&quot;) +
    theme(legend.position = &quot;none&quot;,
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
  
  # Save the volcano plot
  png(paste0(&quot;volcano_plot_&quot;, trait, &quot;.png&quot;), width = 800, height = 600)
  print(volcano_plot)
  dev.off()
}</code></pre>
<p><strong>Quantile PRS:</strong></p>
<pre class="r"><code>metadata_file &lt;- &quot;analysis/metadata_quantile.txt&quot;
metadata &lt;- read.csv(metadata_file, header = T, sep = &quot;\t&quot;, stringsAsFactors = T)
metadata$sex &lt;- as.factor(metadata$sex)

traits &lt;- metadata[, 7:43]

# Loop through each trait and run DESeq2
for (trait in colnames(traits)) {
  
  # Create the DESeqDataSet for the current trait
  dds &lt;- DESeqDataSetFromMatrix(
    countData = as.matrix(final_count),  # Raw counts
    colData = metadata[, c(1:6, which(colnames(metadata) == trait))],
    design = as.formula(paste(&quot;~ PC1 + PC2 + PC3 + PC4 + PC5 + sex +&quot;, trait))
  )
  rownames(dds) &lt;- id
  
  # Run DESeq2 analysis
  dds &lt;- DESeq(dds, parallel = TRUE, BPPARAM = MulticoreParam(4))
  
  # Get the results for the current trait
  res &lt;- results(dds)
  
  # Save the results to a file
  write.csv(res, paste0(&quot;differential_expression_&quot;, trait, &quot;_quantile_results.csv&quot;))
  
  # print a summary of the results
  print(paste(&quot;Results for trait:&quot;, trait))
  print(summary(res))
  
  # plot the MA-plot for the current trait
  png(paste0(&quot;ma_plot_quantile_&quot;, trait, &quot;.png&quot;), width = 800, height = 600)
  plotMA(res, main = paste(&quot;Quantile: MA Plot for&quot;, trait))
  dev.off()
  
  # volcano plot
  res_tableOE &lt;- as.data.frame(res)
  res_tableOE$gene_name &lt;- raw_count_df$Description[keep_genes]
  res_tableOE &lt;- mutate(res_tableOE, threshold_OE = padj &lt; 0.1 &amp; 
                          abs(log2FoldChange) &gt;= 0.5)
  res_tableOE &lt;- res_tableOE %&gt;% arrange(padj) %&gt;% mutate(genelabels = &quot;&quot;)
  res_tableOE$genelabels[1:10] &lt;- res_tableOE$gene_name[1:10]
  
  volcano_plot &lt;- ggplot(res_tableOE, aes(x = log2FoldChange, y = -log10(padj))) +
    geom_point(aes(colour = threshold_OE)) +
    geom_text_repel(aes(label = genelabels)) +
    ggtitle(paste(&quot;Quantile:Volcano Plot for&quot;, trait)) +
    xlab(&quot;log2 fold change&quot;) + 
    ylab(&quot;-log10 adjusted p-value&quot;) +
    theme(legend.position = &quot;none&quot;,
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
  
  # Save the volcano plot
  png(paste0(&quot;volcano_plot_quantile_&quot;, trait, &quot;.png&quot;), width = 800, height = 600)
  print(volcano_plot)
  dev.off()
}</code></pre>
</div>
<div id="model-2-expression-prs-sex-expression-pcs-2-genotype-pcs"
class="section level2">
<h2>Model 2: expression ~ PRS + sex + expression PCs + 2 genotype
PCs</h2>
<p><strong>Continuous PRS:</strong></p>
<pre class="r"><code># Load the gene expression data
gene_expr_file &lt;- &quot;data/gene_reads_2017-06-05_v8_whole_blood.gct&quot;
raw_count_df &lt;- fread(gene_expr_file, header = TRUE, sep = &quot;\t&quot;, drop = &quot;id&quot;)

# load protein_coding list
protein_coding &lt;- fread(&quot;data/protein-coding_gene.txt&quot;, 
                        sep = &quot;\t&quot;)
protein_coding &lt;- protein_coding[, c(&quot;symbol&quot;, &quot;ensembl_gene_id&quot;)]

# keep only protein-coding genes
raw_count_df &lt;- raw_count_df[raw_count_df$Description %in% protein_coding$symbol, ]

id &lt;- raw_count_df$Name
raw_count &lt;- raw_count_df[, -c(1:2)]

# modify GTEx sample names matching names used in PRS data
colnames(raw_count) &lt;- sub(&quot;^(GTEX-[^-.]+).*&quot;, &quot;\\1&quot;, colnames(raw_count))

matching_samples &lt;- intersect(rownames(metadata), colnames(raw_count))
final_count &lt;- raw_count[ , ..matching_samples]

# prefilter: keep only rows that have a count of at least 10
keep_genes &lt;- rowSums(final_count &gt;= 10) &gt; 0
final_count &lt;- final_count[keep_genes, ]
id &lt;- id[keep_genes]
dim(final_count)</code></pre>
<pre><code>[1] 16893   670</code></pre>
<pre class="r"><code># obtain genotype PCs
geno_pc &lt;- read.table(&quot;data/pca.eigenvec&quot;)
names(geno_pc) = c(&quot;FID&quot;,&quot;IID&quot;,paste0(&quot;geno_PC&quot;, c(1:(ncol(geno_pc)-2))))

eigenval &lt;- scan(&quot;data/pca.eigenval&quot;)
pve &lt;- data.frame(PC = 1:20, pve = eigenval/sum(eigenval)*100)

ggplot(geno_pc) + geom_point(aes(x = geno_PC1, y = geno_PC2)) + 
  labs(x = paste0(&quot;PC1 (&quot;, signif(pve$pve[1], 3), &quot;%)&quot;), 
       y = paste0(&quot;PC2 (&quot;, signif(pve$pve[2], 3), &quot;%)&quot;)) + theme_classic()</code></pre>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/51ffd48702cd994a05a864da79f2fcc8ad9989b9/docs/figure/differential_expression.Rmd/unnamed-chunk-4-1.png" target="_blank">51ffd48</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(geno_pc) + geom_point(aes(x = geno_PC1, y = geno_PC3)) + 
    labs(x = paste0(&quot;PC1 (&quot;, signif(pve$pve[1], 3), &quot;%)&quot;), 
         y = paste0(&quot;PC3 (&quot;, signif(pve$pve[3], 3), &quot;%)&quot;)) + theme_classic()</code></pre>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/51ffd48702cd994a05a864da79f2fcc8ad9989b9/docs/figure/differential_expression.Rmd/unnamed-chunk-4-2.png" target="_blank">51ffd48</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pve, aes(PC, pve)) + geom_point() + geom_line() +
  labs(x = &quot;Genotype PC&quot;, y = &quot;Percentage variance explained&quot;) + theme_classic()</code></pre>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-3">
Past versions of unnamed-chunk-4-3.png
</button>
</p>
<div id="fig-unnamed-chunk-4-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yixuan-chen-elisa/prs/blob/51ffd48702cd994a05a864da79f2fcc8ad9989b9/docs/figure/differential_expression.Rmd/unnamed-chunk-4-3.png" target="_blank">51ffd48</a>
</td>
<td>
ElisaChen
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>metadata &lt;- cbind(metadata, geno_pc[,3:ncol(geno_pc)])

# Loop through each trait and run DESeq2
for (trait in colnames(traits)) {
  
  # Standardize PRS for the current trait
  prs_trait &lt;- traits[,trait]
  prs_trait &lt;- scale(prs_trait) # Standardize PRS to mean = 0, sd = 1
    
  # Add the standardized PRS to the metadata for continuous trait
  metadata[,trait] &lt;- prs_trait
  
  # Create the DESeqDataSet for the current trait
  dds &lt;- DESeqDataSetFromMatrix(
    countData = as.matrix(final_count),  # Raw counts
    colData = metadata[, c(1:6, 44:45, which(colnames(metadata) == trait))],
    design = as.formula(paste(&quot;~ PC1 + PC2 + PC3 + PC4 + PC5 + sex + geno_PC1 + geno_PC2 +&quot;, trait))
  )
  rownames(dds) &lt;- id
  
  # Run DESeq2 analysis
  dds &lt;- DESeq(dds, parallel = TRUE, BPPARAM = MulticoreParam(4))
  
  # Get the results for the current trait
  res &lt;- results(dds)
  
  # Save the results to a file
  write.csv(res, paste0(&quot;differential_expression_&quot;, trait, &quot;_results_M2.csv&quot;))
  
  # print a summary of the results
  print(paste(&quot;Results for trait:&quot;, trait))
  print(summary(res))
  
  # plot the MA-plot for the current trait
  png(paste0(&quot;ma_plot_&quot;, trait, &quot;_M2.png&quot;), width = 800, height = 600)
  plotMA(res, main = paste(&quot;Continuous (M2): MA Plot for&quot;, trait))
  dev.off()
  
  # volcano plot
  res_tableOE &lt;- as.data.frame(res)
  res_tableOE$gene_name &lt;- raw_count_df$Description[keep_genes]
  res_tableOE &lt;- mutate(res_tableOE, threshold_OE = padj &lt; 0.1)
  res_tableOE &lt;- res_tableOE %&gt;% arrange(padj) %&gt;% mutate(genelabels = &quot;&quot;)
  res_tableOE$genelabels[1:10] &lt;- res_tableOE$gene_name[1:10]
  
  volcano_plot &lt;- ggplot(res_tableOE, aes(x = log2FoldChange, y = -log10(padj))) +
    geom_point(aes(colour = threshold_OE)) +
    geom_text_repel(aes(label = genelabels)) +
    ggtitle(paste(&quot;Continuous (M2): Volcano Plot for&quot;, trait)) +
    xlab(&quot;log2 fold change&quot;) + 
    ylab(&quot;-log10 adjusted p-value&quot;) +
    theme(legend.position = &quot;none&quot;,
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
  
  # Save the volcano plot
  png(paste0(&quot;volcano_plot_&quot;, trait, &quot;_M2.png&quot;), width = 800, height = 600)
  print(volcano_plot)
  dev.off()
}</code></pre>
<p><strong>Quantile PRS:</strong></p>
<pre class="r"><code>metadata_file &lt;- &quot;analysis/metadata_quantile.txt&quot;
metadata &lt;- read.csv(metadata_file, header = T, sep = &quot;\t&quot;, stringsAsFactors = T)
metadata$sex &lt;- as.factor(metadata$sex)

traits &lt;- metadata[, 7:43]

metadata &lt;- cbind(metadata, geno_pc[,3:ncol(geno_pc)])

# Loop through each trait and run DESeq2
for (trait in colnames(traits)) {
  
  # Create the DESeqDataSet for the current trait
  dds &lt;- DESeqDataSetFromMatrix(
    countData = as.matrix(final_count),  # Raw counts
    colData = metadata[, c(1:6, which(colnames(metadata) == trait))],
    design = as.formula(paste(&quot;~ PC1 + PC2 + PC3 + PC4 + PC5 + sex + geno_PC1 + geno_PC2 +&quot;, trait))
  )
  rownames(dds) &lt;- id
  
  # Run DESeq2 analysis
  dds &lt;- DESeq(dds, parallel = TRUE, BPPARAM = MulticoreParam(4))
  
  # Get the results for the current trait
  res &lt;- results(dds)
  
  # Save the results to a file
  write.csv(res, paste0(&quot;differential_expression_&quot;, trait, &quot;_quantile_results_M2.csv&quot;))
  
  # print a summary of the results
  print(paste(&quot;Results for trait:&quot;, trait))
  print(summary(res))
  
  # plot the MA-plot for the current trait
  png(paste0(&quot;ma_plot_quantile_&quot;, trait, &quot;_M2.png&quot;), width = 800, height = 600)
  plotMA(res, main = paste(&quot;Quantile (M2): MA Plot for&quot;, trait))
  dev.off()
  
  # volcano plot
  res_tableOE &lt;- as.data.frame(res)
  res_tableOE$gene_name &lt;- raw_count_df$Description[keep_genes]
  res_tableOE &lt;- mutate(res_tableOE, threshold_OE = padj &lt; 0.1 &amp; 
                          abs(log2FoldChange) &gt;= 0.5)
  res_tableOE &lt;- res_tableOE %&gt;% arrange(padj) %&gt;% mutate(genelabels = &quot;&quot;)
  res_tableOE$genelabels[1:10] &lt;- res_tableOE$gene_name[1:10]
  
  volcano_plot &lt;- ggplot(res_tableOE, aes(x = log2FoldChange, y = -log10(padj))) +
    geom_point(aes(colour = threshold_OE)) +
    geom_text_repel(aes(label = genelabels)) +
    ggtitle(paste(&quot;Quantile (M2):Volcano Plot for&quot;, trait)) +
    xlab(&quot;log2 fold change&quot;) + 
    ylab(&quot;-log10 adjusted p-value&quot;) +
    theme(legend.position = &quot;none&quot;,
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
  
  # Save the volcano plot
  png(paste0(&quot;volcano_plot_quantile_&quot;, trait, &quot;_M2.png&quot;), width = 800, height = 600)
  print(volcano_plot)
  dev.off()
}</code></pre>
</div>
</div>
<div id="go-enrichment-analysis" class="section level1">
<h1>GO enrichment Analysis</h1>
<p>Gene Ontology enrichment is applied to both models, with and without
adjusting 2 genotype PCs, for both continuous &amp; quantile PRS.</p>
<p><strong>Continuous PRS:</strong></p>
<pre class="r"><code>dir_path &lt;- &quot;analysis/continuous&quot;
files &lt;- list.files(dir_path, pattern = &quot;differential_expression_.*_results_M2.csv&quot;, full.names = TRUE)

for (file in files) {
  trait &lt;- gsub(&quot;differential_expression_(.*)_results_M2.csv&quot;, &quot;\\1&quot;, basename(file))
  trait
  res_tableOE &lt;- read.csv(file, header = T, row.names = 1)
  
  deGenes &lt;- res_tableOE[res_tableOE$padj &lt; 0.1 &amp; 
                           abs(res_tableOE$log2FoldChange) &gt;= 0.5, ]
  deGenes$gene_id &lt;- gsub(&quot;\\.\\d+$&quot;, &quot;&quot;, rownames(deGenes))
  
  gse &lt;- enrichGO(gene = deGenes$gene_id, ont = &quot;BP&quot;, 
                OrgDb =&quot;org.Hs.eg.db&quot;, keyType = &quot;ENSEMBL&quot;, readable = T)
  
  write.csv(as.data.frame(gse), file = paste0(&quot;GO_enrichment_&quot;, trait, &quot;_results_M2.csv&quot;))
  
  gse &lt;- as.data.frame(gse)
  gse$GeneRatio_num &lt;- as.numeric(sapply(strsplit(gse$GeneRatio, &quot;/&quot;), 
                                         function(x) x[1])) /
                     as.numeric(sapply(strsplit(gse$GeneRatio, &quot;/&quot;), function(x) x[2]))

  gse$BgRatio_num &lt;- as.numeric(sapply(strsplit(gse$BgRatio, &quot;/&quot;), function(x) x[1])) / 
                     as.numeric(sapply(strsplit(gse$BgRatio, &quot;/&quot;), function(x) x[2]))
  
  gse &lt;- cbind(gse, FoldEnrich = gse$GeneRatio_num/gse$BgRatio_num)
  
  if (nrow(gse) &gt;= 20) {
    enrich_plot &lt;- plotEnrich(gse[1:20,], plot_type = &quot;dot&quot;, scale_ratio = 0.5) +
    labs(title = paste(&quot;Continuous (M2): Enrichment plot for&quot;, trait)) + 
    theme(plot.title = element_text(size = 10)) 
  }else{
    enrich_plot &lt;- plotEnrich(gse, plot_type = &quot;dot&quot;, scale_ratio = 0.5) +
    labs(title = paste(&quot;Continuous (M2): Enrichment plot for&quot;, trait)) + 
    theme(plot.title = element_text(size = 10)) 
  }
  ggsave(paste0(&quot;enrichment_plot_&quot;, trait, &quot;_M2.png&quot;), plot = enrich_plot, width = 8, height = 6)
}</code></pre>
<p><strong>Quantile PRS:</strong></p>
<pre class="r"><code>dir_path &lt;- &quot;analysis/quantile&quot;
files &lt;- list.files(dir_path, pattern = &quot;differential_expression_.*_quantile_results_M2.csv&quot;,
                    full.names = TRUE)

for (file in files) {
  trait &lt;- gsub(&quot;differential_expression_(.*)_quantile_results_M2.csv&quot;, &quot;\\1&quot;, basename(file))
  trait
  res_tableOE &lt;- read.csv(file, header = T, row.names = 1)
  
  deGenes &lt;- res_tableOE[res_tableOE$padj &lt; 0.1 &amp; 
                           abs(res_tableOE$log2FoldChange) &gt;= 0.5, ]
  deGenes$gene_id &lt;- gsub(&quot;\\.\\d+$&quot;, &quot;&quot;, rownames(deGenes))
  
  gse &lt;- enrichGO(gene = deGenes$gene_id, ont = &quot;BP&quot;, 
                OrgDb =&quot;org.Hs.eg.db&quot;, keyType = &quot;ENSEMBL&quot;)
  
  write.csv(as.data.frame(gse), file = paste0(&quot;GO_enrichment_quantile_&quot;, trait, &quot;_results_M2.csv&quot;))
  
  gse &lt;- as.data.frame(gse)
  gse$GeneRatio_num &lt;- as.numeric(sapply(strsplit(gse$GeneRatio, &quot;/&quot;), 
                                         function(x) x[1])) /
                     as.numeric(sapply(strsplit(gse$GeneRatio, &quot;/&quot;), function(x) x[2]))

  gse$BgRatio_num &lt;- as.numeric(sapply(strsplit(gse$BgRatio, &quot;/&quot;), function(x) x[1])) / 
                     as.numeric(sapply(strsplit(gse$BgRatio, &quot;/&quot;), function(x) x[2]))
  
  gse &lt;- cbind(gse, FoldEnrich = gse$GeneRatio_num/gse$BgRatio_num)
  
  if (nrow(gse) &gt;= 20) {
    enrich_plot &lt;- plotEnrich(gse[1:20,], plot_type = &quot;dot&quot;, scale_ratio = 0.3) +
    labs(title = paste(&quot;Quantile (M2): Enrichment plot for&quot;, trait)) + 
    theme(plot.title = element_text(size = 10)) 
  }else{
    enrich_plot &lt;- plotEnrich(gse, plot_type = &quot;dot&quot;, scale_ratio = 0.3) +
    labs(title = paste(&quot;Quantile (M2): Enrichment plot for&quot;, trait)) + 
    theme(plot.title = element_text(size = 10)) 
  }
  
  ggsave(paste0(&quot;enrichment_plot_quantile_&quot;, trait, &quot;_M2.png&quot;), plot = enrich_plot, width = 8, height = 6)
}</code></pre>
</div>
<div id="result" class="section level1">
<h1>Result</h1>
<div id="summary-table" class="section level2">
<h2>Summary table</h2>
<table>
<caption>Continuous PRS without adjusting genotype PCs</caption>
<colgroup>
<col width="40%" />
<col width="29%" />
<col width="3%" />
<col width="3%" />
<col width="23%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Trait</th>
<th align="right">Significant differential expressed genes</th>
<th align="right">Up</th>
<th align="right">Down</th>
<th align="right">Significant GO enriched pathways</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Basophil.count</td>
<td align="right">402</td>
<td align="right">363</td>
<td align="right">39</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Basophil.percentage.of.white.cells</td>
<td align="right">200</td>
<td align="right">89</td>
<td align="right">111</td>
<td align="right">109</td>
</tr>
<tr class="odd">
<td align="left">Celiac_GCST90014442</td>
<td align="right">97</td>
<td align="right">55</td>
<td align="right">42</td>
<td align="right">8</td>
</tr>
<tr class="even">
<td align="left">Celiac_GCST90468120</td>
<td align="right">493</td>
<td align="right">25</td>
<td align="right">468</td>
<td align="right">6</td>
</tr>
<tr class="odd">
<td align="left">Eosinophil.count</td>
<td align="right">338</td>
<td align="right">128</td>
<td align="right">210</td>
<td align="right">75</td>
</tr>
<tr class="even">
<td align="left">Eosinophil.percentage.of.white.cells</td>
<td align="right">209</td>
<td align="right">85</td>
<td align="right">124</td>
<td align="right">13</td>
</tr>
<tr class="odd">
<td align="left">Hematocrit</td>
<td align="right">337</td>
<td align="right">57</td>
<td align="right">280</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">Hemoglobin.concentration</td>
<td align="right">132</td>
<td align="right">57</td>
<td align="right">75</td>
<td align="right">12</td>
</tr>
<tr class="odd">
<td align="left">High.light.scatter.reticulocyte.count</td>
<td align="right">239</td>
<td align="right">52</td>
<td align="right">187</td>
<td align="right">73</td>
</tr>
<tr class="even">
<td
align="left">High.light.scatter.reticulocyte.percentage.of.red.cells</td>
<td align="right">325</td>
<td align="right">94</td>
<td align="right">231</td>
<td align="right">146</td>
</tr>
<tr class="odd">
<td align="left">IBD_GCST90013901</td>
<td align="right">559</td>
<td align="right">522</td>
<td align="right">37</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">IBD_GCST90013951</td>
<td align="right">380</td>
<td align="right">338</td>
<td align="right">42</td>
<td align="right">192</td>
</tr>
<tr class="odd">
<td align="left">Immature.fraction.of.reticulocytes</td>
<td align="right">288</td>
<td align="right">134</td>
<td align="right">154</td>
<td align="right">71</td>
</tr>
<tr class="even">
<td align="left">LUPUS_GCST003156</td>
<td align="right">119</td>
<td align="right">79</td>
<td align="right">40</td>
<td align="right">48</td>
</tr>
<tr class="odd">
<td align="left">LUPUS_GCST011096</td>
<td align="right">131</td>
<td align="right">89</td>
<td align="right">42</td>
<td align="right">85</td>
</tr>
<tr class="even">
<td align="left">Lymphocyte.count</td>
<td align="right">729</td>
<td align="right">347</td>
<td align="right">382</td>
<td align="right">59</td>
</tr>
<tr class="odd">
<td align="left">Lymphocyte.percentage.of.white.cells</td>
<td align="right">927</td>
<td align="right">305</td>
<td align="right">622</td>
<td align="right">31</td>
</tr>
<tr class="even">
<td align="left">Mean.corpuscular.hemoglobin</td>
<td align="right">402</td>
<td align="right">22</td>
<td align="right">380</td>
<td align="right">55</td>
</tr>
<tr class="odd">
<td align="left">Mean.corpuscular.hemoglobin.concentration</td>
<td align="right">788</td>
<td align="right">69</td>
<td align="right">719</td>
<td align="right">189</td>
</tr>
<tr class="even">
<td align="left">Mean.corpuscular.volume</td>
<td align="right">162</td>
<td align="right">22</td>
<td align="right">140</td>
<td align="right">44</td>
</tr>
<tr class="odd">
<td align="left">Mean.platelet.volume</td>
<td align="right">146</td>
<td align="right">30</td>
<td align="right">116</td>
<td align="right">22</td>
</tr>
<tr class="even">
<td align="left">Mean.reticulocyte.volume</td>
<td align="right">312</td>
<td align="right">148</td>
<td align="right">164</td>
<td align="right">99</td>
</tr>
<tr class="odd">
<td align="left">Mean.sphered.corpuscular.volume</td>
<td align="right">533</td>
<td align="right">81</td>
<td align="right">452</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">Monocyte.count</td>
<td align="right">348</td>
<td align="right">180</td>
<td align="right">168</td>
<td align="right">35</td>
</tr>
<tr class="odd">
<td align="left">Monocyte.percentage.of.white.cells</td>
<td align="right">179</td>
<td align="right">42</td>
<td align="right">137</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">Neutrophil.count</td>
<td align="right">1358</td>
<td align="right">1260</td>
<td align="right">98</td>
<td align="right">46</td>
</tr>
<tr class="odd">
<td align="left">Neutrophil.percentage.of.white.cells</td>
<td align="right">421</td>
<td align="right">357</td>
<td align="right">64</td>
<td align="right">20</td>
</tr>
<tr class="even">
<td align="left">Platelet.count</td>
<td align="right">78</td>
<td align="right">51</td>
<td align="right">27</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">Platelet.crit</td>
<td align="right">113</td>
<td align="right">82</td>
<td align="right">31</td>
<td align="right">8</td>
</tr>
<tr class="even">
<td align="left">Platelet.distribution.width</td>
<td align="right">105</td>
<td align="right">32</td>
<td align="right">73</td>
<td align="right">22</td>
</tr>
<tr class="odd">
<td align="left">Red.blood.cell.count</td>
<td align="right">130</td>
<td align="right">81</td>
<td align="right">49</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Red.cell.distribution.width</td>
<td align="right">999</td>
<td align="right">396</td>
<td align="right">603</td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td align="left">Reticulocyte.count</td>
<td align="right">140</td>
<td align="right">60</td>
<td align="right">80</td>
<td align="right">30</td>
</tr>
<tr class="even">
<td align="left">Reticulocyte.fraction.of.red.cells</td>
<td align="right">225</td>
<td align="right">122</td>
<td align="right">103</td>
<td align="right">28</td>
</tr>
<tr class="odd">
<td align="left">T1D_GCST90000529</td>
<td align="right">618</td>
<td align="right">540</td>
<td align="right">78</td>
<td align="right">30</td>
</tr>
<tr class="even">
<td align="left">T1D_GCST90014023</td>
<td align="right">205</td>
<td align="right">135</td>
<td align="right">70</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">White.blood.cell.count</td>
<td align="right">1314</td>
<td align="right">1118</td>
<td align="right">196</td>
<td align="right">34</td>
</tr>
</tbody>
</table>
<table>
<caption>Continuous PRS with adjusting genotype PCs</caption>
<colgroup>
<col width="40%" />
<col width="29%" />
<col width="3%" />
<col width="3%" />
<col width="23%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Trait</th>
<th align="right">Significant differential expressed genes</th>
<th align="right">Up</th>
<th align="right">Down</th>
<th align="right">Significant GO enriched pathways</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Basophil.count</td>
<td align="right">385</td>
<td align="right">345</td>
<td align="right">40</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Basophil.percentage.of.white.cells</td>
<td align="right">203</td>
<td align="right">93</td>
<td align="right">110</td>
<td align="right">107</td>
</tr>
<tr class="odd">
<td align="left">Celiac_GCST90014442</td>
<td align="right">94</td>
<td align="right">55</td>
<td align="right">39</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">Celiac_GCST90468120</td>
<td align="right">493</td>
<td align="right">25</td>
<td align="right">468</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">Eosinophil.count</td>
<td align="right">333</td>
<td align="right">126</td>
<td align="right">207</td>
<td align="right">75</td>
</tr>
<tr class="even">
<td align="left">Eosinophil.percentage.of.white.cells</td>
<td align="right">211</td>
<td align="right">87</td>
<td align="right">124</td>
<td align="right">6</td>
</tr>
<tr class="odd">
<td align="left">Hematocrit</td>
<td align="right">318</td>
<td align="right">51</td>
<td align="right">267</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Hemoglobin.concentration</td>
<td align="right">120</td>
<td align="right">54</td>
<td align="right">66</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">High.light.scatter.reticulocyte.count</td>
<td align="right">237</td>
<td align="right">52</td>
<td align="right">185</td>
<td align="right">72</td>
</tr>
<tr class="even">
<td
align="left">High.light.scatter.reticulocyte.percentage.of.red.cells</td>
<td align="right">328</td>
<td align="right">101</td>
<td align="right">227</td>
<td align="right">139</td>
</tr>
<tr class="odd">
<td align="left">IBD_GCST90013901</td>
<td align="right">561</td>
<td align="right">525</td>
<td align="right">36</td>
<td align="right">18</td>
</tr>
<tr class="even">
<td align="left">IBD_GCST90013951</td>
<td align="right">378</td>
<td align="right">340</td>
<td align="right">38</td>
<td align="right">179</td>
</tr>
<tr class="odd">
<td align="left">Immature.fraction.of.reticulocytes</td>
<td align="right">294</td>
<td align="right">142</td>
<td align="right">152</td>
<td align="right">69</td>
</tr>
<tr class="even">
<td align="left">LUPUS_GCST003156</td>
<td align="right">123</td>
<td align="right">82</td>
<td align="right">41</td>
<td align="right">50</td>
</tr>
<tr class="odd">
<td align="left">LUPUS_GCST011096</td>
<td align="right">138</td>
<td align="right">94</td>
<td align="right">44</td>
<td align="right">109</td>
</tr>
<tr class="even">
<td align="left">Lymphocyte.count</td>
<td align="right">718</td>
<td align="right">341</td>
<td align="right">377</td>
<td align="right">62</td>
</tr>
<tr class="odd">
<td align="left">Lymphocyte.percentage.of.white.cells</td>
<td align="right">923</td>
<td align="right">309</td>
<td align="right">614</td>
<td align="right">32</td>
</tr>
<tr class="even">
<td align="left">Mean.corpuscular.hemoglobin</td>
<td align="right">381</td>
<td align="right">22</td>
<td align="right">359</td>
<td align="right">63</td>
</tr>
<tr class="odd">
<td align="left">Mean.corpuscular.hemoglobin.concentration</td>
<td align="right">785</td>
<td align="right">71</td>
<td align="right">714</td>
<td align="right">243</td>
</tr>
<tr class="even">
<td align="left">Mean.corpuscular.volume</td>
<td align="right">162</td>
<td align="right">22</td>
<td align="right">140</td>
<td align="right">40</td>
</tr>
<tr class="odd">
<td align="left">Mean.platelet.volume</td>
<td align="right">147</td>
<td align="right">31</td>
<td align="right">116</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">Mean.reticulocyte.volume</td>
<td align="right">312</td>
<td align="right">147</td>
<td align="right">165</td>
<td align="right">104</td>
</tr>
<tr class="odd">
<td align="left">Mean.sphered.corpuscular.volume</td>
<td align="right">534</td>
<td align="right">84</td>
<td align="right">450</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">Monocyte.count</td>
<td align="right">366</td>
<td align="right">188</td>
<td align="right">178</td>
<td align="right">41</td>
</tr>
<tr class="odd">
<td align="left">Monocyte.percentage.of.white.cells</td>
<td align="right">182</td>
<td align="right">43</td>
<td align="right">139</td>
<td align="right">17</td>
</tr>
<tr class="even">
<td align="left">Neutrophil.count</td>
<td align="right">1374</td>
<td align="right">1271</td>
<td align="right">103</td>
<td align="right">44</td>
</tr>
<tr class="odd">
<td align="left">Neutrophil.percentage.of.white.cells</td>
<td align="right">428</td>
<td align="right">361</td>
<td align="right">67</td>
<td align="right">21</td>
</tr>
<tr class="even">
<td align="left">Platelet.count</td>
<td align="right">73</td>
<td align="right">47</td>
<td align="right">26</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">Platelet.crit</td>
<td align="right">110</td>
<td align="right">80</td>
<td align="right">30</td>
<td align="right">8</td>
</tr>
<tr class="even">
<td align="left">Platelet.distribution.width</td>
<td align="right">93</td>
<td align="right">31</td>
<td align="right">62</td>
<td align="right">22</td>
</tr>
<tr class="odd">
<td align="left">Red.blood.cell.count</td>
<td align="right">110</td>
<td align="right">67</td>
<td align="right">43</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Red.cell.distribution.width</td>
<td align="right">990</td>
<td align="right">401</td>
<td align="right">589</td>
<td align="right">19</td>
</tr>
<tr class="odd">
<td align="left">Reticulocyte.count</td>
<td align="right">129</td>
<td align="right">57</td>
<td align="right">72</td>
<td align="right">34</td>
</tr>
<tr class="even">
<td align="left">Reticulocyte.fraction.of.red.cells</td>
<td align="right">213</td>
<td align="right">120</td>
<td align="right">93</td>
<td align="right">27</td>
</tr>
<tr class="odd">
<td align="left">T1D_GCST90000529</td>
<td align="right">619</td>
<td align="right">537</td>
<td align="right">82</td>
<td align="right">32</td>
</tr>
<tr class="even">
<td align="left">T1D_GCST90014023</td>
<td align="right">192</td>
<td align="right">123</td>
<td align="right">69</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">White.blood.cell.count</td>
<td align="right">1313</td>
<td align="right">1118</td>
<td align="right">195</td>
<td align="right">35</td>
</tr>
</tbody>
</table>
<table style="width:100%;">
<caption>Quantile PRS without adjusting genotype PCs</caption>
<colgroup>
<col width="43%" />
<col width="27%" />
<col width="3%" />
<col width="3%" />
<col width="22%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Trait</th>
<th align="right">Significant differential expressed genes</th>
<th align="right">Up</th>
<th align="right">Down</th>
<th align="right">Significant GO enriched pathways</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Basophil.count_quantile</td>
<td align="right">712</td>
<td align="right">685</td>
<td align="right">27</td>
<td align="right">157</td>
</tr>
<tr class="even">
<td align="left">Basophil.percentage.of.white.cells_quantile</td>
<td align="right">184</td>
<td align="right">64</td>
<td align="right">120</td>
<td align="right">75</td>
</tr>
<tr class="odd">
<td align="left">Celiac_GCST90014442_quantile</td>
<td align="right">336</td>
<td align="right">124</td>
<td align="right">212</td>
<td align="right">301</td>
</tr>
<tr class="even">
<td align="left">Celiac_GCST90468120_quantile</td>
<td align="right">212</td>
<td align="right">42</td>
<td align="right">170</td>
<td align="right">35</td>
</tr>
<tr class="odd">
<td align="left">Eosinophil.count_quantile</td>
<td align="right">507</td>
<td align="right">197</td>
<td align="right">310</td>
<td align="right">152</td>
</tr>
<tr class="even">
<td align="left">Eosinophil.percentage.of.white.cells_quantile</td>
<td align="right">431</td>
<td align="right">218</td>
<td align="right">213</td>
<td align="right">120</td>
</tr>
<tr class="odd">
<td align="left">Hematocrit_quantile</td>
<td align="right">269</td>
<td align="right">118</td>
<td align="right">151</td>
<td align="right">87</td>
</tr>
<tr class="even">
<td align="left">Hemoglobin.concentration_quantile</td>
<td align="right">160</td>
<td align="right">37</td>
<td align="right">123</td>
<td align="right">62</td>
</tr>
<tr class="odd">
<td align="left">High.light.scatter.reticulocyte.count_quantile</td>
<td align="right">225</td>
<td align="right">93</td>
<td align="right">132</td>
<td align="right">113</td>
</tr>
<tr class="even">
<td
align="left">High.light.scatter.reticulocyte.percentage.of.red.cells_quantile</td>
<td align="right">312</td>
<td align="right">141</td>
<td align="right">171</td>
<td align="right">170</td>
</tr>
<tr class="odd">
<td align="left">IBD_GCST90013901_quantile</td>
<td align="right">272</td>
<td align="right">204</td>
<td align="right">68</td>
<td align="right">177</td>
</tr>
<tr class="even">
<td align="left">IBD_GCST90013951_quantile</td>
<td align="right">346</td>
<td align="right">256</td>
<td align="right">90</td>
<td align="right">186</td>
</tr>
<tr class="odd">
<td align="left">Immature.fraction.of.reticulocytes_quantile</td>
<td align="right">473</td>
<td align="right">128</td>
<td align="right">345</td>
<td align="right">200</td>
</tr>
<tr class="even">
<td align="left">LUPUS_GCST003156_quantile</td>
<td align="right">205</td>
<td align="right">96</td>
<td align="right">109</td>
<td align="right">21</td>
</tr>
<tr class="odd">
<td align="left">LUPUS_GCST011096_quantile</td>
<td align="right">493</td>
<td align="right">52</td>
<td align="right">441</td>
<td align="right">179</td>
</tr>
<tr class="even">
<td align="left">Lymphocyte.count_quantile</td>
<td align="right">1456</td>
<td align="right">852</td>
<td align="right">604</td>
<td align="right">207</td>
</tr>
<tr class="odd">
<td align="left">Lymphocyte.percentage.of.white.cells_quantile</td>
<td align="right">532</td>
<td align="right">289</td>
<td align="right">243</td>
<td align="right">129</td>
</tr>
<tr class="even">
<td align="left">Mean.corpuscular.hemoglobin_quantile</td>
<td align="right">410</td>
<td align="right">23</td>
<td align="right">387</td>
<td align="right">89</td>
</tr>
<tr class="odd">
<td align="left">Mean.corpuscular.hemoglobin.concentration_quantile</td>
<td align="right">317</td>
<td align="right">86</td>
<td align="right">231</td>
<td align="right">292</td>
</tr>
<tr class="even">
<td align="left">Mean.corpuscular.volume_quantile</td>
<td align="right">453</td>
<td align="right">58</td>
<td align="right">395</td>
<td align="right">149</td>
</tr>
<tr class="odd">
<td align="left">Mean.platelet.volume_quantile</td>
<td align="right">209</td>
<td align="right">85</td>
<td align="right">124</td>
<td align="right">90</td>
</tr>
<tr class="even">
<td align="left">Mean.reticulocyte.volume_quantile</td>
<td align="right">406</td>
<td align="right">115</td>
<td align="right">291</td>
<td align="right">123</td>
</tr>
<tr class="odd">
<td align="left">Mean.sphered.corpuscular.volume_quantile</td>
<td align="right">355</td>
<td align="right">76</td>
<td align="right">279</td>
<td align="right">66</td>
</tr>
<tr class="even">
<td align="left">Monocyte.count_quantile</td>
<td align="right">1175</td>
<td align="right">845</td>
<td align="right">330</td>
<td align="right">138</td>
</tr>
<tr class="odd">
<td align="left">Monocyte.percentage.of.white.cells_quantile</td>
<td align="right">566</td>
<td align="right">50</td>
<td align="right">516</td>
<td align="right">90</td>
</tr>
<tr class="even">
<td align="left">Neutrophil.count_quantile</td>
<td align="right">806</td>
<td align="right">736</td>
<td align="right">70</td>
<td align="right">99</td>
</tr>
<tr class="odd">
<td align="left">Neutrophil.percentage.of.white.cells_quantile</td>
<td align="right">1451</td>
<td align="right">1330</td>
<td align="right">121</td>
<td align="right">385</td>
</tr>
<tr class="even">
<td align="left">Platelet.count_quantile</td>
<td align="right">248</td>
<td align="right">59</td>
<td align="right">189</td>
<td align="right">114</td>
</tr>
<tr class="odd">
<td align="left">Platelet.crit_quantile</td>
<td align="right">314</td>
<td align="right">100</td>
<td align="right">214</td>
<td align="right">116</td>
</tr>
<tr class="even">
<td align="left">Platelet.distribution.width_quantile</td>
<td align="right">469</td>
<td align="right">149</td>
<td align="right">320</td>
<td align="right">126</td>
</tr>
<tr class="odd">
<td align="left">Red.blood.cell.count_quantile</td>
<td align="right">291</td>
<td align="right">157</td>
<td align="right">134</td>
<td align="right">81</td>
</tr>
<tr class="even">
<td align="left">Red.cell.distribution.width_quantile</td>
<td align="right">399</td>
<td align="right">287</td>
<td align="right">112</td>
<td align="right">45</td>
</tr>
<tr class="odd">
<td align="left">Reticulocyte.count_quantile</td>
<td align="right">170</td>
<td align="right">74</td>
<td align="right">96</td>
<td align="right">88</td>
</tr>
<tr class="even">
<td align="left">Reticulocyte.fraction.of.red.cells_quantile</td>
<td align="right">517</td>
<td align="right">458</td>
<td align="right">59</td>
<td align="right">171</td>
</tr>
<tr class="odd">
<td align="left">T1D_GCST90000529_quantile</td>
<td align="right">1559</td>
<td align="right">1438</td>
<td align="right">121</td>
<td align="right">300</td>
</tr>
<tr class="even">
<td align="left">T1D_GCST90014023_quantile</td>
<td align="right">575</td>
<td align="right">504</td>
<td align="right">71</td>
<td align="right">87</td>
</tr>
<tr class="odd">
<td align="left">White.blood.cell.count_quantile</td>
<td align="right">1216</td>
<td align="right">1082</td>
<td align="right">134</td>
<td align="right">154</td>
</tr>
</tbody>
</table>
<table style="width:100%;">
<caption>Quantile PRS with adjusting genotype PCs</caption>
<colgroup>
<col width="43%" />
<col width="27%" />
<col width="3%" />
<col width="3%" />
<col width="22%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Trait</th>
<th align="right">Significant differential expressed genes</th>
<th align="right">Up</th>
<th align="right">Down</th>
<th align="right">Significant GO enriched pathways</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Basophil.count_quantile</td>
<td align="right">707</td>
<td align="right">682</td>
<td align="right">25</td>
<td align="right">171</td>
</tr>
<tr class="even">
<td align="left">Basophil.percentage.of.white.cells_quantile</td>
<td align="right">174</td>
<td align="right">62</td>
<td align="right">112</td>
<td align="right">95</td>
</tr>
<tr class="odd">
<td align="left">Celiac_GCST90014442_quantile</td>
<td align="right">335</td>
<td align="right">124</td>
<td align="right">211</td>
<td align="right">315</td>
</tr>
<tr class="even">
<td align="left">Celiac_GCST90468120_quantile</td>
<td align="right">208</td>
<td align="right">43</td>
<td align="right">165</td>
<td align="right">38</td>
</tr>
<tr class="odd">
<td align="left">Eosinophil.count_quantile</td>
<td align="right">498</td>
<td align="right">191</td>
<td align="right">307</td>
<td align="right">159</td>
</tr>
<tr class="even">
<td align="left">Eosinophil.percentage.of.white.cells_quantile</td>
<td align="right">426</td>
<td align="right">219</td>
<td align="right">207</td>
<td align="right">118</td>
</tr>
<tr class="odd">
<td align="left">Hematocrit_quantile</td>
<td align="right">235</td>
<td align="right">108</td>
<td align="right">127</td>
<td align="right">93</td>
</tr>
<tr class="even">
<td align="left">Hemoglobin.concentration_quantile</td>
<td align="right">156</td>
<td align="right">35</td>
<td align="right">121</td>
<td align="right">65</td>
</tr>
<tr class="odd">
<td align="left">High.light.scatter.reticulocyte.count_quantile</td>
<td align="right">218</td>
<td align="right">94</td>
<td align="right">124</td>
<td align="right">110</td>
</tr>
<tr class="even">
<td
align="left">High.light.scatter.reticulocyte.percentage.of.red.cells_quantile</td>
<td align="right">316</td>
<td align="right">149</td>
<td align="right">167</td>
<td align="right">138</td>
</tr>
<tr class="odd">
<td align="left">IBD_GCST90013901_quantile</td>
<td align="right">269</td>
<td align="right">200</td>
<td align="right">69</td>
<td align="right">174</td>
</tr>
<tr class="even">
<td align="left">IBD_GCST90013951_quantile</td>
<td align="right">337</td>
<td align="right">250</td>
<td align="right">87</td>
<td align="right">177</td>
</tr>
<tr class="odd">
<td align="left">Immature.fraction.of.reticulocytes_quantile</td>
<td align="right">456</td>
<td align="right">127</td>
<td align="right">329</td>
<td align="right">195</td>
</tr>
<tr class="even">
<td align="left">LUPUS_GCST003156_quantile</td>
<td align="right">202</td>
<td align="right">97</td>
<td align="right">105</td>
<td align="right">15</td>
</tr>
<tr class="odd">
<td align="left">LUPUS_GCST011096_quantile</td>
<td align="right">471</td>
<td align="right">47</td>
<td align="right">424</td>
<td align="right">136</td>
</tr>
<tr class="even">
<td align="left">Lymphocyte.count_quantile</td>
<td align="right">1603</td>
<td align="right">907</td>
<td align="right">696</td>
<td align="right">243</td>
</tr>
<tr class="odd">
<td align="left">Lymphocyte.percentage.of.white.cells_quantile</td>
<td align="right">511</td>
<td align="right">285</td>
<td align="right">226</td>
<td align="right">124</td>
</tr>
<tr class="even">
<td align="left">Mean.corpuscular.hemoglobin_quantile</td>
<td align="right">390</td>
<td align="right">25</td>
<td align="right">365</td>
<td align="right">107</td>
</tr>
<tr class="odd">
<td align="left">Mean.corpuscular.hemoglobin.concentration_quantile</td>
<td align="right">316</td>
<td align="right">91</td>
<td align="right">225</td>
<td align="right">294</td>
</tr>
<tr class="even">
<td align="left">Mean.corpuscular.volume_quantile</td>
<td align="right">425</td>
<td align="right">55</td>
<td align="right">370</td>
<td align="right">146</td>
</tr>
<tr class="odd">
<td align="left">Mean.platelet.volume_quantile</td>
<td align="right">205</td>
<td align="right">85</td>
<td align="right">120</td>
<td align="right">86</td>
</tr>
<tr class="even">
<td align="left">Mean.reticulocyte.volume_quantile</td>
<td align="right">403</td>
<td align="right">111</td>
<td align="right">292</td>
<td align="right">116</td>
</tr>
<tr class="odd">
<td align="left">Mean.sphered.corpuscular.volume_quantile</td>
<td align="right">366</td>
<td align="right">78</td>
<td align="right">288</td>
<td align="right">76</td>
</tr>
<tr class="even">
<td align="left">Monocyte.count_quantile</td>
<td align="right">1186</td>
<td align="right">844</td>
<td align="right">342</td>
<td align="right">135</td>
</tr>
<tr class="odd">
<td align="left">Monocyte.percentage.of.white.cells_quantile</td>
<td align="right">589</td>
<td align="right">50</td>
<td align="right">539</td>
<td align="right">89</td>
</tr>
<tr class="even">
<td align="left">Neutrophil.count_quantile</td>
<td align="right">808</td>
<td align="right">736</td>
<td align="right">72</td>
<td align="right">114</td>
</tr>
<tr class="odd">
<td align="left">Neutrophil.percentage.of.white.cells_quantile</td>
<td align="right">1467</td>
<td align="right">1335</td>
<td align="right">132</td>
<td align="right">394</td>
</tr>
<tr class="even">
<td align="left">Platelet.count_quantile</td>
<td align="right">235</td>
<td align="right">53</td>
<td align="right">182</td>
<td align="right">109</td>
</tr>
<tr class="odd">
<td align="left">Platelet.crit_quantile</td>
<td align="right">309</td>
<td align="right">93</td>
<td align="right">216</td>
<td align="right">101</td>
</tr>
<tr class="even">
<td align="left">Platelet.distribution.width_quantile</td>
<td align="right">450</td>
<td align="right">147</td>
<td align="right">303</td>
<td align="right">120</td>
</tr>
<tr class="odd">
<td align="left">Red.blood.cell.count_quantile</td>
<td align="right">264</td>
<td align="right">143</td>
<td align="right">121</td>
<td align="right">78</td>
</tr>
<tr class="even">
<td align="left">Red.cell.distribution.width_quantile</td>
<td align="right">405</td>
<td align="right">293</td>
<td align="right">112</td>
<td align="right">40</td>
</tr>
<tr class="odd">
<td align="left">Reticulocyte.count_quantile</td>
<td align="right">169</td>
<td align="right">74</td>
<td align="right">95</td>
<td align="right">81</td>
</tr>
<tr class="even">
<td align="left">Reticulocyte.fraction.of.red.cells_quantile</td>
<td align="right">516</td>
<td align="right">461</td>
<td align="right">55</td>
<td align="right">211</td>
</tr>
<tr class="odd">
<td align="left">T1D_GCST90000529_quantile</td>
<td align="right">1543</td>
<td align="right">1417</td>
<td align="right">126</td>
<td align="right">280</td>
</tr>
<tr class="even">
<td align="left">T1D_GCST90014023_quantile</td>
<td align="right">554</td>
<td align="right">482</td>
<td align="right">72</td>
<td align="right">89</td>
</tr>
<tr class="odd">
<td align="left">White.blood.cell.count_quantile</td>
<td align="right">1173</td>
<td align="right">1042</td>
<td align="right">131</td>
<td align="right">147</td>
</tr>
</tbody>
</table>
<p><strong>Interpretation:</strong></p>
<p>In the quantile analysis, the number of significant differential
expressed genes is generally higher compared to the continuous analysis,
which suggests that stratifying by PRS (top 25%) may identify more genes
with stronger associations to the traits. We observed that certain genes
were identified in both the continuous and quantile PRS analyses, though
not all genes overlapped. Specifically, for the Eosinophil count trait,
ACTN3, HAMP, SCGB3A1, CH25H, and NPIPA8 appeared among the top 10 DEGs
across both analyses, suggesting that these genes might play a key role
in the trait regardless of the analysis approach. For example:</p>
<ul>
<li><p>HAMP encodes Hepcidin, which is involved in the maintenance of
iron homeostasis.</p></li>
<li><p>SCGB3A1 predicts to enable cytokin activity.</p></li>
<li><p>CH25H encodes for cholesterol 25-hydroxylase, involved in lipid
metabolism and modulating immune responses.</p></li>
</ul>
<p>Additionally, the GO enrichment analysis provides insights into the
biological pathways and processes that are potentially involved in these
traits. BgRatio is the ratio of the total number of genes in a specific
pathway (GO term) to the total number of genes. GeneRatio is the ratio
of the number of genes enriched in specific pathway to the number of
genes that were tested in the analysis. FoldEnrichment is the ratio of
GeneRatio to BgRatio, showing how much more enriched a pathway is in
your genes of interest compared to the total gene set. Higher
FoldEnrichment values indicate that a particular GO term is
overrepresented in your gene set compared to the general gene pool,
suggesting it may be biologically relevant for the trait. Some traits
show higher pathway enrichment in the quantile analysis, indicating that
stratification by PRS may provide more biologically meaningful
results.</p>
<p>For highly correlated traits, they tend to share similar GO enriched
pathways. For example, mean corpuscular hemoglobin and mean corpucular
volume are both involved in read blood cell biology, so it’s expected
that their GO enrichment would point to similar biological processes,
such as hemoglobin metabolism, iron homeostasis and erythropoiesis.</p>
<p>To account for population stratification, we adjusted the model by
including the first two genotype principal components (PCs), as they
explain most of the genetic variation in the data. For Eosinophil count,
we observed that genes were consistently identified as top
differentially expressed with and without adjustment. The only
difference is that instead of SLC6A9, WFDC2 was detected as top 10 DEGs
when adjusting for the first two genotype PCs. WFBC2 is a protease
inhibitor with a potential role in innate immune defense (<a
href="https://doi.org/10.1186/1465-9921-7-61">L Bingle, 2006</a>).
Although same genes are identified as top DEGs for continuous analysis,
additional pathways such as response to inactivity were highlighted when
adjusting for the first two genotype PCs.</p>
</div>
<div id="basophil-count" class="section level2">
<h2>Basophil count</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Basophil.count.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Basophil.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Basophil.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Basophil.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Basophil.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Basophil.count.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Basophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Basophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Basophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Basophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Basophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Basophil.count_M2.png" /></p>
</div>
<div id="basophil.percentage.of.white.cells" class="section level2">
<h2>Basophil.percentage.of.white.cells</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Basophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Basophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Basophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Basophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Basophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Basophil.percentage.of.white.cells.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Basophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Basophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Basophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Basophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Basophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Basophil.percentage.of.white.cells_M2.png" /></p>
</div>
<div id="eosinophil.count" class="section level2">
<h2>Eosinophil.count</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Eosinophil.count.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Eosinophil.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Eosinophil.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Eosinophil.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Eosinophil.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Eosinophil.count.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Eosinophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Eosinophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Eosinophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Eosinophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Eosinophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Eosinophil.count_M2.png" /></p>
</div>
<div id="eosinophil.percentage.of.white.cells" class="section level2">
<h2>Eosinophil.percentage.of.white.cells</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Eosinophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Eosinophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Eosinophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Eosinophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Eosinophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Eosinophil.percentage.of.white.cells.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Eosinophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Eosinophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Eosinophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Eosinophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Eosinophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Eosinophil.percentage.of.white.cells_M2.png" /></p>
</div>
<div id="hematocrit" class="section level2">
<h2>Hematocrit</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Hematocrit.png" /> <img
src="figure/differential_expression.Rmd/ma_plot_quantile_Hematocrit.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Hematocrit.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Hematocrit.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Hematocrit.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Hematocrit.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Hematocrit_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Hematocrit_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Hematocrit_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Hematocrit_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Hematocrit_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Hematocrit_M2.png" /></p>
</div>
<div id="hemoglobin.concentration" class="section level2">
<h2>Hemoglobin.concentration</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Hemoglobin.concentration.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Hemoglobin.concentration.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Hemoglobin.concentration.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Hemoglobin.concentration.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Hemoglobin.concentration.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Hemoglobin.concentration.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Hemoglobin.concentration_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Hemoglobin.concentration_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Hemoglobin.concentration_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Hemoglobin.concentration_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Hemoglobin.concentration_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Hemoglobin.concentration.png" /></p>
</div>
<div id="high.light.scatter.reticulocyte.count" class="section level2">
<h2>High.light.scatter.reticulocyte.count</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_High.light.scatter.reticulocyte.count.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_High.light.scatter.reticulocyte.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_High.light.scatter.reticulocyte.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_High.light.scatter.reticulocyte.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_High.light.scatter.reticulocyte.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_High.light.scatter.reticulocyte.count.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_High.light.scatter.reticulocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_High.light.scatter.reticulocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_High.light.scatter.reticulocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_High.light.scatter.reticulocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_High.light.scatter.reticulocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_High.light.scatter.reticulocyte.count_M2.png" /></p>
</div>
<div id="high.light.scatter.reticulocyte.percentage.of.red.cells"
class="section level2">
<h2>High.light.scatter.reticulocyte.percentage.of.red.cells</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_High.light.scatter.reticulocyte.percentage.of.red.cells.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_High.light.scatter.reticulocyte.percentage.of.red.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_High.light.scatter.reticulocyte.percentage.of.red.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_High.light.scatter.reticulocyte.percentage.of.red.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_High.light.scatter.reticulocyte.percentage.of.red.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_High.light.scatter.reticulocyte.percentage.of.red.cells.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_High.light.scatter.reticulocyte.percentage.of.red.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_High.light.scatter.reticulocyte.percentage.of.red.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_High.light.scatter.reticulocyte.percentage.of.red.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_High.light.scatter.reticulocyte.percentage.of.red.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_High.light.scatter.reticulocyte.percentage.of.red.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_High.light.scatter.reticulocyte.percentage.of.red.cells_M2.png" /></p>
</div>
<div id="immature.fraction.of.reticulocytes" class="section level2">
<h2>Immature.fraction.of.reticulocytes</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Immature.fraction.of.reticulocytes.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Immature.fraction.of.reticulocytes.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Immature.fraction.of.reticulocytes.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Immature.fraction.of.reticulocytes.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Immature.fraction.of.reticulocytes.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Immature.fraction.of.reticulocytes.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Immature.fraction.of.reticulocytes_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Immature.fraction.of.reticulocytes_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Immature.fraction.of.reticulocytes_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Immature.fraction.of.reticulocytes_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Immature.fraction.of.reticulocytes_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Immature.fraction.of.reticulocytes_M2.png" /></p>
</div>
<div id="lymphocyte.count" class="section level2">
<h2>Lymphocyte.count</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Lymphocyte.count.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Lymphocyte.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Lymphocyte.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Lymphocyte.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Lymphocyte.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Lymphocyte.count.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Lymphocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Lymphocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Lymphocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Lymphocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Lymphocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Lymphocyte.count_M2.png" /></p>
</div>
<div id="lymphocyte.percentage.of.white.cells" class="section level2">
<h2>Lymphocyte.percentage.of.white.cells</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Lymphocyte.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Lymphocyte.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Lymphocyte.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Lymphocyte.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Lymphocyte.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Lymphocyte.percentage.of.white.cells.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Lymphocyte.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Lymphocyte.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Lymphocyte.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Lymphocyte.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Lymphocyte.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Lymphocyte.percentage.of.white.cells_M2.png" /></p>
</div>
<div id="mean.corpuscular.hemoglobin.concentration"
class="section level2">
<h2>Mean.corpuscular.hemoglobin.concentration</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Mean.corpuscular.hemoglobin.concentration.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Mean.corpuscular.hemoglobin.concentration.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Mean.corpuscular.hemoglobin.concentration.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Mean.corpuscular.hemoglobin.concentration.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Mean.corpuscular.hemoglobin.concentration.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.corpuscular.hemoglobin.concentration.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Mean.corpuscular.hemoglobin.concentration_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Mean.corpuscular.hemoglobin.concentration_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Mean.corpuscular.hemoglobin.concentration_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Mean.corpuscular.hemoglobin.concentration_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Mean.corpuscular.hemoglobin.concentration_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.corpuscular.hemoglobin.concentration_M2.png" /></p>
</div>
<div id="mean.corpuscular.hemoglobin" class="section level2">
<h2>Mean.corpuscular.hemoglobin</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Mean.corpuscular.hemoglobin.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Mean.corpuscular.hemoglobin.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Mean.corpuscular.hemoglobin.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Mean.corpuscular.hemoglobin.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Mean.corpuscular.hemoglobin.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.corpuscular.hemoglobin.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Mean.corpuscular.hemoglobin_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Mean.corpuscular.hemoglobin_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Mean.corpuscular.hemoglobin_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Mean.corpuscular.hemoglobin_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Mean.corpuscular.hemoglobin_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.corpuscular.hemoglobin_M2.png" /></p>
</div>
<div id="mean.corpuscular.volume" class="section level2">
<h2>Mean.corpuscular.volume</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Mean.corpuscular.volume.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Mean.corpuscular.volume.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Mean.corpuscular.volume.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Mean.corpuscular.volume.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Mean.corpuscular.volume.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.corpuscular.volume.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Mean.corpuscular.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Mean.corpuscular.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Mean.corpuscular.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Mean.corpuscular.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Mean.corpuscular.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.corpuscular.volume_M2.png" /></p>
</div>
<div id="mean.platelet.volume" class="section level2">
<h2>Mean.platelet.volume</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Mean.platelet.volume.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Mean.platelet.volume.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Mean.platelet.volume.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Mean.platelet.volume.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Mean.platelet.volume.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.platelet.volume.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Mean.platelet.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Mean.platelet.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Mean.platelet.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Mean.platelet.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Mean.platelet.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.platelet.volume_M2.png" /></p>
</div>
<div id="mean.reticulocyte.volume" class="section level2">
<h2>Mean.reticulocyte.volume</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Mean.reticulocyte.volume.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Mean.reticulocyte.volume.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Mean.reticulocyte.volume.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Mean.reticulocyte.volume.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Mean.reticulocyte.volume.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.reticulocyte.volume.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Mean.reticulocyte.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Mean.reticulocyte.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Mean.reticulocyte.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Mean.reticulocyte.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Mean.reticulocyte.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.reticulocyte.volume_M2.png" /></p>
</div>
<div id="mean.sphered.corpuscular.volume" class="section level2">
<h2>Mean.sphered.corpuscular.volume</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Mean.sphered.corpuscular.volume.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Mean.sphered.corpuscular.volume.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Mean.sphered.corpuscular.volume.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Mean.sphered.corpuscular.volume.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Mean.sphered.corpuscular.volume.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.sphered.corpuscular.volume.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Mean.sphered.corpuscular.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Mean.sphered.corpuscular.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Mean.sphered.corpuscular.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Mean.sphered.corpuscular.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Mean.sphered.corpuscular.volume_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.sphered.corpuscular.volume_M2.png" /></p>
</div>
<div id="monocyte.count" class="section level2">
<h2>Monocyte.count</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Monocyte.count.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Monocyte.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Monocyte.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Monocyte.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Monocyte.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Monocyte.count.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Monocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Monocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Monocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Monocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Monocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Monocyte.count_M2.png" /></p>
</div>
<div id="monocyte.percentage.of.white.cells" class="section level2">
<h2>Monocyte.percentage.of.white.cells</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Monocyte.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Monocyte.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Monocyte.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Monocyte.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Monocyte.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Monocyte.percentage.of.white.cells.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Monocyte.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Monocyte.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Monocyte.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Monocyte.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Monocyte.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Monocyte.percentage.of.white.cells_M2.png" /></p>
</div>
<div id="neutrophil.count" class="section level2">
<h2>Neutrophil.count</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Neutrophil.count.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Neutrophil.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Neutrophil.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Neutrophil.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Neutrophil.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Neutrophil.count.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Neutrophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Neutrophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Neutrophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Neutrophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Neutrophil.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Neutrophil.count_M2.png" /></p>
</div>
<div id="neutrophil.percentage.of.white.cells" class="section level2">
<h2>Neutrophil.percentage.of.white.cells</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Neutrophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Neutrophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Neutrophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Neutrophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Neutrophil.percentage.of.white.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Neutrophil.percentage.of.white.cells.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Neutrophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Neutrophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Neutrophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Neutrophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Neutrophil.percentage.of.white.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Neutrophil.percentage.of.white.cells_M2.png" /></p>
</div>
<div id="platelet.count" class="section level2">
<h2>Platelet.count</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Platelet.count.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Platelet.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Platelet.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Platelet.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Platelet.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Platelet.count.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Platelet.count_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Platelet.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Platelet.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Platelet.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Platelet.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Platelet.count_M2.png" /></p>
</div>
<div id="platelet.crit" class="section level2">
<h2>Platelet.crit</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Platelet.crit.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Platelet.crit.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Platelet.crit.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Platelet.crit.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Platelet.crit.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Platelet.crit.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Platelet.crit_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Platelet.crit_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Platelet.crit_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Platelet.crit_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Platelet.crit_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Platelet.crit_M2.png" /></p>
</div>
<div id="platelet.distribution.width" class="section level2">
<h2>Platelet.distribution.width</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Platelet.distribution.width.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Platelet.distribution.width.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Platelet.distribution.width.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Platelet.distribution.width.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Platelet.distribution.width.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Platelet.distribution.width.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Platelet.distribution.width_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Platelet.distribution.width_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Platelet.distribution.width_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Platelet.distribution.width_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Platelet.distribution.width_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Platelet.distribution.width_M2.png" /></p>
</div>
<div id="red.blood.cell.count" class="section level2">
<h2>Red.blood.cell.count</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Red.blood.cell.count.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Red.blood.cell.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Red.blood.cell.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Red.blood.cell.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Red.blood.cell.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Red.blood.cell.count.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Red.blood.cell.count_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Red.blood.cell.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Red.blood.cell.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Red.blood.cell.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Red.blood.cell.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Red.blood.cell.count_M2.png" /></p>
</div>
<div id="red.cell.distribution.width" class="section level2">
<h2>Red.cell.distribution.width</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Red.cell.distribution.width.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Red.cell.distribution.width.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Red.cell.distribution.width.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Red.cell.distribution.width.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Red.cell.distribution.width.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Red.cell.distribution.width.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Red.cell.distribution.width_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Red.cell.distribution.width_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Red.cell.distribution.width_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Red.cell.distribution.width_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Red.cell.distribution.width_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Red.cell.distribution.width_M2.png" /></p>
</div>
<div id="reticulocyte.count" class="section level2">
<h2>Reticulocyte.count</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Reticulocyte.count.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Reticulocyte.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Reticulocyte.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Reticulocyte.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Reticulocyte.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Reticulocyte.count.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Reticulocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Reticulocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Reticulocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Reticulocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Reticulocyte.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Reticulocyte.count_M2.png" /></p>
</div>
<div id="reticulocyte.fraction.of.red.cells" class="section level2">
<h2>Reticulocyte.fraction.of.red.cells</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Reticulocyte.fraction.of.red.cells.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Reticulocyte.fraction.of.red.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Reticulocyte.fraction.of.red.cells.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Reticulocyte.fraction.of.red.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Reticulocyte.fraction.of.red.cells.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Reticulocyte.fraction.of.red.cells.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Reticulocyte.fraction.of.red.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Reticulocyte.fraction.of.red.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Reticulocyte.fraction.of.red.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Reticulocyte.fraction.of.red.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Reticulocyte.fraction.of.red.cells_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Reticulocyte.fraction.of.red.cells_M2.png" /></p>
</div>
<div id="white.blood.cell.count" class="section level2">
<h2>White.blood.cell.count</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_White.blood.cell.count.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_White.blood.cell.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_White.blood.cell.count.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_White.blood.cell.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_White.blood.cell.count.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_White.blood.cell.count.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_White.blood.cell.count_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_White.blood.cell.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_White.blood.cell.count_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_White.blood.cell.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_White.blood.cell.count_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_White.blood.cell.count_M2.png" /></p>
</div>
<div id="celiac_gcst90014442" class="section level2">
<h2>Celiac_GCST90014442</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Celiac_GCST90014442.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Celiac_GCST90014442.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Celiac_GCST90014442.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Celiac_GCST90014442.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Celiac_GCST90014442.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Celiac_GCST90014442.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Celiac_GCST90014442_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Celiac_GCST90014442_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Celiac_GCST90014442_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Celiac_GCST90014442_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Celiac_GCST90014442_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Celiac_GCST90014442_M2.png" /></p>
</div>
<div id="celiac_gcst90468120" class="section level2">
<h2>Celiac_GCST90468120</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Celiac_GCST90468120.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Celiac_GCST90468120.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Celiac_GCST90468120.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Celiac_GCST90468120.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Celiac_GCST90468120.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Celiac_GCST90468120.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_Celiac_GCST90468120_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_Celiac_GCST90468120_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_Celiac_GCST90468120_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_Celiac_GCST90468120_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_Celiac_GCST90468120_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_Celiac_GCST90468120_M2.png" /></p>
</div>
<div id="ibd_gcst90013901" class="section level2">
<h2>IBD_GCST90013901</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_IBD_GCST90013901.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_IBD_GCST90013901.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_IBD_GCST90013901.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_IBD_GCST90013901.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_IBD_GCST90013901.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_IBD_GCST90013901.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_IBD_GCST90013901_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_IBD_GCST90013901_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_IBD_GCST90013901_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_IBD_GCST90013901_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_IBD_GCST90013901_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_IBD_GCST90013901_M2.png" /></p>
</div>
<div id="ibd_gcst90013951" class="section level2">
<h2>IBD_GCST90013951</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_IBD_GCST90013951.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_IBD_GCST90013951.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_IBD_GCST90013951.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_IBD_GCST90013951.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_IBD_GCST90013951.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_IBD_GCST90013951.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_IBD_GCST90013951_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_IBD_GCST90013951_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_IBD_GCST90013951_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_IBD_GCST90013951_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_IBD_GCST90013951_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_IBD_GCST90013951_M2.png" /></p>
</div>
<div id="lupus_gcst003156" class="section level2">
<h2>LUPUS_GCST003156</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_LUPUS_GCST003156.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_LUPUS_GCST003156.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_LUPUS_GCST003156.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_LUPUS_GCST003156.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_LUPUS_GCST003156.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_LUPUS_GCST003156.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_LUPUS_GCST003156_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_LUPUS_GCST003156_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_LUPUS_GCST003156_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_LUPUS_GCST003156_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_LUPUS_GCST003156_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_LUPUS_GCST003156_M2.png" /></p>
</div>
<div id="lupus_gcst011096" class="section level2">
<h2>LUPUS_GCST011096</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_LUPUS_GCST011096.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_LUPUS_GCST011096.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_LUPUS_GCST011096.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_LUPUS_GCST011096.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_LUPUS_GCST011096.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_LUPUS_GCST011096.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_LUPUS_GCST011096_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_LUPUS_GCST011096_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_LUPUS_GCST011096_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_LUPUS_GCST011096_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_LUPUS_GCST011096_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_LUPUS_GCST011096_M2.png" /></p>
</div>
<div id="t1d_gcst90000529" class="section level2">
<h2>T1D_GCST90000529</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_T1D_GCST90000529.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_T1D_GCST90000529.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_T1D_GCST90000529.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_T1D_GCST90000529.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_T1D_GCST90000529.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_T1D_GCST90000529.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_T1D_GCST90000529_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_T1D_GCST90000529_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_T1D_GCST90000529_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_T1D_GCST90000529_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_T1D_GCST90000529_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_T1D_GCST90000529_M2.png" /></p>
</div>
<div id="t1d_gcst90014023" class="section level2">
<h2>T1D_GCST90014023</h2>
<p><img
src="figure/differential_expression.Rmd/ma_plot_T1D_GCST90014023.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_T1D_GCST90014023.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_T1D_GCST90014023.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_T1D_GCST90014023.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_T1D_GCST90014023.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_T1D_GCST90014023.png" /></p>
<p><img
src="figure/differential_expression.Rmd/ma_plot_T1D_GCST90014023_M2.png" />
<img
src="figure/differential_expression.Rmd/ma_plot_quantile_T1D_GCST90014023_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_T1D_GCST90014023_M2.png" />
<img
src="figure/differential_expression.Rmd/volcano_plot_quantile_T1D_GCST90014023_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_T1D_GCST90014023_M2.png" />
<img
src="figure/differential_expression.Rmd/enrichment_plot_quantile_T1D_GCST90014023_M2.png" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] clusterProfiler_4.6.2       enrichplot_1.18.4          
 [3] org.Hs.eg.db_3.16.0         AnnotationDbi_1.60.2       
 [5] genekitr_1.2.8              ggrepel_0.9.6              
 [7] BiocParallel_1.32.6         DESeq2_1.38.3              
 [9] SummarizedExperiment_1.28.0 Biobase_2.58.0             
[11] MatrixGenerics_1.10.0       matrixStats_1.2.0          
[13] GenomicRanges_1.50.2        GenomeInfoDb_1.34.9        
[15] IRanges_2.32.0              S4Vectors_0.36.2           
[17] BiocGenerics_0.44.0         corrplot_0.95              
[19] ggplot2_3.5.1               dplyr_1.1.4                
[21] data.table_1.16.4           workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] shadowtext_0.1.4       fastmatch_1.1-6        plyr_1.8.9            
  [4] igraph_1.5.1           lazyeval_0.2.2         splines_4.2.2         
  [7] usethis_3.1.0          urltools_1.7.3         digest_0.6.37         
 [10] yulab.utils_0.2.0      htmltools_0.5.8.1      GOSemSim_2.24.0       
 [13] viridis_0.6.5          GO.db_3.16.0           magrittr_2.0.3        
 [16] memoise_2.0.1          remotes_2.5.0          openxlsx_4.2.5.2      
 [19] Biostrings_2.66.0      annotate_1.76.0        graphlayouts_1.0.1    
 [22] prettyunits_1.2.0      colorspace_2.1-1       blob_1.2.4            
 [25] xfun_0.50              callr_3.7.6            crayon_1.5.3          
 [28] RCurl_1.98-1.16        jsonlite_1.8.9         scatterpie_0.2.4      
 [31] ape_5.7-1              glue_1.8.0             polyclip_1.10-7       
 [34] gtable_0.3.6           zlibbioc_1.44.0        XVector_0.38.0        
 [37] DelayedArray_0.24.0    pkgbuild_1.4.6         scales_1.3.0          
 [40] DOSE_3.24.2            DBI_1.2.3              miniUI_0.1.1.1        
 [43] Rcpp_1.0.14            viridisLite_0.4.2      xtable_1.8-4          
 [46] progress_1.2.3         gridGraphics_0.5-1     tidytree_0.4.6        
 [49] bit_4.5.0.1            europepmc_0.4.3        profvis_0.4.0         
 [52] htmlwidgets_1.6.4      httr_1.4.7             fgsea_1.24.0          
 [55] RColorBrewer_1.1-3     ellipsis_0.3.2         urlchecker_1.0.1      
 [58] pkgconfig_2.0.3        XML_3.99-0.18          farver_2.1.2          
 [61] sass_0.4.9             locfit_1.5-9.8         labeling_0.4.3        
 [64] ggplotify_0.1.2        tidyselect_1.2.1       rlang_1.1.5           
 [67] reshape2_1.4.4         later_1.4.1            munsell_0.5.1         
 [70] tools_4.2.2            cachem_1.1.0           downloader_0.4        
 [73] cli_3.6.3              generics_0.1.3         RSQLite_2.3.9         
 [76] gson_0.1.0             devtools_2.4.5         evaluate_1.0.3        
 [79] stringr_1.5.1          fastmap_1.2.0          yaml_2.3.10           
 [82] ggtree_3.6.2           processx_3.8.5         knitr_1.49            
 [85] bit64_4.6.0-1          fs_1.6.5               tidygraph_1.3.0       
 [88] zip_2.3.2              purrr_1.0.2            KEGGREST_1.38.0       
 [91] ggraph_2.1.0           nlme_3.1-160           mime_0.12             
 [94] whisker_0.4.1          aplot_0.2.4            ggvenn_0.1.10         
 [97] xml2_1.3.6             compiler_4.2.2         rstudioapi_0.17.1     
[100] png_0.1-8              treeio_1.22.0          tibble_3.2.1          
[103] tweenr_2.0.3           geneplotter_1.76.0     bslib_0.9.0           
[106] stringi_1.8.4          ps_1.8.1               lattice_0.22-6        
[109] Matrix_1.5-1           vctrs_0.6.5            pillar_1.10.1         
[112] lifecycle_1.0.4        triebeard_0.4.1        jquerylib_0.1.4       
[115] cowplot_1.1.3          bitops_1.0-9           httpuv_1.6.15         
[118] patchwork_1.3.0        qvalue_2.30.0          R6_2.5.1              
[121] promises_1.3.2         gridExtra_2.3          sessioninfo_1.2.2     
[124] codetools_0.2-20       pkgload_1.4.0          MASS_7.3-58.1         
[127] rprojroot_2.0.4        withr_3.0.2            GenomeInfoDbData_1.2.9
[130] parallel_4.2.2         hms_1.1.3              grid_4.2.2            
[133] ggfun_0.1.8            tidyr_1.3.1            HDO.db_0.99.1         
[136] rmarkdown_2.29         git2r_0.33.0           getPass_0.2-4         
[139] ggforce_0.4.1          shiny_1.10.0           geneset_0.2.7         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
