<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>expression_normalization</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">expression_normalization</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-17
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>myproject/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250327code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250327)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250327code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250327)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong3c6673f">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 3c6673f
</a>
</p>
</div>
<div id="strongRepositoryversionstrong3c6673f"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 3c6673f.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  analysis/normalized_counts.txt
    Untracked:  analysis/prs_blood_cell.txt
    Untracked:  analysis/prs_immune.txt
    Untracked:  data/GTEx_v8.bed
    Untracked:  data/GTEx_v8.bim
    Untracked:  data/GTEx_v8.bk
    Untracked:  data/GTEx_v8.fam
    Untracked:  data/GTEx_v8.rds
    Untracked:  data/Whole_Blood.v8.covariates.txt
    Untracked:  data/blood_cell/
    Untracked:  data/gene_reads_2017-06-05_v8_whole_blood.gct
    Untracked:  data/gene_tpm_2017-06-05_v8_whole_blood.gct.gz
    Untracked:  data/immune/

Unstaged changes:
    Modified:   myproject.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/expression_normalization.Rmd</code>) and HTML
(<code>docs/expression_normalization.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
3c6673f
</td>
<td>
ElisaChen
</td>
<td>
2025-04-17
</td>
<td>
wflow_publish("analysis/expression_normalization.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
a14b960
</td>
<td>
ElisaChen
</td>
<td>
2025-04-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
3d07189
</td>
<td>
ElisaChen
</td>
<td>
2025-04-17
</td>
<td>
wflow_publish(c("analysis/about.Rmd",
"analysis/expression_normalization.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="raw-count" class="section level1">
<h1>Raw count</h1>
<pre class="r"><code># Load the gene expression data
gene_expr_file &lt;- &quot;data/gene_reads_2017-06-05_v8_whole_blood.gct&quot;
raw_count_df &lt;- fread(gene_expr_file, header = TRUE)

gene_name &lt;- raw_count_df$Description
raw_count_df &lt;- raw_count_df[, -c(1:3)]

# modify GTEx sample names matching names used in PRS data
colnames(raw_count_df) &lt;- sub(&quot;^(GTEX-[^-.]+).*&quot;, &quot;\\1&quot;, colnames(raw_count_df))</code></pre>
</div>
<div id="match-samples-between-prs-and-count-matrix"
class="section level1">
<h1>Match samples between PRS and count matrix</h1>
<pre class="r"><code># load PRS data for blood trait
prs_file &lt;- &quot;analysis/prs_blood_cell.txt&quot;
prs_blood &lt;- fread(prs_file, header = T, stringsAsFactors = FALSE)

# obtain GTEx sample id
id &lt;- prs_blood$sample_id

# Load the PRS data for immune trait
prs_file &lt;- &quot;analysis/prs_immune.txt&quot;
prs_immune &lt;- fread(prs_file, header = T, stringsAsFactors = FALSE)

# subset samples
matching_samples &lt;- intersect(id, colnames(raw_count_df))
prs_blood &lt;- prs_blood[prs_blood$sample_id %in% matching_samples, ]
prs_immune &lt;- prs_immune[prs_immune$sample_id %in% matching_samples, ]
final_count &lt;- raw_count_df[,  ..matching_samples]
dim(final_count) </code></pre>
<pre><code>[1] 56200   670</code></pre>
</div>
<div id="stratify-by-prs-level-high-vs-median-vs-low"
class="section level1">
<h1>Stratify by PRS level (High vs Median vs Low)</h1>
<div id="blood-trait" class="section level2">
<h2>Blood trait</h2>
<pre class="r"><code># Loop through each column (starting from the second column, as the first column is sample_id)
for (i in 2:ncol(prs_blood)) {
  # Extract the PRS values for the current trait
  prs_trait &lt;- prs_blood[[i]]

  # Calculate the 25th and 75th percentiles for the trait
  p25 &lt;- quantile(prs_trait, 0.25, na.rm = TRUE)
  p75 &lt;- quantile(prs_trait, 0.75, na.rm = TRUE)

  # Create a new column for the group classification based on the percentiles
  group &lt;- ifelse(prs_trait &gt; p75, &quot;High&quot;, 
                  ifelse(prs_trait &lt; p25, &quot;Low&quot;, &quot;Median&quot;))

  # Assign the group to the new column for this trait
  prs_blood[[paste0(&quot;group_&quot;, colnames(prs_blood)[i])]] &lt;- as.factor(group)
}</code></pre>
</div>
<div id="immune-trait" class="section level2">
<h2>Immune trait</h2>
<pre class="r"><code># Loop through each column (starting from the second column, as the first column is sample_id)
for (i in 2:ncol(prs_immune)) {
  # Extract the PRS values for the current trait
  prs_trait &lt;- prs_immune[[i]]

  # Calculate the 25th and 75th percentiles for the trait
  p25 &lt;- quantile(prs_trait, 0.25, na.rm = TRUE)
  p75 &lt;- quantile(prs_trait, 0.75, na.rm = TRUE)

  # Create a new column for the group classification based on the percentiles
  group &lt;- ifelse(prs_trait &gt; p75, &quot;High&quot;, 
                  ifelse(prs_trait &lt; p25, &quot;Low&quot;, &quot;Median&quot;))

  # Assign the group to the new column for this trait
  prs_immune[[paste0(&quot;group_&quot;, colnames(prs_immune)[i])]] &lt;- as.factor(group)
}</code></pre>
</div>
</div>
<div id="deseq2-normalization-by-median-of-ratios"
class="section level1">
<h1>DESeq2 normalization by median of ratios</h1>
<p>The size factor is calculated as follows:</p>
<ol style="list-style-type: decimal">
<li><p>For each gene, the geometric mean of counts across all samples is
computed (this serves as the pseudo baseline expression).</p></li>
<li><p>For each gene, the ratio of its count in a specific sample to the
pseudo-baseline expression is calculated (e.g., Sample A/pseudo
baseline, Sample B/pseudo baseline).</p></li>
<li><p>For each sample, the median of these ratios is computed, which
results in the size factor for that sample.</p></li>
</ol>
<p>Thus, DESeq2 normalizes for both sequencing depth and RNA composition
differences, and this process is independent of the design matrix,
meaning it is unaffected by the specific traits (whether blood or immune
traits) used in the analysis.</p>
<pre class="r"><code># Create the DESeq2 object (DESeqDataSet) from the raw count matrix and PRS
dds &lt;- DESeqDataSetFromMatrix(countData = as.matrix(final_count), 
                              colData = prs_blood, 
                              design = ~ group_PGS000163)
rownames(dds) &lt;- gene_name

# remove genes with no reads
keep_genes &lt;- rowSums(counts(dds)) &gt; 0
dds &lt;- dds[keep_genes, ]
dds</code></pre>
<pre><code>class: DESeqDataSet 
dim: 53136 670 
metadata(1): version
assays(1): counts
rownames(53136): DDX11L1 WASH7P ... MT-TT MT-TP
rowData names(0):
colnames(670): GTEX-111YS GTEX-1122O ... GTEX-ZXES GTEX-ZXG5
colData names(59): sample_id PGS000163 ... group_PGS000190
  group_PGS000191</code></pre>
<pre class="r"><code># Estimate size factors for normalization
dds &lt;- estimateSizeFactors(dds)
outliers &lt;- which(sizeFactors(dds) &gt; 4)
plot(sizeFactors(dds), colSums(counts(dds)), ylab = &quot;library sizes&quot;, 
     xlab = &quot;size factors&quot;, cex = .5)
text(sizeFactors(dds)[outliers], colSums(counts(dds))[outliers], 
     labels = colnames(dds)[outliers], pos = 2, cex = .5, col = &quot;red&quot;)</code></pre>
<p><img src="figure/expression_normalization.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
a14b960
</td>
<td>
ElisaChen
</td>
<td>
2025-04-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># obtain normalized counts
normalized_counts &lt;- counts(dds, normalized=TRUE)
#write.table(normalized_counts, &quot;analysis/normalized_counts.txt&quot;, 
#            sep=&quot;\t&quot;, row.names = T, col.names = T)</code></pre>
</div>
<div id="quantile-normalization" class="section level1">
<h1>Quantile normalization</h1>
<pre class="r"><code># load TPM counts data
tpm_file &lt;- &quot;data/gene_tpm_2017-06-05_v8_whole_blood.gct.gz&quot;
tpm_count_df &lt;- fread(tpm_file, header = TRUE, stringsAsFactors = FALSE)
colnames(tpm_count_df) &lt;- sub(&quot;^(GTEX-[^-.]+).*&quot;, &quot;\\1&quot;, colnames(tpm_count_df))

# subset samples &amp; remove no read genes
tpm_subset_df &lt;- tpm_count_df[, matching_samples, with = FALSE]
tpm_subset_df &lt;- tpm_subset_df[rowSums(tpm_subset_df[, -1, with = FALSE]) &gt; 0, ]

# convert into matrix
tpm_count &lt;- as.matrix(tpm_subset_df)

# perform quantile normalization
qn_counts &lt;- preprocessCore::normalize.quantiles(tpm_count)
colnames(qn_counts) &lt;- matching_samples</code></pre>
</div>
<div
id="compare-raw-counts-deseq2-normalized-count-quantile-normalized-count"
class="section level1">
<h1>Compare raw counts, DESeq2 normalized count &amp; quantile
normalized count</h1>
<pre class="r"><code>boxplot(log2(counts(dds) + 1)[, c(1:10, outliers)], 
        main = &quot;Raw read counts&quot;, ylab = &quot;log2(read counts)&quot;, cex = .6, 
        col = c(rep(&quot;lightblue&quot;, 10), &quot;red&quot;), las = 2) </code></pre>
<p><img src="figure/expression_normalization.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
a14b960
</td>
<td>
ElisaChen
</td>
<td>
2025-04-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>boxplot(log2(counts(dds, normalize = TRUE) + 1)[, c(1:10, outliers)], 
        main = &quot;DESeq2 normalized read counts&quot;, ylab = &quot;log2(read counts)&quot;, 
        cex = .6, col = c(rep(&quot;lightblue&quot;, 10), &quot;red&quot;), las = 2)</code></pre>
<p><img src="figure/expression_normalization.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
a14b960
</td>
<td>
ElisaChen
</td>
<td>
2025-04-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>boxplot(log2(qn_counts + 1)[, c(1:10, outliers)], 
        main = &quot;Quantile normalized read counts&quot;, ylab = &quot;log2(read counts)&quot;, 
        cex = .6, col = c(rep(&quot;lightblue&quot;, 10), &quot;red&quot;), las = 2) </code></pre>
<p><img src="figure/expression_normalization.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-3">
Past versions of unnamed-chunk-7-3.png
</button>
</p>
<div id="fig-unnamed-chunk-7-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
a14b960
</td>
<td>
ElisaChen
</td>
<td>
2025-04-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggplot2_3.5.1               preprocessCore_1.60.2      
 [3] DESeq2_1.38.3               SummarizedExperiment_1.28.0
 [5] Biobase_2.58.0              MatrixGenerics_1.10.0      
 [7] matrixStats_1.2.0           GenomicRanges_1.50.2       
 [9] GenomeInfoDb_1.34.9         IRanges_2.32.0             
[11] S4Vectors_0.36.2            BiocGenerics_0.44.0        
[13] dplyr_1.1.4                 data.table_1.16.4          
[15] workflowr_1.7.1            

loaded via a namespace (and not attached):
 [1] httr_1.4.7             sass_0.4.9             bit64_4.6.0-1         
 [4] jsonlite_1.8.9         R.utils_2.12.3         bslib_0.9.0           
 [7] getPass_0.2-4          blob_1.2.4             GenomeInfoDbData_1.2.9
[10] yaml_2.3.10            pillar_1.10.1          RSQLite_2.3.9         
[13] lattice_0.22-6         glue_1.8.0             digest_0.6.37         
[16] RColorBrewer_1.1-3     promises_1.3.2         XVector_0.38.0        
[19] colorspace_2.1-1       R.oo_1.27.0            htmltools_0.5.8.1     
[22] httpuv_1.6.15          Matrix_1.5-1           XML_3.99-0.18         
[25] pkgconfig_2.0.3        zlibbioc_1.44.0        xtable_1.8-4          
[28] scales_1.3.0           processx_3.8.5         whisker_0.4.1         
[31] later_1.4.1            BiocParallel_1.32.6    git2r_0.33.0          
[34] tibble_3.2.1           annotate_1.76.0        KEGGREST_1.38.0       
[37] generics_0.1.3         withr_3.0.2            cachem_1.1.0          
[40] cli_3.6.3              magrittr_2.0.3         crayon_1.5.3          
[43] memoise_2.0.1          evaluate_1.0.3         ps_1.8.1              
[46] R.methodsS3_1.8.2      fs_1.6.5               tools_4.2.2           
[49] lifecycle_1.0.4        stringr_1.5.1          locfit_1.5-9.8        
[52] munsell_0.5.1          DelayedArray_0.24.0    AnnotationDbi_1.60.2  
[55] callr_3.7.6            Biostrings_2.66.0      compiler_4.2.2        
[58] jquerylib_0.1.4        rlang_1.1.5            grid_4.2.2            
[61] RCurl_1.98-1.16        rstudioapi_0.17.1      bitops_1.0-9          
[64] rmarkdown_2.29         gtable_0.3.6           codetools_0.2-20      
[67] DBI_1.2.3              R6_2.5.1               knitr_1.49            
[70] fastmap_1.2.0          bit_4.5.0.1            rprojroot_2.0.4       
[73] stringi_1.8.4          parallel_4.2.2         Rcpp_1.0.14           
[76] vctrs_0.6.5            geneplotter_1.76.0     png_0.1-8             
[79] tidyselect_1.2.1       xfun_0.50             </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
