---
title: "Differential expression"
output: html_document
---

```{r setup, include=FALSE, message = F, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(data.table)
library(dplyr)
library(ggplot2)
library(corrplot)
library(DESeq2)
library(SummarizedExperiment)
library(BiocParallel)
library(ggrepel)
library(genekitr)
library(org.Hs.eg.db)
library(enrichplot)
library(clusterProfiler)
```

# Correlation between PRS & expression PCs (whole blood)

```{r}
# load prs & pcs
metadata_file <- "analysis/metadata.txt"
metadata <- read.csv(metadata_file, header = T, sep = "\t", stringsAsFactors = T)
metadata$sex <- as.factor(metadata$sex)

traits <- metadata[, 7:43]
pc <- metadata[, c(1:5)]

# Calculate the correlation between each trait and each PC
correlation_matrix <- cor(traits, pc)
range(correlation_matrix)
correlation_matrix <- t(correlation_matrix)

# Create the heatmap using corrplot
corrplot(correlation_matrix, method = "color", 
         col = colorRampPalette(c("blue", "white", "red"))(200), # color palette
         addCoef.col = "black", # Add correlation coefficients to the plot
         number.cex = 0.4, # Adjust the font size of the numbers
         tl.col = "black", # text label color
         tl.srt = 90, # rotate text labels
         tl.cex = 0.5, 
         title = "Correlation between Traits and expression PCs", 
         mar = c(0, 0, 1, 0)
)
```

# Perform DESeq2 differential expression analysis for each trait

## Model 1: expression \~ PRS + sex + expression PCs

**Continuous PRS:**

```{r, warning=FALSE, eval=FALSE}
# Load the gene expression data
gene_expr_file <- "data/gene_reads_2017-06-05_v8_whole_blood.gct"
raw_count_df <- fread(gene_expr_file, header = TRUE, sep = "\t", drop = "id")

# load protein_coding list
protein_coding <- fread("data/protein-coding_gene.txt", sep = "\t")
protein_coding <- protein_coding[, c("symbol", "ensembl_gene_id")]

# keep only protein-coding genes
raw_count_df <- raw_count_df[raw_count_df$Description %in% protein_coding$symbol, ]

id <- raw_count_df$Name
raw_count <- raw_count_df[, -c(1:2)]

# modify GTEx sample names matching names used in PRS data
colnames(raw_count) <- sub("^(GTEX-[^-.]+).*", "\\1", colnames(raw_count))

matching_samples <- intersect(rownames(metadata), colnames(raw_count))
final_count <- raw_count[ , ..matching_samples]

# prefilter: keep only rows that have a count of at least 10
keep_genes <- rowSums(final_count >= 10) > 0
final_count <- final_count[keep_genes, ]
id <- id[keep_genes]
dim(final_count)

# Loop through each trait and run DESeq2
for (trait in colnames(traits)) {
  
  # Standardize PRS for the current trait
  prs_trait <- traits[,trait]
  prs_trait <- scale(prs_trait) # Standardize PRS to mean = 0, sd = 1
    
  # Add the standardized PRS to the metadata for continuous trait
  metadata[,trait] <- prs_trait
  
  # Create the DESeqDataSet for the current trait
  dds <- DESeqDataSetFromMatrix(
    countData = as.matrix(final_count),  # Raw counts
    colData = metadata[, c(1:6, which(colnames(metadata) == trait))],
    design = as.formula(paste("~ PC1 + PC2 + PC3 + PC4 + PC5 + sex +", trait))
  )
  rownames(dds) <- id
  
  # Run DESeq2 analysis
  dds <- DESeq(dds, parallel = TRUE, BPPARAM = MulticoreParam(4))
  
  # Get the results for the current trait
  res <- results(dds)
  
  # Save the results to a file
  write.csv(res, paste0("differential_expression_", trait, "_results.csv"))
  
  # print a summary of the results
  print(paste("Results for trait:", trait))
  print(summary(res))
  
  # plot the MA-plot for the current trait
  png(paste0("ma_plot_", trait, ".png"), width = 800, height = 600)
  plotMA(res, main = paste("Continuous: MA Plot for", trait))
  dev.off()
  
  # volcano plot
  res_tableOE <- as.data.frame(res)
  res_tableOE$gene_name <- raw_count_df$Description[keep_genes]
  res_tableOE <- mutate(res_tableOE, threshold_OE = padj < 0.1)
  res_tableOE <- res_tableOE %>% arrange(padj) %>% mutate(genelabels = "")
  res_tableOE$genelabels[1:10] <- res_tableOE$gene_name[1:10]
  
  volcano_plot <- ggplot(res_tableOE, aes(x = log2FoldChange, y = -log10(padj))) +
    geom_point(aes(colour = threshold_OE)) +
    geom_text_repel(aes(label = genelabels)) +
    ggtitle(paste("Continuous: Volcano Plot for", trait)) +
    xlab("log2 fold change") + 
    ylab("-log10 adjusted p-value") +
    theme(legend.position = "none",
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
  
  # Save the volcano plot
  png(paste0("volcano_plot_", trait, ".png"), width = 800, height = 600)
  print(volcano_plot)
  dev.off()
}
```

**Quantile PRS:**

```{r, warning=FALSE, eval=FALSE}
metadata_file <- "analysis/metadata_quantile.txt"
metadata <- read.csv(metadata_file, header = T, sep = "\t", stringsAsFactors = T)
metadata$sex <- as.factor(metadata$sex)

traits <- metadata[, 7:43]

# Loop through each trait and run DESeq2
for (trait in colnames(traits)) {
  
  # Create the DESeqDataSet for the current trait
  dds <- DESeqDataSetFromMatrix(
    countData = as.matrix(final_count),  # Raw counts
    colData = metadata[, c(1:6, which(colnames(metadata) == trait))],
    design = as.formula(paste("~ PC1 + PC2 + PC3 + PC4 + PC5 + sex +", trait))
  )
  rownames(dds) <- id
  
  # Run DESeq2 analysis
  dds <- DESeq(dds, parallel = TRUE, BPPARAM = MulticoreParam(4))
  
  # Get the results for the current trait
  res <- results(dds)
  
  # Save the results to a file
  write.csv(res, paste0("differential_expression_", trait, "_quantile_results.csv"))
  
  # print a summary of the results
  print(paste("Results for trait:", trait))
  print(summary(res))
  
  # plot the MA-plot for the current trait
  png(paste0("ma_plot_quantile_", trait, ".png"), width = 800, height = 600)
  plotMA(res, main = paste("Quantile: MA Plot for", trait))
  dev.off()
  
  # volcano plot
  res_tableOE <- as.data.frame(res)
  res_tableOE$gene_name <- raw_count_df$Description[keep_genes]
  res_tableOE <- mutate(res_tableOE, threshold_OE = padj < 0.1 & 
                          abs(log2FoldChange) >= 0.5)
  res_tableOE <- res_tableOE %>% arrange(padj) %>% mutate(genelabels = "")
  res_tableOE$genelabels[1:10] <- res_tableOE$gene_name[1:10]
  
  volcano_plot <- ggplot(res_tableOE, aes(x = log2FoldChange, y = -log10(padj))) +
    geom_point(aes(colour = threshold_OE)) +
    geom_text_repel(aes(label = genelabels)) +
    ggtitle(paste("Quantile:Volcano Plot for", trait)) +
    xlab("log2 fold change") + 
    ylab("-log10 adjusted p-value") +
    theme(legend.position = "none",
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
  
  # Save the volcano plot
  png(paste0("volcano_plot_quantile_", trait, ".png"), width = 800, height = 600)
  print(volcano_plot)
  dev.off()
}
```

## Model 2: expression \~ PRS + sex + expression PCs + 2 genotype PCs

**Continuous PRS:**

```{r}
# Load the gene expression data
gene_expr_file <- "data/gene_reads_2017-06-05_v8_whole_blood.gct"
raw_count_df <- fread(gene_expr_file, header = TRUE, sep = "\t", drop = "id")

# load protein_coding list
protein_coding <- fread("data/protein-coding_gene.txt", 
                        sep = "\t")
protein_coding <- protein_coding[, c("symbol", "ensembl_gene_id")]

# keep only protein-coding genes
raw_count_df <- raw_count_df[raw_count_df$Description %in% protein_coding$symbol, ]

id <- raw_count_df$Name
raw_count <- raw_count_df[, -c(1:2)]

# modify GTEx sample names matching names used in PRS data
colnames(raw_count) <- sub("^(GTEX-[^-.]+).*", "\\1", colnames(raw_count))

matching_samples <- intersect(rownames(metadata), colnames(raw_count))
final_count <- raw_count[ , ..matching_samples]

# prefilter: keep only rows that have a count of at least 10
keep_genes <- rowSums(final_count >= 10) > 0
final_count <- final_count[keep_genes, ]
id <- id[keep_genes]
dim(final_count)

# obtain genotype PCs
geno_pc <- read.table("data/pca.eigenvec")
names(geno_pc) = c("FID","IID",paste0("geno_PC", c(1:(ncol(geno_pc)-2))))

eigenval <- scan("data/pca.eigenval")
pve <- data.frame(PC = 1:20, pve = eigenval/sum(eigenval)*100)

ggplot(geno_pc) + geom_point(aes(x = geno_PC1, y = geno_PC2)) + 
  labs(x = paste0("PC1 (", signif(pve$pve[1], 3), "%)"), 
       y = paste0("PC2 (", signif(pve$pve[2], 3), "%)")) + theme_classic()
ggplot(geno_pc) + geom_point(aes(x = geno_PC1, y = geno_PC3)) + 
    labs(x = paste0("PC1 (", signif(pve$pve[1], 3), "%)"), 
         y = paste0("PC3 (", signif(pve$pve[3], 3), "%)")) + theme_classic()

ggplot(pve, aes(PC, pve)) + geom_point() + geom_line() +
  labs(x = "Genotype PC", y = "Percentage variance explained") + theme_classic()
```

```{r, warning=FALSE, eval=FALSE}
metadata <- cbind(metadata, geno_pc[,3:ncol(geno_pc)])

# Loop through each trait and run DESeq2
for (trait in colnames(traits)) {
  
  # Standardize PRS for the current trait
  prs_trait <- traits[,trait]
  prs_trait <- scale(prs_trait) # Standardize PRS to mean = 0, sd = 1
    
  # Add the standardized PRS to the metadata for continuous trait
  metadata[,trait] <- prs_trait
  
  # Create the DESeqDataSet for the current trait
  dds <- DESeqDataSetFromMatrix(
    countData = as.matrix(final_count),  # Raw counts
    colData = metadata[, c(1:6, 44:45, which(colnames(metadata) == trait))],
    design = as.formula(paste("~ PC1 + PC2 + PC3 + PC4 + PC5 + sex + geno_PC1 + geno_PC2 +", trait))
  )
  rownames(dds) <- id
  
  # Run DESeq2 analysis
  dds <- DESeq(dds, parallel = TRUE, BPPARAM = MulticoreParam(4))
  
  # Get the results for the current trait
  res <- results(dds)
  
  # Save the results to a file
  write.csv(res, paste0("differential_expression_", trait, "_results_M2.csv"))
  
  # print a summary of the results
  print(paste("Results for trait:", trait))
  print(summary(res))
  
  # plot the MA-plot for the current trait
  png(paste0("ma_plot_", trait, "_M2.png"), width = 800, height = 600)
  plotMA(res, main = paste("Continuous (M2): MA Plot for", trait))
  dev.off()
  
  # volcano plot
  res_tableOE <- as.data.frame(res)
  res_tableOE$gene_name <- raw_count_df$Description[keep_genes]
  res_tableOE <- mutate(res_tableOE, threshold_OE = padj < 0.1)
  res_tableOE <- res_tableOE %>% arrange(padj) %>% mutate(genelabels = "")
  res_tableOE$genelabels[1:10] <- res_tableOE$gene_name[1:10]
  
  volcano_plot <- ggplot(res_tableOE, aes(x = log2FoldChange, y = -log10(padj))) +
    geom_point(aes(colour = threshold_OE)) +
    geom_text_repel(aes(label = genelabels)) +
    ggtitle(paste("Continuous (M2): Volcano Plot for", trait)) +
    xlab("log2 fold change") + 
    ylab("-log10 adjusted p-value") +
    theme(legend.position = "none",
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
  
  # Save the volcano plot
  png(paste0("volcano_plot_", trait, "_M2.png"), width = 800, height = 600)
  print(volcano_plot)
  dev.off()
}
```

**Quantile PRS:**

```{r, warning=FALSE, eval=FALSE}
metadata_file <- "analysis/metadata_quantile.txt"
metadata <- read.csv(metadata_file, header = T, sep = "\t", stringsAsFactors = T)
metadata$sex <- as.factor(metadata$sex)

traits <- metadata[, 7:43]

metadata <- cbind(metadata, geno_pc[,3:ncol(geno_pc)])

# Loop through each trait and run DESeq2
for (trait in colnames(traits)) {
  
  # Create the DESeqDataSet for the current trait
  dds <- DESeqDataSetFromMatrix(
    countData = as.matrix(final_count),  # Raw counts
    colData = metadata[, c(1:6, which(colnames(metadata) == trait))],
    design = as.formula(paste("~ PC1 + PC2 + PC3 + PC4 + PC5 + sex + geno_PC1 + geno_PC2 +", trait))
  )
  rownames(dds) <- id
  
  # Run DESeq2 analysis
  dds <- DESeq(dds, parallel = TRUE, BPPARAM = MulticoreParam(4))
  
  # Get the results for the current trait
  res <- results(dds)
  
  # Save the results to a file
  write.csv(res, paste0("differential_expression_", trait, "_quantile_results_M2.csv"))
  
  # print a summary of the results
  print(paste("Results for trait:", trait))
  print(summary(res))
  
  # plot the MA-plot for the current trait
  png(paste0("ma_plot_quantile_", trait, "_M2.png"), width = 800, height = 600)
  plotMA(res, main = paste("Quantile (M2): MA Plot for", trait))
  dev.off()
  
  # volcano plot
  res_tableOE <- as.data.frame(res)
  res_tableOE$gene_name <- raw_count_df$Description[keep_genes]
  res_tableOE <- mutate(res_tableOE, threshold_OE = padj < 0.1 & 
                          abs(log2FoldChange) >= 0.5)
  res_tableOE <- res_tableOE %>% arrange(padj) %>% mutate(genelabels = "")
  res_tableOE$genelabels[1:10] <- res_tableOE$gene_name[1:10]
  
  volcano_plot <- ggplot(res_tableOE, aes(x = log2FoldChange, y = -log10(padj))) +
    geom_point(aes(colour = threshold_OE)) +
    geom_text_repel(aes(label = genelabels)) +
    ggtitle(paste("Quantile (M2):Volcano Plot for", trait)) +
    xlab("log2 fold change") + 
    ylab("-log10 adjusted p-value") +
    theme(legend.position = "none",
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
  
  # Save the volcano plot
  png(paste0("volcano_plot_quantile_", trait, "_M2.png"), width = 800, height = 600)
  print(volcano_plot)
  dev.off()
}
```

# GO enrichment Analysis

Gene Ontology enrichment is applied to both models, with and without adjusting 2 genotype PCs, for both continuous & quantile PRS.

**Continuous PRS:**

```{r, warning=FALSE, eval=FALSE}
dir_path <- "analysis/continuous_wb"
files <- list.files(dir_path, pattern = "differential_expression_.*_results.csv", 
                    full.names = TRUE)

for (file in files) {
  trait <- gsub("differential_expression_(.*)_results.csv", "\\1", basename(file))
  trait
  res_tableOE <- read.csv(file, header = T, row.names = 1)
  
  deGenes <- res_tableOE[res_tableOE$padj < 0.1 & 
                           abs(res_tableOE$log2FoldChange) >= 0.5, ]
  deGenes$gene_id <- gsub("\\.\\d+$", "", rownames(deGenes))
  
  # Separate upregulated and downregulated genes
  upregulated_genes <- deGenes[deGenes$log2FoldChange > 0, ]$gene_id
  downregulated_genes <- deGenes[deGenes$log2FoldChange < 0, ]$gene_id
  
  # Run GO enrichment for upregulated genes
  gse_up <- enrichGO(gene = upregulated_genes, ont = "BP", 
                   OrgDb = "org.Hs.eg.db", keyType = "ENSEMBL", readable = T)
  
  # Run GO enrichment for downregulated genes
  gse_down <- enrichGO(gene = downregulated_genes, ont = "BP", 
                     OrgDb = "org.Hs.eg.db", keyType = "ENSEMBL", readable = T)
  
  # Convert enrichment results to data frames and calculate additional ratios
  if (is.null(gse_up)) {
    gse_up <- data.frame(ID = character(), Description = character(),
                       GeneRatio = character(), BgRatio = character(),
                       pvalue = numeric(), p.adjust = numeric(),
                       qvalue = numeric(), geneID = character(),
                       Count = integer(), stringsAsFactors = FALSE)
  } else { 
    gse_up <- as.data.frame(gse_up)}
  
  gse_down <- as.data.frame(gse_down)
  
  gse_up$GeneRatio_num <- as.numeric(sapply(strsplit(gse_up$GeneRatio, "/"), function(x) x[1])) / 
                        as.numeric(sapply(strsplit(gse_up$GeneRatio, "/"), function(x) x[2]))
  gse_up$BgRatio_num <- as.numeric(sapply(strsplit(gse_up$BgRatio, "/"), function(x) x[1])) / 
                        as.numeric(sapply(strsplit(gse_up$BgRatio, "/"), function(x) x[2]))
  gse_up <- cbind(gse_up, FoldEnrich = gse_up$GeneRatio_num / gse_up$BgRatio_num)

  gse_down$GeneRatio_num <- as.numeric(sapply(strsplit(gse_down$GeneRatio, "/"), function(x) x[1])) / 
                          as.numeric(sapply(strsplit(gse_down$GeneRatio, "/"), function(x) x[2]))
  gse_down$BgRatio_num <- as.numeric(sapply(strsplit(gse_down$BgRatio, "/"), function(x) x[1])) / 
                          as.numeric(sapply(strsplit(gse_down$BgRatio, "/"), function(x) x[2]))
  gse_down <- cbind(gse_down, FoldEnrich = gse_down$GeneRatio_num / gse_down$BgRatio_num)
  
if (nrow(gse_up) >= 20) {
  enrich_plot_up <- plotEnrich(gse_up[1:20, ], plot_type = "dot", scale_ratio = 0.4) +
    labs(title = paste("Continuous: Upregulated Enrichment Pathways for", trait)) +
    theme(plot.title = element_text(size = 6), axis.text.y   = element_text(size = 5))
} else {
  enrich_plot_up <- plotEnrich(gse_up, plot_type = "dot", scale_ratio = 0.4) +
    labs(title = paste("Continuous: Upregulated Enrichment Pathways for", trait)) +
    theme(plot.title = element_text(size = 6), axis.text.y   = element_text(size = 5))
}
  
if (nrow(gse_down) >= 20) {
  enrich_plot_down <- plotEnrich(gse_down[1:20, ], plot_type = "dot", scale_ratio = 0.4) +
    labs(title = paste("Continuous: Downregulated Enrichment Pathways for", trait)) +
    theme(plot.title = element_text(size = 6), axis.text.y   = element_text(size = 5))
} else {
  enrich_plot_down <- plotEnrich(gse_down, plot_type = "dot", scale_ratio = 0.4) +
    labs(title = paste("Continuous: Downregulated Enrichment Pathways for", trait)) +
    theme(plot.title = element_text(size = 6), axis.text.y   = element_text(size = 5))
}
  
  # Arrange the two plots side by side
  combined_plot <- grid.arrange(enrich_plot_up, enrich_plot_down, ncol = 2)
  
  # Save the combined plot
  ggsave(paste0("enrichment_plot_", trait, ".png"), plot = combined_plot, 
         width = 15, height = 6)
  
  # Save the GO enrichment results to CSV
  write.csv(gse_up, file = paste0("GO_enrichment_", trait, "_upregulated.csv"))
  write.csv(gse_down, file = paste0("GO_enrichment_", trait, "_downregulated.csv"))
}
```

**Quantile PRS:**

```{r, warning=FALSE, eval=FALSE}
dir_path <- "analysis/quantile_wb"
files <- list.files(dir_path, pattern = "differential_expression_.*_quantile_results.csv", 
                    full.names = TRUE)

for (file in files) {
  trait <- gsub("differential_expression_(.*)_quantile_results.csv", "\\1", basename(file))
  trait
  res_tableOE <- read.csv(file, header = T, row.names = 1)
  
  deGenes <- res_tableOE[res_tableOE$padj < 0.1 & 
                           abs(res_tableOE$log2FoldChange) >= 0.5, ]
  deGenes$gene_id <- gsub("\\.\\d+$", "", rownames(deGenes))
  
  # Separate upregulated and downregulated genes
  upregulated_genes <- deGenes[deGenes$log2FoldChange > 0, ]$gene_id
  downregulated_genes <- deGenes[deGenes$log2FoldChange < 0, ]$gene_id
  
  # Run GO enrichment for upregulated genes
  gse_up <- enrichGO(gene = upregulated_genes, ont = "BP", 
                   OrgDb = "org.Hs.eg.db", keyType = "ENSEMBL", readable = T)
  
  # Run GO enrichment for downregulated genes
  gse_down <- enrichGO(gene = downregulated_genes, ont = "BP", 
                     OrgDb = "org.Hs.eg.db", keyType = "ENSEMBL", readable = T)
  
  # Convert enrichment results to data frames and calculate additional ratios
  if (is.null(gse_up)) {
    gse_up <- data.frame(ID = character(), Description = character(),
                       GeneRatio = character(), BgRatio = character(),
                       pvalue = numeric(), p.adjust = numeric(),
                       qvalue = numeric(), geneID = character(),
                       Count = integer(), stringsAsFactors = FALSE)
  } else { 
    gse_up <- as.data.frame(gse_up)}
  
  gse_down <- as.data.frame(gse_down)
  
  gse_up$GeneRatio_num <- as.numeric(sapply(strsplit(gse_up$GeneRatio, "/"), function(x) x[1])) / 
                        as.numeric(sapply(strsplit(gse_up$GeneRatio, "/"), function(x) x[2]))
  gse_up$BgRatio_num <- as.numeric(sapply(strsplit(gse_up$BgRatio, "/"), function(x) x[1])) / 
                        as.numeric(sapply(strsplit(gse_up$BgRatio, "/"), function(x) x[2]))
  gse_up <- cbind(gse_up, FoldEnrich = gse_up$GeneRatio_num / gse_up$BgRatio_num)

  gse_down$GeneRatio_num <- as.numeric(sapply(strsplit(gse_down$GeneRatio, "/"), function(x) x[1])) / 
                          as.numeric(sapply(strsplit(gse_down$GeneRatio, "/"), function(x) x[2]))
  gse_down$BgRatio_num <- as.numeric(sapply(strsplit(gse_down$BgRatio, "/"), function(x) x[1])) / 
                          as.numeric(sapply(strsplit(gse_down$BgRatio, "/"), function(x) x[2]))
  gse_down <- cbind(gse_down, FoldEnrich = gse_down$GeneRatio_num / gse_down$BgRatio_num)
  
if (nrow(gse_up) >= 20) {
  enrich_plot_up <- plotEnrich(gse_up[1:20, ], plot_type = "dot", scale_ratio = 0.4) +
    labs(title = paste("Quantile: Upregulated Enrichment Pathways for", trait)) +
    theme(plot.title = element_text(size = 6), axis.text.y   = element_text(size = 5))
} else {
  enrich_plot_up <- plotEnrich(gse_up, plot_type = "dot", scale_ratio = 0.4) +
    labs(title = paste("Quantile: Upregulated Enrichment Pathways for", trait)) +
    theme(plot.title = element_text(size = 6), axis.text.y   = element_text(size = 5))
}
  
if (nrow(gse_down) >= 20) {
  enrich_plot_down <- plotEnrich(gse_down[1:20, ], plot_type = "dot", scale_ratio = 0.4) +
    labs(title = paste("Quantile: Downregulated Enrichment Pathways for", trait)) +
    theme(plot.title = element_text(size = 6), axis.text.y   = element_text(size = 5))
} else {
  enrich_plot_down <- plotEnrich(gse_down, plot_type = "dot", scale_ratio = 0.4) +
    labs(title = paste("Quantile: Downregulated Enrichment Pathways for", trait)) +
    theme(plot.title = element_text(size = 6), axis.text.y   = element_text(size = 5))
}
  
  # Arrange the two plots side by side
  combined_plot <- grid.arrange(enrich_plot_up, enrich_plot_down, ncol = 2)
  
  # Save the combined plot
  ggsave(paste0("enrichment_plot_quantile_", trait, ".png"), plot = combined_plot, 
         width = 15, height = 6)
  
  # Save the GO enrichment results to CSV
  write.csv(gse_up, file = paste0("GO_enrichment_", trait, "_quantile_upregulated.csv"))
  write.csv(gse_down, file = paste0("GO_enrichment_", trait, "_quantile_downregulated.csv"))
}
```

# Result

## Summary table

```{r, echo = F}
continuous_dir <- "analysis/continuous_wb"
continuous_m2_dir <- "analysis/continuous_wb_m2"
quantile_dir <- "analysis/quantile_wb"
quantile_m2_dir <- "analysis/quantile_wb_m2"

# Get the list of differential expression results and GO enrichment results
de_files_continuous <- list.files(continuous_dir, pattern = "differential_expression_.*_results.csv", 
                                  full.names = TRUE)
up_files_continuous <- list.files(continuous_dir, pattern = "GO_enrichment_.*_upregulated.csv", 
                                  full.names = TRUE)
down_files_continuous <- list.files(continuous_dir, pattern = "GO_enrichment_.*_downregulated.csv", 
                                  full.names = TRUE)
de_files_continuous_m2 <- list.files(continuous_m2_dir, pattern = "differential_expression_.*_results_M2.csv", 
                                     full.names = TRUE)
up_files_continuous_m2 <- list.files(continuous_m2_dir, pattern = "GO_enrichment_.*_upregulated_M2.csv", 
                                  full.names = TRUE)
down_files_continuous_m2 <- list.files(continuous_m2_dir, pattern = "GO_enrichment_.*_downregulated_M2.csv", 
                                  full.names = TRUE)

de_files_quantile <- list.files(quantile_dir, pattern = "differential_expression_.*_quantile_results.csv", 
                                full.names = TRUE)
up_files_quantile <- list.files(quantile_dir, pattern = "GO_enrichment_.*_quantile_upregulated.csv", 
                                full.names = TRUE)
down_files_quantile <- list.files(quantile_dir, pattern = "GO_enrichment_.*_quantile_downregulated.csv", 
                                full.names = TRUE)
de_files_quantile_m2 <- list.files(quantile_m2_dir, pattern = "differential_expression_.*_quantile_results_M2.csv", 
                                full.names = TRUE)
up_files_quantile_m2 <- list.files(quantile_m2_dir, pattern = "GO_enrichment_.*_quantile_upregulated_M2.csv", 
                                full.names = TRUE)
down_files_quantile_m2 <- list.files(quantile_m2_dir, pattern = "GO_enrichment.*_quantile_downregulated_M2.csv", 
                                full.names = TRUE)

# Initialize a data frame to store the results
results_df_continuous <- data.frame(trait = character(),
                                    num_significant_de = integer(),
                                    upregulated = integer(),
                                    downregulated = integer(),
                                    up_enriched_go = integer(),
                                    down_enriched_go = integer(),
                                    stringsAsFactors = FALSE)

results_df_quantile <- data.frame(trait = character(),
                                  num_significant_de = integer(),
                                  upregulated = integer(),
                                  downregulated = integer(),
                                  up_enriched_go = integer(),
                                  down_enriched_go = integer(),
                                  stringsAsFactors = FALSE)

results_df_continuous_m2 <- data.frame(trait = character(),
                                    num_significant_de = integer(),
                                    upregulated = integer(),
                                    downregulated = integer(),
                                    up_enriched_go = integer(),
                                    down_enriched_go = integer(),
                                    stringsAsFactors = FALSE)

results_df_quantile_m2 <- data.frame(trait = character(),
                                  num_significant_de = integer(),
                                  upregulated = integer(),
                                  downregulated = integer(),
                                  up_enriched_go = integer(),
                                  down_enriched_go = integer(),
                                  stringsAsFactors = FALSE)

# Function to extract significant DE genes and GO enriched pathways
extract_results <- function(de_file, up_go_file, down_go_file, analysis_type) {
  # Load the differential expression results
  res <- read.csv(de_file, header = T, row.names = 1)
  
  # Filter for significant DE genes (padj < 0.1)
  significant_de <- res[!is.na(res$padj) & res$padj < 0.1, ]
  
  # Count number of upregulated and downregulated genes
  upregulated <- sum(significant_de$log2FoldChange > 0)
  downregulated <- sum(significant_de$log2FoldChange < 0)
  
  # Load the GO enrichment results
  up_go   <- read.csv(up_go_file,   header = TRUE, row.names = 1)
  down_go <- read.csv(down_go_file, header = TRUE, row.names = 1)
  
  up_paths   <- sum(up_go$p.adjust < 0.05, na.rm = TRUE)
  down_paths <- sum(down_go$p.adjust < 0.05, na.rm = TRUE)
  
  # Extract trait name from the file name
  trait_name <- gsub("differential_expression_|_results(.*).csv", "", basename(de_file))
  
  # Create a row for this trait and add it to the results data frame
  new_row <- data.frame(
    trait = trait_name,
    num_significant_de = nrow(significant_de),
    upregulated = upregulated,
    downregulated = downregulated,
    up_enriched_go = up_paths,
    down_enriched_go = down_paths 
  )
  if (analysis_type == "continuous") {
    results_df_continuous <<- rbind(results_df_continuous, new_row)
  } else if (analysis_type == "quantile") {
    results_df_quantile <<- rbind(results_df_quantile, new_row)
  } else if (analysis_type == "continuous_m2") {
    results_df_continuous_m2 <<- rbind(results_df_continuous_m2, new_row)
  } else if (analysis_type == "quantile_m2") {
    results_df_quantile_m2 <<- rbind(results_df_quantile_m2, new_row)
  }
}

# Loop through the continuous files and extract results
for (i in seq_along(de_files_continuous)) {
  extract_results(de_files_continuous[i], up_files_continuous[i], down_files_continuous[i], "continuous")
}

for (i in seq_along(de_files_continuous_m2)) {
  extract_results(de_files_continuous_m2[i], up_files_continuous_m2[i], down_files_continuous_m2[i], "continuous_m2")
}

# Loop through the quantile files and extract results
for (i in seq_along(de_files_quantile)) {
  extract_results(de_files_quantile[i], up_files_quantile[i], down_files_quantile[i], "quantile")
}

for (i in seq_along(de_files_quantile)) {
  extract_results(de_files_quantile_m2[i], up_files_quantile_m2[i], down_files_quantile_m2[i], "quantile_m2")
}

colnames(results_df_continuous) <- c("Trait", "Significant DE genes", 
                                     "Up-regulated Genes", "Down-regulated Genes", 
                                     "Up-regulated GO pathways", "Down-regulated GO pathways")
colnames(results_df_quantile) <- c("Trait", "Significant DE genes", 
                                   "Up-regulated Genes", "Down-regulated Genes", 
                                   "Up-regulated GO pathways", "Down-regulated GO pathways")
colnames(results_df_continuous_m2) <- c("Trait", "Significant DE genes", 
                                     "Up-regulated Genes", "Down-regulated Genes", 
                                     "Up-regulated GO pathways", "Down-regulated GO pathways")
colnames(results_df_quantile_m2) <- c("Trait", "Significant DE genes", 
                                     "Up-regulated Genes", "Down-regulated Genes", 
                                     "Up-regulated GO pathways", "Down-regulated GO pathways")

knitr::kable(results_df_continuous, caption = "Continuous PRS without adjusting genotype PCs")
knitr::kable(results_df_continuous_m2, caption = "Continuous PRS with adjusting genotype PCs")
knitr::kable(results_df_quantile, caption = "Quantile PRS without adjusting genotype PCs")
knitr::kable(results_df_quantile_m2, caption = "Quantile PRS with adjusting genotype PCs")
```

**Interpretation:**

In the quantile analysis, the number of significant differential expressed genes is generally higher compared to the continuous analysis, which suggests that stratifying by PRS (top 25%) may identify more genes with stronger associations to the traits. We observed that certain genes were identified in both the continuous and quantile PRS analyses, though not all genes overlapped. Specifically, for the Eosinophil count trait, ACTN3, HAMP, SCGB3A1, CH25H, and NPIPA8 appeared among the top 10 DEGs across both analyses, suggesting that these genes might play a key role in the trait regardless of the analysis approach. For example:

-   HAMP encodes Hepcidin, which is involved in the maintenance of iron homeostasis.

-   SCGB3A1 predicts to enable cytokin activity.

-   CH25H encodes for cholesterol 25-hydroxylase, involved in lipid metabolism and modulating immune responses.

Additionally, the GO enrichment analysis provides insights into the biological pathways and processes that are potentially involved in these traits. BgRatio is the ratio of the total number of genes in a specific pathway (GO term) to the total number of genes. GeneRatio is the ratio of the number of genes enriched in specific pathway to the number of genes that were tested in the analysis. FoldEnrichment is the ratio of GeneRatio to BgRatio, showing how much more enriched a pathway is in your genes of interest compared to the total gene set. Higher FoldEnrichment values indicate that a particular GO term is overrepresented in your gene set compared to the general gene pool, suggesting it may be biologically relevant for the trait. Some traits show higher pathway enrichment in the quantile analysis, indicating that stratification by PRS may provide more biologically meaningful results.

For highly correlated traits, they tend to share similar GO enriched pathways. For example, mean corpuscular hemoglobin and mean corpucular volume are both involved in read blood cell biology, so it's expected that their GO enrichment would point to similar biological processes, such as hemoglobin metabolism, iron homeostasis and erythropoiesis.

To account for population stratification, we adjusted the model by including the first two genotype principal components (PCs), as they explain most of the genetic variation in the data. For Eosinophil count, we observed that genes were consistently identified as top differentially expressed with and without adjustment. The only difference is that instead of SLC6A9, WFDC2 was detected as top 10 DEGs when adjusting for the first two genotype PCs. WFBC2 is a protease inhibitor with a potential role in innate immune defense ([L Bingle, 2006](https://doi.org/10.1186/1465-9921-7-61)). Although same genes are identified as top DEGs for continuous analysis, additional pathways such as response to inactivity were highlighted when adjusting for the first two genotype PCs.

## Basophil count

![](figure/differential_expression.Rmd/ma_plot_Basophil.count.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Basophil.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_Basophil.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Basophil.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Basophil.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Basophil.count.png)
 
![](figure/differential_expression.Rmd/ma_plot_Basophil.count_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Basophil.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Basophil.count_M2.png)
![](figure/differential_expression.Rmd/volcano_plot_quantile_Basophil.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Basophil.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Basophil.count_M2.png)

## Basophil.percentage.of.white.cells

![](figure/differential_expression.Rmd/ma_plot_Basophil.percentage.of.white.cells.png)
![](figure/differential_expression.Rmd/ma_plot_quantile_Basophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_Basophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Basophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Basophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Basophil.percentage.of.white.cells.png)

![](figure/differential_expression.Rmd/ma_plot_Basophil.percentage.of.white.cells_M2.png)
![](figure/differential_expression.Rmd/ma_plot_quantile_Basophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Basophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Basophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Basophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Basophil.percentage.of.white.cells_M2.png)

## Eosinophil.count

![](figure/differential_expression.Rmd/ma_plot_Eosinophil.count.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Eosinophil.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_Eosinophil.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Eosinophil.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Eosinophil.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Eosinophil.count.png)

![](figure/differential_expression.Rmd/ma_plot_Eosinophil.count_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Eosinophil.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Eosinophil.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Eosinophil.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Eosinophil.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Eosinophil.count_M2.png)

## Eosinophil.percentage.of.white.cells

![](figure/differential_expression.Rmd/ma_plot_Eosinophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Eosinophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_Eosinophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Eosinophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Eosinophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Eosinophil.percentage.of.white.cells.png)

![](figure/differential_expression.Rmd/ma_plot_Eosinophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Eosinophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Eosinophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Eosinophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Eosinophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Eosinophil.percentage.of.white.cells_M2.png)

## Hematocrit

![](figure/differential_expression.Rmd/ma_plot_Hematocrit.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Hematocrit.png) 
![](figure/differential_expression.Rmd/volcano_plot_Hematocrit.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Hematocrit.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Hematocrit.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Hematocrit.png)

![](figure/differential_expression.Rmd/ma_plot_Hematocrit_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Hematocrit_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Hematocrit_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Hematocrit_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Hematocrit_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Hematocrit_M2.png)

## Hemoglobin.concentration

![](figure/differential_expression.Rmd/ma_plot_Hemoglobin.concentration.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Hemoglobin.concentration.png) 
![](figure/differential_expression.Rmd/volcano_plot_Hemoglobin.concentration.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Hemoglobin.concentration.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Hemoglobin.concentration.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Hemoglobin.concentration.png)

![](figure/differential_expression.Rmd/ma_plot_Hemoglobin.concentration_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Hemoglobin.concentration_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Hemoglobin.concentration_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Hemoglobin.concentration_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Hemoglobin.concentration_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Hemoglobin.concentration.png)

## High.light.scatter.reticulocyte.count

![](figure/differential_expression.Rmd/ma_plot_High.light.scatter.reticulocyte.count.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_High.light.scatter.reticulocyte.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_High.light.scatter.reticulocyte.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_High.light.scatter.reticulocyte.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_High.light.scatter.reticulocyte.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_High.light.scatter.reticulocyte.count.png)

![](figure/differential_expression.Rmd/ma_plot_High.light.scatter.reticulocyte.count_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_High.light.scatter.reticulocyte.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_High.light.scatter.reticulocyte.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_High.light.scatter.reticulocyte.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_High.light.scatter.reticulocyte.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_High.light.scatter.reticulocyte.count_M2.png)

## High.light.scatter.reticulocyte.percentage.of.red.cells

![](figure/differential_expression.Rmd/ma_plot_High.light.scatter.reticulocyte.percentage.of.red.cells.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_High.light.scatter.reticulocyte.percentage.of.red.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_High.light.scatter.reticulocyte.percentage.of.red.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_High.light.scatter.reticulocyte.percentage.of.red.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_High.light.scatter.reticulocyte.percentage.of.red.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_High.light.scatter.reticulocyte.percentage.of.red.cells.png)

![](figure/differential_expression.Rmd/ma_plot_High.light.scatter.reticulocyte.percentage.of.red.cells_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_High.light.scatter.reticulocyte.percentage.of.red.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_High.light.scatter.reticulocyte.percentage.of.red.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_High.light.scatter.reticulocyte.percentage.of.red.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_High.light.scatter.reticulocyte.percentage.of.red.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_High.light.scatter.reticulocyte.percentage.of.red.cells_M2.png)

## Immature.fraction.of.reticulocytes

![](figure/differential_expression.Rmd/ma_plot_Immature.fraction.of.reticulocytes.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Immature.fraction.of.reticulocytes.png) 
![](figure/differential_expression.Rmd/volcano_plot_Immature.fraction.of.reticulocytes.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Immature.fraction.of.reticulocytes.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Immature.fraction.of.reticulocytes.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Immature.fraction.of.reticulocytes.png)

![](figure/differential_expression.Rmd/ma_plot_Immature.fraction.of.reticulocytes_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Immature.fraction.of.reticulocytes_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Immature.fraction.of.reticulocytes_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Immature.fraction.of.reticulocytes_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Immature.fraction.of.reticulocytes_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Immature.fraction.of.reticulocytes_M2.png)

## Lymphocyte.count

![](figure/differential_expression.Rmd/ma_plot_Lymphocyte.count.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Lymphocyte.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_Lymphocyte.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Lymphocyte.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Lymphocyte.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Lymphocyte.count.png)

![](figure/differential_expression.Rmd/ma_plot_Lymphocyte.count_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Lymphocyte.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Lymphocyte.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Lymphocyte.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Lymphocyte.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Lymphocyte.count_M2.png)

## Lymphocyte.percentage.of.white.cells

![](figure/differential_expression.Rmd/ma_plot_Lymphocyte.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Lymphocyte.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_Lymphocyte.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Lymphocyte.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Lymphocyte.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Lymphocyte.percentage.of.white.cells.png)

![](figure/differential_expression.Rmd/ma_plot_Lymphocyte.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Lymphocyte.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Lymphocyte.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Lymphocyte.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Lymphocyte.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Lymphocyte.percentage.of.white.cells_M2.png)

## Mean.corpuscular.hemoglobin.concentration

![](figure/differential_expression.Rmd/ma_plot_Mean.corpuscular.hemoglobin.concentration.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Mean.corpuscular.hemoglobin.concentration.png) 
![](figure/differential_expression.Rmd/volcano_plot_Mean.corpuscular.hemoglobin.concentration.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Mean.corpuscular.hemoglobin.concentration.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Mean.corpuscular.hemoglobin.concentration.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.corpuscular.hemoglobin.concentration.png)

![](figure/differential_expression.Rmd/ma_plot_Mean.corpuscular.hemoglobin.concentration_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Mean.corpuscular.hemoglobin.concentration_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Mean.corpuscular.hemoglobin.concentration_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Mean.corpuscular.hemoglobin.concentration_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Mean.corpuscular.hemoglobin.concentration_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.corpuscular.hemoglobin.concentration_M2.png)

## Mean.corpuscular.hemoglobin

![](figure/differential_expression.Rmd/ma_plot_Mean.corpuscular.hemoglobin.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Mean.corpuscular.hemoglobin.png) 
![](figure/differential_expression.Rmd/volcano_plot_Mean.corpuscular.hemoglobin.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Mean.corpuscular.hemoglobin.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Mean.corpuscular.hemoglobin.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.corpuscular.hemoglobin.png)

![](figure/differential_expression.Rmd/ma_plot_Mean.corpuscular.hemoglobin_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Mean.corpuscular.hemoglobin_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Mean.corpuscular.hemoglobin_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Mean.corpuscular.hemoglobin_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Mean.corpuscular.hemoglobin_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.corpuscular.hemoglobin_M2.png)

## Mean.corpuscular.volume

![](figure/differential_expression.Rmd/ma_plot_Mean.corpuscular.volume.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Mean.corpuscular.volume.png) 
![](figure/differential_expression.Rmd/volcano_plot_Mean.corpuscular.volume.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Mean.corpuscular.volume.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Mean.corpuscular.volume.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.corpuscular.volume.png)

![](figure/differential_expression.Rmd/ma_plot_Mean.corpuscular.volume_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Mean.corpuscular.volume_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Mean.corpuscular.volume_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Mean.corpuscular.volume_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Mean.corpuscular.volume_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.corpuscular.volume_M2.png)

## Mean.platelet.volume

![](figure/differential_expression.Rmd/ma_plot_Mean.platelet.volume.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Mean.platelet.volume.png) 
![](figure/differential_expression.Rmd/volcano_plot_Mean.platelet.volume.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Mean.platelet.volume.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Mean.platelet.volume.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.platelet.volume.png)


![](figure/differential_expression.Rmd/ma_plot_Mean.platelet.volume_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Mean.platelet.volume_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Mean.platelet.volume_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Mean.platelet.volume_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Mean.platelet.volume_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.platelet.volume_M2.png)

## Mean.reticulocyte.volume

![](figure/differential_expression.Rmd/ma_plot_Mean.reticulocyte.volume.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Mean.reticulocyte.volume.png) 
![](figure/differential_expression.Rmd/volcano_plot_Mean.reticulocyte.volume.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Mean.reticulocyte.volume.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Mean.reticulocyte.volume.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.reticulocyte.volume.png)

![](figure/differential_expression.Rmd/ma_plot_Mean.reticulocyte.volume_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Mean.reticulocyte.volume_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Mean.reticulocyte.volume_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Mean.reticulocyte.volume_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Mean.reticulocyte.volume_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.reticulocyte.volume_M2.png)

## Mean.sphered.corpuscular.volume

![](figure/differential_expression.Rmd/ma_plot_Mean.sphered.corpuscular.volume.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Mean.sphered.corpuscular.volume.png) 
![](figure/differential_expression.Rmd/volcano_plot_Mean.sphered.corpuscular.volume.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Mean.sphered.corpuscular.volume.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Mean.sphered.corpuscular.volume.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.sphered.corpuscular.volume.png)

![](figure/differential_expression.Rmd/ma_plot_Mean.sphered.corpuscular.volume_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Mean.sphered.corpuscular.volume_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Mean.sphered.corpuscular.volume_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Mean.sphered.corpuscular.volume_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Mean.sphered.corpuscular.volume_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Mean.sphered.corpuscular.volume_M2.png)

## Monocyte.count

![](figure/differential_expression.Rmd/ma_plot_Monocyte.count.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Monocyte.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_Monocyte.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Monocyte.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Monocyte.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Monocyte.count.png)

![](figure/differential_expression.Rmd/ma_plot_Monocyte.count_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Monocyte.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Monocyte.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Monocyte.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Monocyte.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Monocyte.count_M2.png)

## Monocyte.percentage.of.white.cells

![](figure/differential_expression.Rmd/ma_plot_Monocyte.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Monocyte.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_Monocyte.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Monocyte.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Monocyte.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Monocyte.percentage.of.white.cells.png)

![](figure/differential_expression.Rmd/ma_plot_Monocyte.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Monocyte.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Monocyte.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Monocyte.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Monocyte.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Monocyte.percentage.of.white.cells_M2.png)

## Neutrophil.count

![](figure/differential_expression.Rmd/ma_plot_Neutrophil.count.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Neutrophil.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_Neutrophil.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Neutrophil.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Neutrophil.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Neutrophil.count.png)

![](figure/differential_expression.Rmd/ma_plot_Neutrophil.count_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Neutrophil.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Neutrophil.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Neutrophil.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Neutrophil.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Neutrophil.count_M2.png)

## Neutrophil.percentage.of.white.cells

![](figure/differential_expression.Rmd/ma_plot_Neutrophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Neutrophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_Neutrophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Neutrophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Neutrophil.percentage.of.white.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Neutrophil.percentage.of.white.cells.png)

![](figure/differential_expression.Rmd/ma_plot_Neutrophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Neutrophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Neutrophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Neutrophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Neutrophil.percentage.of.white.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Neutrophil.percentage.of.white.cells_M2.png)

## Platelet.count

![](figure/differential_expression.Rmd/ma_plot_Platelet.count.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Platelet.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_Platelet.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Platelet.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Platelet.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Platelet.count.png)

![](figure/differential_expression.Rmd/ma_plot_Platelet.count_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Platelet.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Platelet.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Platelet.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Platelet.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Platelet.count_M2.png)

## Platelet.crit

![](figure/differential_expression.Rmd/ma_plot_Platelet.crit.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Platelet.crit.png) 
![](figure/differential_expression.Rmd/volcano_plot_Platelet.crit.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Platelet.crit.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Platelet.crit.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Platelet.crit.png)

![](figure/differential_expression.Rmd/ma_plot_Platelet.crit_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Platelet.crit_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Platelet.crit_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Platelet.crit_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Platelet.crit_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Platelet.crit_M2.png)

## Platelet.distribution.width

![](figure/differential_expression.Rmd/ma_plot_Platelet.distribution.width.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Platelet.distribution.width.png) 
![](figure/differential_expression.Rmd/volcano_plot_Platelet.distribution.width.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Platelet.distribution.width.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Platelet.distribution.width.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Platelet.distribution.width.png)

![](figure/differential_expression.Rmd/ma_plot_Platelet.distribution.width_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Platelet.distribution.width_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Platelet.distribution.width_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Platelet.distribution.width_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Platelet.distribution.width_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Platelet.distribution.width_M2.png)

## Red.blood.cell.count

![](figure/differential_expression.Rmd/ma_plot_Red.blood.cell.count.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Red.blood.cell.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_Red.blood.cell.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Red.blood.cell.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Red.blood.cell.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Red.blood.cell.count.png)

![](figure/differential_expression.Rmd/ma_plot_Red.blood.cell.count_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Red.blood.cell.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Red.blood.cell.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Red.blood.cell.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Red.blood.cell.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Red.blood.cell.count_M2.png)

## Red.cell.distribution.width

![](figure/differential_expression.Rmd/ma_plot_Red.cell.distribution.width.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Red.cell.distribution.width.png) 
![](figure/differential_expression.Rmd/volcano_plot_Red.cell.distribution.width.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Red.cell.distribution.width.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Red.cell.distribution.width.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Red.cell.distribution.width.png)

![](figure/differential_expression.Rmd/ma_plot_Red.cell.distribution.width_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Red.cell.distribution.width_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Red.cell.distribution.width_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Red.cell.distribution.width_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Red.cell.distribution.width_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Red.cell.distribution.width_M2.png)

## Reticulocyte.count

![](figure/differential_expression.Rmd/ma_plot_Reticulocyte.count.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Reticulocyte.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_Reticulocyte.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Reticulocyte.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Reticulocyte.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Reticulocyte.count.png)

![](figure/differential_expression.Rmd/ma_plot_Reticulocyte.count_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Reticulocyte.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Reticulocyte.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Reticulocyte.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Reticulocyte.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Reticulocyte.count_M2.png)

## Reticulocyte.fraction.of.red.cells

![](figure/differential_expression.Rmd/ma_plot_Reticulocyte.fraction.of.red.cells.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Reticulocyte.fraction.of.red.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_Reticulocyte.fraction.of.red.cells.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Reticulocyte.fraction.of.red.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Reticulocyte.fraction.of.red.cells.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Reticulocyte.fraction.of.red.cells.png)

![](figure/differential_expression.Rmd/ma_plot_Reticulocyte.fraction.of.red.cells_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Reticulocyte.fraction.of.red.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Reticulocyte.fraction.of.red.cells_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Reticulocyte.fraction.of.red.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Reticulocyte.fraction.of.red.cells_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Reticulocyte.fraction.of.red.cells_M2.png)

## White.blood.cell.count

![](figure/differential_expression.Rmd/ma_plot_White.blood.cell.count.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_White.blood.cell.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_White.blood.cell.count.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_White.blood.cell.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_White.blood.cell.count.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_White.blood.cell.count.png)

![](figure/differential_expression.Rmd/ma_plot_White.blood.cell.count_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_White.blood.cell.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_White.blood.cell.count_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_White.blood.cell.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_White.blood.cell.count_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_White.blood.cell.count_M2.png)

## Celiac_GCST90014442

![](figure/differential_expression.Rmd/ma_plot_Celiac_GCST90014442.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Celiac_GCST90014442.png) 
![](figure/differential_expression.Rmd/volcano_plot_Celiac_GCST90014442.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Celiac_GCST90014442.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Celiac_GCST90014442.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Celiac_GCST90014442.png)

![](figure/differential_expression.Rmd/ma_plot_Celiac_GCST90014442_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Celiac_GCST90014442_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Celiac_GCST90014442_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Celiac_GCST90014442_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Celiac_GCST90014442_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Celiac_GCST90014442_M2.png)

## Celiac_GCST90468120

![](figure/differential_expression.Rmd/ma_plot_Celiac_GCST90468120.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Celiac_GCST90468120.png) 
![](figure/differential_expression.Rmd/volcano_plot_Celiac_GCST90468120.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Celiac_GCST90468120.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Celiac_GCST90468120.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Celiac_GCST90468120.png)

![](figure/differential_expression.Rmd/ma_plot_Celiac_GCST90468120_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_Celiac_GCST90468120_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_Celiac_GCST90468120_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_Celiac_GCST90468120_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_Celiac_GCST90468120_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_Celiac_GCST90468120_M2.png)

## IBD_GCST90013901

![](figure/differential_expression.Rmd/ma_plot_IBD_GCST90013901.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_IBD_GCST90013901.png) 
![](figure/differential_expression.Rmd/volcano_plot_IBD_GCST90013901.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_IBD_GCST90013901.png) 
![](figure/differential_expression.Rmd/enrichment_plot_IBD_GCST90013901.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_IBD_GCST90013901.png)

![](figure/differential_expression.Rmd/ma_plot_IBD_GCST90013901_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_IBD_GCST90013901_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_IBD_GCST90013901_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_IBD_GCST90013901_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_IBD_GCST90013901_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_IBD_GCST90013901_M2.png)

## IBD_GCST90013951

![](figure/differential_expression.Rmd/ma_plot_IBD_GCST90013951.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_IBD_GCST90013951.png) 
![](figure/differential_expression.Rmd/volcano_plot_IBD_GCST90013951.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_IBD_GCST90013951.png) 
![](figure/differential_expression.Rmd/enrichment_plot_IBD_GCST90013951.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_IBD_GCST90013951.png)

![](figure/differential_expression.Rmd/ma_plot_IBD_GCST90013951_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_IBD_GCST90013951_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_IBD_GCST90013951_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_IBD_GCST90013951_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_IBD_GCST90013951_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_IBD_GCST90013951_M2.png)

## LUPUS_GCST003156

![](figure/differential_expression.Rmd/ma_plot_LUPUS_GCST003156.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_LUPUS_GCST003156.png) 
![](figure/differential_expression.Rmd/volcano_plot_LUPUS_GCST003156.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_LUPUS_GCST003156.png) 
![](figure/differential_expression.Rmd/enrichment_plot_LUPUS_GCST003156.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_LUPUS_GCST003156.png)

![](figure/differential_expression.Rmd/ma_plot_LUPUS_GCST003156_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_LUPUS_GCST003156_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_LUPUS_GCST003156_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_LUPUS_GCST003156_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_LUPUS_GCST003156_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_LUPUS_GCST003156_M2.png)

## LUPUS_GCST011096

![](figure/differential_expression.Rmd/ma_plot_LUPUS_GCST011096.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_LUPUS_GCST011096.png) 
![](figure/differential_expression.Rmd/volcano_plot_LUPUS_GCST011096.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_LUPUS_GCST011096.png) 
![](figure/differential_expression.Rmd/enrichment_plot_LUPUS_GCST011096.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_LUPUS_GCST011096.png)

![](figure/differential_expression.Rmd/ma_plot_LUPUS_GCST011096_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_LUPUS_GCST011096_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_LUPUS_GCST011096_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_LUPUS_GCST011096_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_LUPUS_GCST011096_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_LUPUS_GCST011096_M2.png)

## T1D_GCST90000529

![](figure/differential_expression.Rmd/ma_plot_T1D_GCST90000529.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_T1D_GCST90000529.png) 
![](figure/differential_expression.Rmd/volcano_plot_T1D_GCST90000529.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_T1D_GCST90000529.png) 
![](figure/differential_expression.Rmd/enrichment_plot_T1D_GCST90000529.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_T1D_GCST90000529.png)

![](figure/differential_expression.Rmd/ma_plot_T1D_GCST90000529_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_T1D_GCST90000529_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_T1D_GCST90000529_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_T1D_GCST90000529_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_T1D_GCST90000529_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_T1D_GCST90000529_M2.png)

## T1D_GCST90014023

![](figure/differential_expression.Rmd/ma_plot_T1D_GCST90014023.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_T1D_GCST90014023.png) 
![](figure/differential_expression.Rmd/volcano_plot_T1D_GCST90014023.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_T1D_GCST90014023.png) 
![](figure/differential_expression.Rmd/enrichment_plot_T1D_GCST90014023.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_T1D_GCST90014023.png)

![](figure/differential_expression.Rmd/ma_plot_T1D_GCST90014023_M2.png) 
![](figure/differential_expression.Rmd/ma_plot_quantile_T1D_GCST90014023_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_T1D_GCST90014023_M2.png) 
![](figure/differential_expression.Rmd/volcano_plot_quantile_T1D_GCST90014023_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_T1D_GCST90014023_M2.png) 
![](figure/differential_expression.Rmd/enrichment_plot_quantile_T1D_GCST90014023_M2.png)
