---
title: "Differential expression"
output: html_document
---

```{r setup, include=FALSE, message = F, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(data.table)
library(dplyr)
library(ggplot2)
library(corrplot)
library(DESeq2)
library(SummarizedExperiment)
library(BiocParallel)
library(ggrepel)
```

# Correlation between PRS & PCs

```{r}
# load prs & pcs
metadata_file <- "analysis/metadata.txt"
metadata <- read.csv(metadata_file, header = T, sep = "\t", stringsAsFactors = T)
metadata$sex <- as.factor(metadata$sex)

traits <- metadata[, 7:43]
pc <- metadata[, 1:5]

# Calculate the correlation between each trait and each PC
correlation_matrix <- cor(traits, pc)
range(correlation_matrix)
correlation_matrix <- t(correlation_matrix)

# Create the heatmap using corrplot
corrplot(correlation_matrix, method = "color", 
         col = colorRampPalette(c("blue", "white", "red"))(200), # color palette
         addCoef.col = "black", # Add correlation coefficients to the plot
         number.cex = 0.4, # Adjust the font size of the numbers
         tl.col = "black", # text label color
         tl.srt = 90, # rotate text labels
         tl.cex = 0.5, 
         title = "Correlation between Traits and PCs", 
         mar = c(0, 0, 1, 0)
)
```

# Perform DESeq2 differential expression analysis for each trait

```{r, warning=FALSE, eval=FALSE}
# Load the gene expression data
gene_expr_file <- "data/gene_reads_2017-06-05_v8_whole_blood.gct"
raw_count_df <- fread(gene_expr_file, header = TRUE, sep = "\t", drop = "id")

id <- raw_count_df$Name
raw_count <- raw_count_df[, -c(1:2)]

# modify GTEx sample names matching names used in PRS data
colnames(raw_count) <- sub("^(GTEX-[^-.]+).*", "\\1", colnames(raw_count))

matching_samples <- intersect(rownames(metadata), colnames(raw_count))
final_count <- raw_count[ , ..matching_samples]

# prefilter: keep only rows that have a count of at least 10 for a minimal number of samples
keep_genes <- rowSums(final_count >= 10) >= 100
final_count <- final_count[keep_genes, ]
id <- id[keep_genes]

# Loop through each trait and run DESeq2
for (trait in colnames(traits)) {
  
  # Create the DESeqDataSet for the current trait
  dds <- DESeqDataSetFromMatrix(
    countData = as.matrix(final_count),  # Raw counts
    colData = metadata[, c(1:6, which(colnames(metadata) == trait))],
    design = as.formula(paste("~ PC1 + PC2 + PC3 + PC4 + PC5 + sex +", trait))
  )
  rownames(dds) <- id
  
  # Run DESeq2 analysis
  dds <- DESeq(dds, parallel = TRUE, BPPARAM = MulticoreParam(4))
  
  # Get the results for the current trait
  res <- results(dds)
  
  # Save the results to a file
  write.csv(res, paste0("differential_expression_", trait, "_results.csv"))
  
  # print a summary of the results
  print(paste("Results for trait:", trait))
  print(summary(res))
  
  # plot the MA-plot for the current trait
  png(paste0("ma_plot_", trait, ".png"), width = 800, height = 600)
  plotMA(res, main = paste("MA Plot for", trait))
  dev.off()
  
  # volcano plot
  res_tableOE <- as.data.frame(res)
  res_tableOE$gene_name <- raw_count_df$Description[match(rownames(res_tableOE), id)]
  res_tableOE <- mutate(res_tableOE, threshold_OE = padj < 0.05 & 
                          abs(log2FoldChange) >= 1)
  res_tableOE <- res_tableOE %>% arrange(padj) %>% mutate(genelabels = "")
  res_tableOE$genelabels[1:10] <- res_tableOE$gene_name[1:10]
  
  volcano_plot <- ggplot(res_tableOE, aes(x = log2FoldChange, y = -log10(padj))) +
    geom_point(aes(colour = threshold_OE)) +
    geom_text_repel(aes(label = genelabels)) +
    ggtitle(paste("Volcano Plot for", trait)) +
    xlab("log2 fold change") + 
    ylab("-log10 adjusted p-value") +
    theme(legend.position = "none",
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
  
  # Save the volcano plot
  png(paste0("volcano_plot_", trait, ".png"), width = 800, height = 600)
  print(volcano_plot)
  dev.off()
}
```

```{r}
for (trait in colnames(traits)) {
  res <- read.csv(paste0("analysis/differential_expression_", trait, "_results.csv"), row.names = 1, header = T)
  print(paste0(trait, ": number of significant differential genes is ", sum(res$padj < 0.05, na.rm=TRUE)))
}
```

![MA plot PGS000163](figure/differential_expression.Rmd/ma_plot_PGS000163.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000163.png)

![MA plot PGS000164](figure/differential_expression.Rmd/ma_plot_PGS000164.png)
![](figure/differential_expression.Rmd/volcano_plot_PGS000164.png)

![MA plot PGS000165](figure/differential_expression.Rmd/ma_plot_PGS000165.png)
![](figure/differential_expression.Rmd/volcano_plot_PGS000165.png)

![MA plot PGS000166](figure/differential_expression.Rmd/ma_plot_PGS000166.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000166.png)

![MA plot PGS000167](figure/differential_expression.Rmd/ma_plot_PGS000167.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000167.png)

![MA plot PGS000168](figure/differential_expression.Rmd/ma_plot_PGS000168.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000168.png)

![MA plot PGS000169](figure/differential_expression.Rmd/ma_plot_PGS000169.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000169.png)

![MA plot PGS000170](figure/differential_expression.Rmd/ma_plot_PGS000170.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000170.png)

![MA plot PGS000171](figure/differential_expression.Rmd/ma_plot_PGS000171.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000171.png)

![MA plot PGS000172](figure/differential_expression.Rmd/ma_plot_PGS000172.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000172.png)

![MA plot PGS000173](figure/differential_expression.Rmd/ma_plot_PGS000173.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000173.png)

![MA plot PGS000174](figure/differential_expression.Rmd/ma_plot_PGS000174.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000174.png)

![MA plot PGS000175](figure/differential_expression.Rmd/ma_plot_PGS000175.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000175.png)

![MA plot PGS000176](figure/differential_expression.Rmd/ma_plot_PGS000176.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000176.png)

![MA plot PGS000177](figure/differential_expression.Rmd/ma_plot_PGS000177.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000177.png)

![MA plot PGS000178](figure/differential_expression.Rmd/ma_plot_PGS000178.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000178.png)

![MA plot PGS000179](figure/differential_expression.Rmd/ma_plot_PGS000179.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000179.png)

![MA plot PGS000180](figure/differential_expression.Rmd/ma_plot_PGS000180.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000180.png)

![MA plot PGS000181](figure/differential_expression.Rmd/ma_plot_PGS000181.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000181.png)

![MA plot PGS000182](figure/differential_expression.Rmd/ma_plot_PGS000182.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000182.png)

![MA plot PGS000183](figure/differential_expression.Rmd/ma_plot_PGS000183.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000183.png)

![MA plot PGS000184](figure/differential_expression.Rmd/ma_plot_PGS000184.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000184.png)

![MA plot PGS000185](figure/differential_expression.Rmd/ma_plot_PGS000185.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000185.png)

![MA plot PGS000186](figure/differential_expression.Rmd/ma_plot_PGS000186.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000186.png)

![MA plot PGS000187](figure/differential_expression.Rmd/ma_plot_PGS000187.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000187.png)

![MA plot PGS000188](figure/differential_expression.Rmd/ma_plot_PGS000188.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000188.png)

![MA plot PGS000189](figure/differential_expression.Rmd/ma_plot_PGS000189.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000189.png)

![MA plot PGS000190](figure/differential_expression.Rmd/ma_plot_PGS000190.png) 
![](figure/differential_expression.Rmd/volcano_plot_PGS000190.png)

![MA plot Celiac_GCST90014442](figure/differential_expression.Rmd/ma_plot_Celiac_GCST90014442.png) 
![](figure/differential_expression.Rmd/volcano_plot_Celiac_GCST90014442.png)

![MA plot Celiac_GCST90014442](figure/differential_expression.Rmd/ma_plot_Celiac_GCST90014442.png) 
![](figure/differential_expression.Rmd/volcano_plot_Celiac_GCST90014442.png)

![MA plot Celiac_GCST90468120](figure/differential_expression.Rmd/ma_plot_Celiac_GCST90468120.png) 
![](figure/differential_expression.Rmd/volcano_plot_Celiac_GCST90468120.png)

![MA plot IBD_GCST90013901](figure/differential_expression.Rmd/ma_plot_IBD_GCST90013901.png) 
![](figure/differential_expression.Rmd/volcano_plot_IBD_GCST90013901.png)

![MA plot IBD_GCST90013951](figure/differential_expression.Rmd/ma_plot_IBD_GCST90013951.png) 
![](figure/differential_expression.Rmd/volcano_plot_IBD_GCST90013951.png)

![MA plot LUPUS_GCST003156](figure/differential_expression.Rmd/ma_plot_LUPUS_GCST003156.png) 
![](figure/differential_expression.Rmd/volcano_plot_LUPUS_GCST003156.png)

![MA plot LUPUS_GCST011096](figure/differential_expression.Rmd/ma_plot_LUPUS_GCST011096.png) 
![](figure/differential_expression.Rmd/volcano_plot_LUPUS_GCST011096.png)

![MA plot T1D_GCST90000529](figure/differential_expression.Rmd/ma_plot_T1D_GCST90000529.png) 
![](figure/differential_expression.Rmd/volcano_plot_T1D_GCST90000529.png)

![MA plot T1D_GCST90014023](figure/differential_expression.Rmd/ma_plot_T1D_GCST90014023.png) 
![](figure/differential_expression.Rmd/volcano_plot_T1D_GCST90014023.png)

