---
title: "expression_normalization"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(data.table)
library(dplyr)
library(DESeq2)
library(preprocessCore)
library(ggplot2)
```

# Raw count
```{r}
# Load the gene expression data
gene_expr_file <- "data/gene_reads_2017-06-05_v8_whole_blood.gct"
raw_count_df <- fread(gene_expr_file, header = TRUE)

gene_name <- raw_count_df$Description
raw_count_df <- raw_count_df[, -c(1:3)]

# modify GTEx sample names matching names used in PRS data
colnames(raw_count_df) <- sub("^(GTEX-[^-.]+).*", "\\1", colnames(raw_count_df))
```

# Match samples between PRS and count matrix
```{r}
# load PRS data for blood trait
prs_file <- "analysis/prs_blood_cell.txt"
prs_blood <- fread(prs_file, header = T, stringsAsFactors = FALSE)

# obtain GTEx sample id
id <- prs_blood$sample_id

# Load the PRS data for immune trait
prs_file <- "analysis/prs_immune.txt"
prs_immune <- fread(prs_file, header = T, stringsAsFactors = FALSE)

# subset samples
matching_samples <- intersect(id, colnames(raw_count_df))
prs_blood <- prs_blood[prs_blood$sample_id %in% matching_samples, ]
prs_immune <- prs_immune[prs_immune$sample_id %in% matching_samples, ]
final_count <- raw_count_df[,  ..matching_samples]
dim(final_count) 
```

# Stratify by PRS level (High vs Median vs Low)
## Blood trait
```{r}
# Loop through each column (starting from the second column, as the first column is sample_id)
for (i in 2:ncol(prs_blood)) {
  # Extract the PRS values for the current trait
  prs_trait <- prs_blood[[i]]

  # Calculate the 25th and 75th percentiles for the trait
  p25 <- quantile(prs_trait, 0.25, na.rm = TRUE)
  p75 <- quantile(prs_trait, 0.75, na.rm = TRUE)

  # Create a new column for the group classification based on the percentiles
  group <- ifelse(prs_trait > p75, "High", 
                  ifelse(prs_trait < p25, "Low", "Median"))

  # Assign the group to the new column for this trait
  prs_blood[[paste0("group_", colnames(prs_blood)[i])]] <- as.factor(group)
}
```

## Immune trait
```{r}
# Loop through each column (starting from the second column, as the first column is sample_id)
for (i in 2:ncol(prs_immune)) {
  # Extract the PRS values for the current trait
  prs_trait <- prs_immune[[i]]

  # Calculate the 25th and 75th percentiles for the trait
  p25 <- quantile(prs_trait, 0.25, na.rm = TRUE)
  p75 <- quantile(prs_trait, 0.75, na.rm = TRUE)

  # Create a new column for the group classification based on the percentiles
  group <- ifelse(prs_trait > p75, "High", 
                  ifelse(prs_trait < p25, "Low", "Median"))

  # Assign the group to the new column for this trait
  prs_immune[[paste0("group_", colnames(prs_immune)[i])]] <- as.factor(group)
}
```

# DESeq2 normalization by median of ratios

The size factor is calculated as follows:

1. For each gene, the geometric mean of counts across all samples is computed (this serves as the pseudo baseline expression).

2. For each gene, the ratio of its count in a specific sample to the pseudo-baseline expression is calculated (e.g., Sample A/pseudo baseline, Sample B/pseudo baseline).

3. For each sample, the median of these ratios is computed, which results in the size factor for that sample.

Thus, DESeq2 normalizes for both sequencing depth and RNA composition differences, and this process is independent of the design matrix, meaning it is unaffected by the specific traits (whether blood or immune traits) used in the analysis.

```{r}
# Create the DESeq2 object (DESeqDataSet) from the raw count matrix and PRS
dds <- DESeqDataSetFromMatrix(countData = as.matrix(final_count), 
                              colData = prs_blood, 
                              design = ~ group_PGS000163)
rownames(dds) <- gene_name

# remove genes with no reads
keep_genes <- rowSums(counts(dds)) > 0
dds <- dds[keep_genes, ]
dds

# Estimate size factors for normalization
dds <- estimateSizeFactors(dds)
outliers <- which(sizeFactors(dds) > 4)
plot(sizeFactors(dds), colSums(counts(dds)), ylab = "library sizes", 
     xlab = "size factors", cex = .5)
text(sizeFactors(dds)[outliers], colSums(counts(dds))[outliers], 
     labels = colnames(dds)[outliers], pos = 2, cex = .5, col = "red")

# obtain normalized counts
normalized_counts <- counts(dds, normalized=TRUE)
#write.table(normalized_counts, "analysis/normalized_counts.txt", 
#            sep="\t", row.names = T, col.names = T)
```

# Quantile normalization

```{r}
# load TPM counts data
tpm_file <- "data/gene_tpm_2017-06-05_v8_whole_blood.gct.gz"
tpm_count_df <- fread(tpm_file, header = TRUE, stringsAsFactors = FALSE)
colnames(tpm_count_df) <- sub("^(GTEX-[^-.]+).*", "\\1", colnames(tpm_count_df))

# subset samples & remove no read genes
tpm_subset_df <- tpm_count_df[, matching_samples, with = FALSE]
tpm_subset_df <- tpm_subset_df[rowSums(tpm_subset_df[, -1, with = FALSE]) > 0, ]

# convert into matrix
tpm_count <- as.matrix(tpm_subset_df)

# perform quantile normalization
qn_counts <- preprocessCore::normalize.quantiles(tpm_count)
colnames(qn_counts) <- matching_samples
```


# Compare raw counts, DESeq2 normalized count & quantile normalized count
```{r}
boxplot(log2(counts(dds) + 1)[, c(1:10, outliers)], 
        main = "Raw read counts", ylab = "log2(read counts)", cex = .6, 
        col = c(rep("lightblue", 10), "red"), las = 2) 

boxplot(log2(counts(dds, normalize = TRUE) + 1)[, c(1:10, outliers)], 
        main = "DESeq2 normalized read counts", ylab = "log2(read counts)", 
        cex = .6, col = c(rep("lightblue", 10), "red"), las = 2)

boxplot(log2(qn_counts + 1)[, c(1:10, outliers)], 
        main = "Quantile normalized read counts", ylab = "log2(read counts)", 
        cex = .6, col = c(rep("lightblue", 10), "red"), las = 2) 
```

